\chapter{Computational morphology-driven cellular archetypes}

\section{Contribution and disclaimer}

In this chapter I use the method developed and applied in the last chapter to two distinct cohorts of \ac{wbs} and develop methods that enable the prediction of clinically relevant conditions from the distributions of features within \ac{wbs} and using individual cells. I also outline \ac{mile-vice}, a weakly-supervised \ac{ml} framework to cluster cells into virtual cell types, and show how this can be used to derive cellular archetypes that are enriched in specific conditions, enabling the discovery of novel and clinically relevant morphological phenotypes.

\section{Introduction}

\ac{mds} are a heterogeneous group of myeloid neoplasms. The characterization of \ac{mds} encompasses several distinct clinical observations: consistently decreased numbers of mature blood cells (cytopenia) associated with a decreased ability of the bone marrow to produce blood cells (bone marrow failure), identification of blood cells from one or more specific lineages with abnormal cytomorphology (dysplasia) and an increase in the probability of \ac{aml} onset due to genetic instability \cite{Valent2017-uh,Hofmann2005-vv,Aster2020-cu}. In other words, \ac{mds} are simultaneously an oncological malignancy and a precursor to more severe forms of cancer.

\paragraph{Expert diagnostic of myelodysplastic syndromes.} The diagnosis of \ac{mds} is done through a number of complementary assessments and analyses. \Ac{cbc}, generally obtained through the use of an automated blood counter and can detect cytopenias (reduction in circulating mature blood cells) such as anaemias, leukopenias and/or thrombocytopenias (cytopenias affecting \ac{rbc}, \ac{wbc} and platelets, respectively), all of which are common in \ac{mds} \cite{Najean1989-qm,Campo2017-wi}. Megaloblastic anaemias, in particular, can be indicative of or easily confounded with \ac{mds} \cite{Kaferle2009-pl,Vasekova2016-vo,Corey2007-cs}. The analysis of \ac{wbs} or bone marrow slides with different stains allows the detection of specific dysplastic features in blood cells. The former allows the characterization of abnormalities such as granulocytes with reduced nuclear segmentation, abnormal granularity in neutrophils or abnormal \ac{rbc} (such as macrocytes, elliptocytes and dacrocytes) \cite{Campo2017-wi,Langenhuijsen1984-qx,Kuriyama1986-ts,Davey1988-zn}. The analysis of bone marrow slides enables the identification of abnormal cell maturation or an excessive count of blasts \cite{Aster2020-cu}. Cytochemistry and immunohistochemistry can also be used with bone marrow slides and allows the identification of, among others, ring sideroblasts, which define a subtype of \ac{mds} \cite{Campo2017-wi,Mufti2008-ye}. Additionally, flow cytometry, cytogenetic characterization and molecular sequencing are necessary for the classification of \ac{mds} \cite{Aster2020-cu,Porwit2014-zi,Greenberg2012-en}, with multiparametric flow cytometry being often times considered essential for the diagnosis of \ac{mds} \cite{Cremers2016-fs}.

\paragraph{Confounders in \ac{mds} diagnostic and the relevance of \ac{sf3b1} mutation status in prognostic.} \Ac{mds} can be complicated to diagnose when little information is available --- as mentioned, anaemias are a common symptom of \ac{mds}, but can be caused by multiple reasons such as nutrient deficiency \cite{Castle1978-ky,Short2013-pz,Aslinia2006-en}. In other words, anaemias can be a symptom of \ac{mds} or caused by other, non-oncological factors and diagnosing between anaemias and \ac{mds} from a \ac{wbs} alone is non-trivial and unfeasible in a clinical setting. Additionally, \ac{mds} subtypes characterized by specific mutations can be hard to characterize when no molecular sequencing or bone marrow slides are available. Such a \ac{mds} subtype is \ac{sf3b1}, characterized clinically by longer survival times \cite{Malcovati2020-no} and morphologically by the presence of ring sideroblasts in the bone marrow using specific stains that allow their identification \cite{Hellstrom_Lindberg2015-zs}. However, no morphological phenotypes have been identified for \ac{sf3b1}-mutant \ac{mds} in the blood. Considering the clinical relevance of this mutation, it is important to better understand the cytomorphological manifestations of \ac{sf3b1}-mutant \ac{mds} in the blood. Computational methods such as the ones outlined in Chapter 4 and in the present Chapter can represent a strategy to identify novel cellular archetypes for specific conditions.

\paragraph{Computational cytomorphology approaches to \ac{mds} diagnostic.} A recent work showed how previously established and expert-derived cytomorphological descriptions of \ac{wbs} could be used to classify \ac{mds} into specific and genomically-relevant subtypes with limited concordance in a validation cohort \cite{Nagata2020-lh}. While this approach can be potentially useful, it represents a laborious and time-consuming task for the trained expert. Additionally, the subpar inter-individual concordance in classifying cell types can hinder the possibility for finer-grained identification of cytomorphological determinants of \ac{mds} subtypes, novel or otherwise. Other studies have shown how the analysis of bone marrow slides using \ac{dl} can be leveraged to not only classify diseases, but also to detect the presence of specific mutations in both \ac{aml} and \ac{mds} \cite{Bruck2021-fx,Eckardt2021-fb}. These approaches are mostly possible due to the elevated density of nucleated cells in bone marrow: this guarantees that high resolution images (50x) with a relatively small size (2560*1920 pixels) are representative of the whole bone marrow as in \cite{Eckardt2021-fb}.  \etal{Bruck} also use this high cellular density to their advantage by splitting relatively large bone marrow slides into small image tiles and training a \ac{dl} model at this level (224x224 and 299x299) \cite{Bruck2021-fx}. The low density of nucleated cells in \ac{wbs} makes these approaches unfeasible.

In this work, I used computational cytomorphology to predict different conditions and identify novel cellular phenotypes associated with each. For this, I defined four distinct tasks to assess how well computational cytomorphology could be used to predict different clinical conditions based on the data available from the MLLC:

\begin{enumerate}
    \item Disease detection --- identifying the presence of either anaemia or \ac{mds} (easy for trained experts);
    \item Disease classification --- distinguishing between anaemia and \ac{mds} (hard for trained experts --- anaemia can be a symptom of \ac{mds});
    \item \ac{sf3b1}mut \ac{mds} detection --- distinguishing between \ac{sf3b1}-mutant \ac{mds} and other subtypes of \ac{mds} (hard for trained experts --- no known morphological manifestation of \ac{sf3b1} mutations exist);
    \item Anaemia classification --- distinguishing between iron deficiency anaemia and megaloblastic anaemia (easy for trained experts).
\end{enumerate}

In addition to the cytomorphological features already described, I also had access to a few \ac{cbc} parameters, particularly \ac{wbcc} (\ac{wbc}/uL), haemoglobin concentration (g/dL) and platelet counts (platelets/uL). Here, I also assessed their impact on the predictive performance of all the classification methods. For clarity, I do not report training metrics and focus exclusively on cross-validated metrics or external validation metrics.

\section{The MLLC recapitulates blood count trends observed in other studies}

To confirm that the MLLC is a representative cohort, I start by analysing a few simple aspects of this cohort. Particularly, I start by confirming that individuals with \ac{mds} are generally much older than the rest of the population with a particular bias towards male individuals, as previously reported \cite{Rollison2008-yg}. Indeed, the chance of having \ac{mds} in our cohort increases by 12\% every year, with male individuals being more than twice as likely to contract \ac{mds} (according to a binomial regression with binary \ac{mds} status as the dependent variable and age and sex as the covariates, $p=8.47*10^{-16}$ and $p=0.000173$, respectively; \figref{fig:mds-age} and \figref{fig:mds-sex}). While recapitulating these patterns confirms that our cohort is representative, it also prevents us from using both parameters as features in predictive modelling as they might bias the results.

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/mllc-age.pdf}
    \placecaption{Age distribution for the different conditions in MLLC.}
    \label{fig:mds-age}
\end{figure}

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/mllc-sex.pdf}
    \placecaption{Sex distribution for the different conditions in MLLC.}
    \label{fig:mds-sex}
\end{figure}

I additionally observe that individuals with either \ac{mds} or anaemia have a tendency to be leukopenic, with individuals having approximately 1,200 and 1,800 fewer \ac{wbc}/uL in these conditions, respectively (according to a linear regression where WBC count is the dependent variable and binary status as \ac{mds} and anaemia are the dependent variables, $p=0.0379$ and $p=0.0088$, respectively; \figref{fig:mds-wbcc}). However, this leukopenic tendency in anaemias is driven by individuals with megaloblastic anaemia --- indeed, whereas iron-deficient individuals are indistinguishable from normal individuals, individuals with megaloblastic anaemia have approximately 3,200 WBC/uL fewer than normal individuals ($p=6.224*10^{-14}$ for a two sample t-test) --- an effect that has been reported elsewhere and that can be associated with folic acid or vitamin B12 deficiencies \cite{Kaferle2009-pl,Castle1978-ky} and that further highlights how megaloblastic anaemias can be confounded with \ac{mds}. Haemoglobin concentrations are also much lower in \ac{mds} and, expectedly, anaemias (\figref{fig:mds-hb}) --- indeed, these individuals have 4.34 and 6.38 fewer haemoglobin g/dL, respectively (according to a linear regression where Hb concentration is the dependent variable and binary status as \ac{mds} and anaemia are the dependent variables, $p<2*10^{-16}$ and $p<2*10^{-16}$, respectively). No detectable difference between normal individuals and individuals with either \ac{mds} or anaemia is observable when it comes to platelet counts, but it should be noted that individuals with megaloblastic anaemia have approximately an expected 146,000 fewer platelets/uL than normal individuals ($p=3.312*10^{-12}$ for a two sample t-test; \figref{fig:mds-plt}), concordant with previous reports \cite{Castle1978-ky}. 

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/mllc-wbcc.pdf}
    \placecaption{White blood cell count distribution for the different conditions in MLLC.}
    \label{fig:mds-wbcc}
\end{figure}

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/mllc-hb.pdf}
    \placecaption{Haemoglobin concentration distribution for the different conditions in MLLC.}
    \label{fig:mds-hb}
\end{figure}

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/mllc-platelet.pdf}
    \placecaption{Platelet count distribution for the different conditions in MLLC.}
    \label{fig:mds-plt}
\end{figure}

Finally, I also observe that \ac{sf3b1}-mutant \ac{mds} presents clinically distinct features from other \ac{mds} subtypes --- particularly, WBC and platelet counts are indistinguishable from those of normal individuals while being much higher than those found in other \ac{mds} subtypes ($p=0.3372$ and $p=0.1506$ for two sample t-tests comparing \ac{wbcc} and platelet counts between individuals with \ac{sf3b1}-mutant \ac{mds} and normal individuals; $p=0.001734$ and $p<2.2*10^{-16}$ for two sample t-tests comparing \ac{wbcc} and platelet counts, respectively, between individuals with \ac{sf3b1}-mutant \ac{mds} and individuals with other \ac{mds} subtypes in our cohort). This is in agreement with what has been previously described in terms of higher platelet and absolute neutrophil counts in \ac{sf3b1}-mutant \ac{mds} \cite{Malcovati2015-tz,Malcovati2020-no} and further highlights the relevance of being able to discriminate between this and other \ac{mds} subtypes. 

\section{Classification of clinically-relevant conditions with morphometric distributions}

I define morphometric distributions as parametric characterizations of the distribution of each feature within each \ac{wbs}. To do so, I calculate the mean and variance of all individual features retrieved from \ac{rbc} and \ac{wbc} collections. By doing so, I obtain a relatively simple characterization of the distribution of each feature, which can be used in predictive modelling tasks as features. To ensure that my estimates for the mean and variance of each feature are adequate, I exclude all cases where fewer than 50 \ac{wbc} or \ac{rbc} were detected. Consequently, for the classification methods, my input was, for each individual, a morphometric distribution --- a vector containing the mean and variance of each feature. This constitutes, for the cytomorphological features, a total of ($53 + 42) * 2 = 190$ features considering both \ac{wbc} (mean and variance of 53 feaures) and \ac{rbc} (mean and variance of 42 feaures) which, when combined with the \ac{cbc} data, yielded a total of 193 features. To get different baselines, I defined three datasets --- one composed of \ac{cbc} features (3 features), one composed of cytomorphological features (190) and one composed of both feature types (193). 

\subsection{Independent assessment of morphometric distributions}

To get a sense on whether the retrieved features could be useful in classification, I first aggregated them on a per-individual basis by calculating the mean and variance of each feature for each cell type (\ac{rbc} and \ac{wbc}), generating what I call morphometric distributions. I then used a relatively simple statistical pipeline to test whether there were significant differences between different clinical conditions on MLLC (normal individuals, individuals with iron deficiency anaemia, individuals with megaloblastic anaemia, individuals with \ac{sf3b1} \ac{mds} and individuals with other \ac{mds} subtypes): first, I perform a Kruskal-Wallis test --- an extension of the Mann-Whitney U-test (a sum of ranks based non-parametric test that tests whether a set of samples originate from the same distribution) for multiple samples --- on each individual feature. Then, considering only the features for which the Kruskal-Wallis test was significant after correcting for multiple testing, I performed a post-hoc Dunn test --- a pairwise sum of ranks test for multiple groups that tests if each group pair originates from the same distribution --- to determine how many features are relevant for each comparison (if $p<0.05$ after multiple testing correction).

This analysis shows that, for most cases, a considerable amount of features has sufficient discriminatory power to discriminate between conditions (\figref{fig:dunn-test-heatmap}). I note here that for the remainder of this paragraph I consider a feature to be significant if either its mean or variance is significant. Identifying the presence of a clinically-relevant condition is the easier task, with at least 42 features (79.2\%) and 22 features (52.4\%) offering discriminatory power for \ac{wbc} and \ac{rbc}, respectively. Additionally, discriminating between \ac{mds} and anaemia is a relatively easy task (at least 38 features (71.7\%) and 20 features (47.6\%) for \ac{wbc} and \ac{rbc}, respectively). Finally, it is worth noting that the greatest challenge --- i.e. the comparison for which there are fewer significant features --- lies in the discrimination between \ac{sf3b1}-mutant \ac{mds} and other \ac{mds} subtypes, with 19 features (35.8\%) and 23 features (54.8\%) for \ac{wbc} and \ac{rbc}, respectively. 

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/dunn-test-heatmap.pdf}
    \placecaption{Number of statistically significant features which offer good discriminating power for \ac{wbc} means (top left), \ac{wbc} variance (top right), \ac{rbc} mean (bottom left) and \ac{rbc} variance (bottom right). Colours represent the percentage of features which are statistically-significant.}
    \label{fig:dunn-test-heatmap}
\end{figure}

Finally, inspecting the distributions of the two most discriminating features (largest Kruskal-Wallis statistic) reveals already striking differences between features depending on the condition --- regarding the mean of \ac{wbc} features, a decrease on both the ellipse variance (i.e. how distinct a shape is from an ellipse) and the first moment of the \ac{cdf} (a relatively abstract quantifier of shape irregularity) are typical of anaemia or \ac{mds} ($p=2.03*10^{-14}$ and $p=8.44*10^{-16}$ for Mann-Whitney tests comparing the distributions of the means of the ellipse variance and the first moment of the \ac{cdf}, respectively, for normal individuals and individuals with a clinically-relevant condition) and can be used to discriminate between both types of anaemia ($p=3.53*10^{-4}$ and $p=1.72*10^{-4}$ for Mann-Whitney tests comparing distributions of the means of the ellipse variance and the first moment of the \ac{cdf}, respectively, for individuals with either iron deficiency or megaloblastic anaemia; \figref{fig:feature-distribution-mean} (top)), whereas the mean of \ac{rbc} features shows that individuals with \ac{sf3b1}-mutant \ac{mds} have \ac{rbc} with larger perimeters (which can be indicative of large size or shape irregularities) and smaller minimum values for the \ac{cdf} (which can be indicative of smaller sizes or shape irregularities; $p=3.14*10^{-13}$ and $p=9.78*10^{-6}$ for Mann-Whitney tests comparing the distributions of the perimeter and minimum value of the \ac{cdf}, respectively; \figref{fig:feature-distribution-mean} (bottom)).

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/top-features-kw.pdf}
    \placecaption{Examples of the two most discriminating features for \ac{wbc} (top) and \ac{rbc} (bottom) stratified by clinically-relevant condition.}
    \label{fig:feature-distribution-mean}
\end{figure}

\subsection{Morphometric distributions as features in supervised learning}

\subsubsection{Methods}

These results show the potential of morphometric distributions for diagnostic. However, they also reveal complex relationships between features --- in \ac{rbc}, the perimeter and minimum value of the \ac{cdf} can both be indicative of size but are increased and decreased, respectively, when comparing \ac{sf3b1}-mutant \ac{mds} with other \ac{mds} subtypes. To study these relationships, I started by splitting the data into 5 non-overlapping test sets using a stratified cross-validation scheme, thus ensuring that each training and validation sets had a sufficient number of each class. Following this, I preprocessed the data on each training set in order to standardize it and remove features which are highly correlated with other features ($|R|>0.9$). Additionally, and considering that \ac{cbc} was available for all individuals, I imputed the missing values for \ac{cbc} data using the median for each parameter. I train two different models for each round of cross-validation --- a binomial regression model with an elastic network regularization (a weighted combination of the $L1$ and $L2$ regularizations) as implemented in the \texttt{glmnet} package for R \cite{Friedman2010-gl} and a random forest model \cite{Breiman2001-yz}. Random forests are ensemble models which combine the weighted predictions from several decision trees trained on a random subset of data and features. The random nature of the training makes them extremely robust to overfitting since it prevents a subset of data or of features from having a disproportionate amount of effect on the trained model. To deal class imbalance I weighted the objective of each sample of $\mathrm{class}_i$ by $1-\frac{\mathrm{number\ of\ elements\ in\ class}_i}{\mathrm{total\ number\ of\ elements}}$. 

After training each model, I calculate the \ac{auroc} by using the concatenated results from all cross-validation rounds to obtain the cross-validated \ac{auroc} \cite{Fawcett2006-mo}. However, for inference and assessment of feature importance I select the best performing model out of each set of cross-validated models. I assessed the feature importance for the binomial regression by calculating the standardized value for each coefficient and for the random forests by using the mean decrease in accuracy as recommended by Breiman in the original paper describing random forests \cite{Breiman2001-yz}. To assess the importance of a group of features --- the total explained variance for each group --- I use the trained binomial regression model and calculate the effect of each group of variables by summing the effects of the standardized variables composing it for each individual. Finally, I calculated the total explained variance by each group of features by calculating the sum of the rows in the covariance matrix between all 5 groups of features --- \ac{wbc} feature means, \ac{wbc} feature variances, \ac{rbc} feature means, \ac{rbc} feature variances and \ac{cbc} parameters.

\subsubsection{Morphometric distributions predict clinically-relevant conditions}

Firstly, I assessed which of the models offered the best performance as this can reveal important aspects of how features are associated with one another. Particularly, while a \ac{rf} model is capable of capturing both linear and non-linear relationships, an elastic network model assumes that the relationship between each variable and the label is linear. In \figref{fig:glmnet-vs-rf-auc} and using both the morphometric distributions and blood count data I show that the elastic regression model outperforms the \ac{rf} model for disease detection and classification, while being slightly worst for \ac{sf3b1}mut detection. Additionally, both perform equally well on the task of anaemia classification. The poorer performance of the \ac{rf} model suggests that the relationships between features are unlikely to be non-linear. I carry on with the elastic regression model, which has the added advantage of allowing the simple calculation of feature group importance.

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/glmnet-vs-rf-auc.pdf}
    \placecaption{Cross-validated \ac{auroc} for the elastic regression model (glmnet; green) and randomized random forest model (RRF; dark red) for the different tasks considered. The average value for each model-task combination is shown as a slanted square.}
    \label{fig:glmnet-vs-rf-auc}
\end{figure}

I then trained models using only blood counts, only morphology (morphometric profiles), or both. I show that morphology performs similarly or outperforms blood counts, with a considerable improvement when considering disease classification and disease detection (\figref{fig:roc-curves-binary} and \figref{fig:auc-binary}), where using only blood counts for prediction performs considerably worst. Importantly, I note that in all cases the performance is best when using both types of data. Interestingly, using blood counts alone is relatively effective at detecting the presence of a \ac{sf3b1} mutation in an individual with \ac{mds} (AUC = 82.9\%), a performance that is driver largely by the increased platelet counts in individuals with \ac{sf3b1}-positive \ac{mds}.

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/roc-curves-binary.pdf}
    \placecaption{Cross-validated receiver operating characteristic curves for the four tasks considered. Coloured depending on the data type used --- blood counts (B.C.), morphology or both (morphology + B.C.).}
    \label{fig:roc-curves-binary}
\end{figure}

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/auc-binary.pdf}
    \placecaption{\Ac{auroc} values for the four tasks considered. Coloured depending on the data type used --- blood counts (B.C.), morphology or both (morphology + B.C.).}
    \label{fig:auc-binary}
\end{figure}

Inspecting how different features impact prediction using both blood counts and morphology can reveal some important trends. For instance and for disease detection, lower haemoglobin levels, an excess of average \ac{rbc} eccentricity (the ratio between major and minor axes) and average \ac{wbc} nuclear green mass displacement (measures how much the red colour distribution is shifted from the centre, thus quantifying nuclear irregularities) contributes considerably to the "normal" classification (\figref{fig:feature-importance-disease-detection-classification} (left)). Within individuals with a disease (anaemia or \ac{mds}), it becomes clear that low platelet and \ac{wbc} counts, together with high haemoglobin concentration are hallmarks of \ac{mds}. An increase in the average \ac{rbc} perimeter, decrease in the average \ac{wbc} nuclear eccentricity, an increase in the variance of \ac{wbc} solidity (the ratio between the object and its convex hull, quantifying deformations to the shape) can also be important in diagnostic of \ac{mds} (\figref{fig:feature-importance-disease-detection-classification} (right)).

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/feature-importance-disease-detection-classification.pdf}
    \placecaption{Coefficients for the five most important features in the elastic network regression (glmnet) model by feature group (blood counts (B.C.), \ac{rbc} mean, \ac{rbc} variance, \ac{wbc} mean and \ac{wbc} variance) for disease detection (left) and disease classification (right).}
    \label{fig:feature-importance-disease-detection-classification}
\end{figure}

As for the discrimination between disease subtypes shows that, for the most part, high platelet counts are indicative of \ac{sf3b1}-mutant \ac{mds} (\figref{fig:feature-importance-mds}) as observed earlier in \figref{fig:mds-plt}. Additionally, an increase in the average \ac{rbc} perimeter is also predictive of \ac{sf3b1}-mutant \ac{mds}. Regarding the classification of different anaemia subtypes, an increase in platelet counts is expected in iron deficiency anaemia, whereas an increase in the \ac{rbc} average minimum of the minor axis peak profile (a measure of the colour distribution along the major axis of the cell) is expected in megaloblastic anaemia \figref{fig:feature-importance-anaemia}. This can point towards a greater prevalence of spherocytes in iron deficiency anaemia. Additionally, a seemingly unintuitive finding which will be further confirmed ahead is that morphological aspects of \ac{wbc} --- particularly concerning the nuclei of \ac{wbc} --- contribute heavily towards the classification of different subtypes of anaemia. However, this can be explained by the fact that hyperlobulated neutrophils --- neutrophils with more than 5 nuclear segments --- are known to be overrepresented in \ac{wbs} in megaloblastic anaemia cases \cite{Hariz2021-qw}.

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/feature-importance-mds.pdf}
    \placecaption{Coefficients for the important features in the elastic network regression (glmnet; abs(coefficient) > 0) model by feature group (blood counts (B.C.), \ac{rbc} mean, \ac{rbc} variance, \ac{wbc} mean and \ac{wbc} variance) for \ac{sf3b1} mutation detection.}
    \label{fig:feature-importance-mds}
\end{figure}

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/feature-importance-anaemia.pdf}
    \placecaption{Coefficients for the important features in the elastic network regression (glmnet; abs(coefficient) > 0) model by feature group (blood counts (B.C.), \ac{rbc} mean, \ac{rbc} variance, \ac{wbc} mean and \ac{wbc} variance) for \ac{sf3b1} mutation detection.}
    \label{fig:feature-importance-anaemia}
\end{figure}

Finally, looking at how groups of features impact prediction shows that morphometric distributions account for most of the variance explained by these predictive models, with aspects pertaining to both mean and variance playing a significant role (\figref{fig:feature-group-importance}). An obvious exception is the detection of \ac{sf3b1} mutations in individuals with \ac{mds} --- here, it is blood counts, particularly platelet counts, that drive the classification, with \ac{wbc} playing a practically negligible part. This may be due to the fact that \ac{wbc} abnormalities are present only in a few cells in \ac{mds}, making the use of relatively generic distribution descriptors (i.e. mean and variance) not adequate for classification.

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/feature-group-importance.pdf}
    \placecaption{Feature group importance for the prediction of different conditions by feature group (blood counts (B.C.), \ac{rbc} mean, \ac{rbc} variance, \ac{wbc} mean and \ac{wbc} variance). Each circle is scaled according to the explained variance.}
    \label{fig:feature-group-importance}
\end{figure}

Finally, reconsidering this problem as a multiple class problem and training models for each data type further highlights the value of morphology in classification --- the multiple class \ac{auroc} is considerably higher morphology (91.3\%) than for blood counts only (83.6\%; \figref{fig:multiclass-auc}), an effect driven largely by the poor predictive ability of blood counts in discriminating between \ac{sf3b1}-mutant \ac{mds} (AUC = 33\%) and between megaloblastic anaemia and non-\ac{sf3b1}-mutant anaemia (AUC = 72\%; \figref{fig:multiclass-auc-heatmap}). As is the case for the binary classification tasks, using both blood counts and morphology as features offers the best performance (AUC = 93.6\%).

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/multiclass-auc.pdf}
    \placecaption{Multiple class \ac{auroc}.}
    \label{fig:multiclass-auc}
\end{figure}

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/multiclass-auc-heatmap.pdf}
    \placecaption{Multiple class pairwise \ac{auroc} for different tasks and data types.}
    \label{fig:multiclass-auc-heatmap}
\end{figure}

This section covers the potential of morphometric profiles in haematological diagnostic. However, it also highlights an important caveat of this method --- while it is relatively simple to understand what an increase on the average perimeter of \ac{rbc} (as observable in cases of \ac{sf3b1}-mutant \ac{mds} when compared with other \ac{mds} subtypes) or even a decrease in the variability of solidity in \ac{wbc} (as observable in cases of \ac{mds} when compared with anaemias), some of these features do not translate immediately to evident visual features, especially when the variation of these features in a single individual is the relevant aspect of classification. Additionally, as is the case for \ac{sf3b1}mut detection, alterations may be present in a small subset of cells, rendering morphometric distributions less suitable to capturing relevant morphological signatures. For this reason, I developed \ac{mile-vice}, a method capable of creating visually coherent groups of cells which are relevant for disease classification with minimal human input.

\section{MILe-ViCe --- Multiple Instance Learning for Virtual Cell quantification}

\subsection{Methods}

\subsubsection{Motivation}

Each slide, from the blood cell detection pipeline specified above, is described as a set of $m$ RBC and $n$ WBC. Typically, when diagnosing \ac{mds}, an expert focuses on identifying abnormal cell types and on quantifying their relative prevalence \cite{Valent2017-uh}. In more abstract terms, given a set of objects, the expert has to characterize and classify each object and use this information to infer the prevalence of specific abnormalities in this set of objects, thus obtaining a classification for the set of objects that is dependent on the prevalence of different classes. In other words, this is essentially a problem of \ac{mil} --- a learning task that focuses on the characterization of objects (or instances) in a group to predict a classification for that group. Some \ac{mil} methods combine this group classification task with the auxiliary task of learning a function that maps instances to a vocabulary and uses the presence or frequency of instances in each “concept” of the vocabulary to predict a classification for the group \cite{Amores2013-ym}. Having this abstraction is helpful to devise an approach that is not only able to classify \ac{wbs} into separate classes (i.e. “normal” vs. “disease”) but also to classify each individual instance into specific concepts (or, in my case, cells into specific virtual cells) that are helpful in the classification process. 

\subsubsection{Multiple instance learning}

While traditional algorithms for supervised \ac{ml}, are dedicated to learning a function that maps a single instance to a class (a one-to-one mapping such as a picture of a cat to the class "cat"), \ac{mil}, a type of supervised \ac{ml}, learns how to map a set of instances to a single class. Essentially, the objective of \ac{mil} algorithms is to take a set of data points --- a bag --- and assign this bag to a class, rather than each individual data point to a class. It is possible to split these methods based on how they handle each individual element (whether they are independent or whether they should be classified individually, for example) and based on how the classification for the bag is calculated (whether the classification should depend on the presence/absence of specific elements or on their relative proportions or whether the interest lies instead in detecting similarity within each bag) \cite{Amores2013-ym,Carbonneau2016-xc}. While problems such as protein sequence classification or sentiment analysis are good examples of \ac{mil} when the bags are inherently structured, I am considerably more interested in the case where no structure is identifiable --- indeed, I am interested in classifying a "bag of cells" stratified by automatically detected \ac{wbc} and \ac{rbc} in a \ac{wbs}. Not only that, but I am also interested in the determination of a vocabulary --- a correspondence between the features that characterize a cell and a virtual cell type, a computational construct that is morphologically coherent --- that is relevant for the classification task. In the ideal case scenario, this vocabulary-based method would map individual cells to specific classes (virtual cell types) and use the proportion of each virtual cell type to classify a bag as having specific haematological conditions, thus combining good predictive performance with the identification of relevant virtual cell types. The main advantage of this is that no it allows for the discovery of novel cytomorphological signatures which can be adapted to the clinic.

\subsubsection{Model specification}

Consider a \ac{wbs} $W$ characterized as a set of $n$ cells $W=\{c_1,c_2,...,c_n\}$, with each cell characterized by a vector of $f$ features. A function $C$ maps each cell $c_i$ to a concept (virtual cell) $v_i$ given a vocabulary $theta$ with $m$ concepts such that $v_i = C(c_i,\theta)$, with $\sum_{j=1}^m{v_i} = 1$ --- in other words, $v_i$ describes the membership of $c_i$ to a set of $m$ concepts. Using this, each \ac{wbs} is characterized as a set of $n$ concept memberships $W'=\{v_1,v_2,...,v_n\}$. Having identified the concept to which each cell belongs to, it is now possible to calculate the proportion of concepts in $W$ by calculating $V = \frac{1}{n}\sum_{i=1}^n{v_i}$. This vector of proportions can now be used to calculate the probability $p$ of $W$ having been prepared from an individual with one of $k$ conditions such that $p = P(V)$.

I am interested in learning $C$ and $P$ at the same time, thus enabling this approach to both predict conditions from specific slides and learn which virtual cell types are relevant to do so. Because of this, I need to define $C$ and $P$ in a way that allows concurrent optimization. I also need to consider that there are \ac{wbs} with tens or hundreds of thousands of detected  cells. Because of this, I am interested in selecting a method that enables randomly selected subsets of cells to be used in each training iteration in such a way that converges towards an optimal solution.  Finally, I have to consider the fact that I am interested in characterizing \ac{wbs} and \ac{rbc}. To this end, I parametrize $C_{\mathrm{WBC}}$, $C_{\mathrm{RBC}}$ and $P$ as linear functions with softmax transforms, such that $C_{\mathrm{WBC}}$ and $C_{\mathrm{RBC}}$ takes vectors of features characterizing \ac{wbc} $\mathrm{wbc}_i$ and \ac{rbc} $\mathrm{rbc}_j$, respectively, with each producing a vector of length $m$ containing the probabilities of $\mathrm{wbc}_i$ and $\mathrm{rbc}_j$ belonging to a specific virtual cell from two distinct vocabularies characterizing \ac{wbc} and \ac{rbc} --- $v_{\mathrm{WBC}(i)}$ and $v_{\mathrm{RBC}(j)}$, respectively. $P$ takes the concatenated vector of virtual cell proportions for \ac{wbc} and \ac{rbc} $V=\mathrm{concatenate}([v_{\mathrm{WBC}(i)},v_{\mathrm{RBC}(j)}])$ and produces a prediction $p$. 

This formulation also enables the concatenation of additional information --- such as blood counts --- to $V$ and guarantees that the entire system can be optimized using stochastic gradient descent, also enabling the utilization of randomly selected subsets of \ac{wbc} and \ac{rbc} on each training step. The missing element in this system is defining the optimal number of virtual cell types; I consider this to be a hyperparameter in this model. In \figref{fig:mile-vice-schematic}, I present a schematic representation of this model.

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/mile-vice-schematic.pdf}
    \placecaption{Schematic representation of the \ac{mile-vice} algorithm. White and red blood cells (WBC and RBC, respectively) are classfied as virtual cell types and the relative proportions of virtual cell types are used to classify an individual as having a specific and clinically relevant condition.}
    \label{fig:mile-vice-schematic}
\end{figure}

\subsubsection{Model optimization}

I test each task using different numbers of virtual cell types --- $[10, 25, 50]$ --- assuming that both \ac{wbc} and \ac{rbc} can be clustered into the same number of virtual cells. I also test whether blood counts (WBC counts (cells/uL), haemoglobin concentration (g/dL) and platelet counts (platelets/uL)) improve the classification performance of \ac{mile-vice}. I standardize (mean = 0 and standard deviation = 1) all additional continuous variables --- \ac{wbcc}, haemoglobin concentration, platelet counts or age. 

From solving these four tasks separately, I can expect the best possible outcome to be four separate models, each of which is primed to solve its task and to identify virtual cell types which are relevant for the detection of a specific condition. However, I was also interested in the virtual cell types which are exclusively useful for one task; to this effect, I train additional models under two specific settings --- one that is a multiclass model, similar to that describe in the previous section, and one that focuses on the concurrent optimization of all four tasks using multi-objective learning. Regarding the multi-objective learning setting, in each training step and for each of the four tasks I sampled a set of relevant \ac{wbs} to be considered, predicted the outcome class and calculated the cross-entropy. I then add all four loss values after multiplying them by a vector sampled from a Dirichlet distribution with concentration [1,1,1,1], a trick proposed in \cite{Ruchte2021-ch} to approximate Pareto optimality (i.e. no improvement can be done to one task without making a different task worse \cite{Censor1977-nd}) and backpropagate this value. For this scenario I also test the inclusion of blood counts and demographic information in the final vector $V$.

To select the best model in each setting I use the cross-validated \ac{auroc}. To get these cross-validated scores, each model is trained over 5 folds and the average value and standard error of each metric is calculated. In each fold, a model is optimized for a maximum 15,000 steps with a learning rate of 0.01 and a weight decay of 0.25. I use a weighted cross-entropy loss and the Adam optimizer \cite{Kingma2014-zd}. The weight for the positive class is calculated as $1 - \frac{\mathrm{no.\ of\ elements\ in\ positive\ class}}{\mathrm{total\ number\ of\ elements}}$. Additionally, the learning rate is decreased every time the training loss stagnates for at least 100 steps by a factor of 0.5. If the learning rate reaches a value of 0.000001 (smaller than the initial learning rate by 4 orders of magnitude). On each training iteration I sample without replacement $500$ \ac{wbc} and $500$ \ac{rbc}. Whenever fewer than $500$ cells were detected, I sample with replacement. This may lead to a situation where bags of cells with a smaller amount of information (fewer cells) contribute equally to those with a large amount of cells. To minimize this, I weigh each instance with a weight $w_{\mathrm{avail}}$ calculated from the number of available \ac{rbc} $m$ and \ac{wbc} $n$ such that $w_{\mathrm{avail}} = 1 + \frac{\mathrm{minimum}(500,m) + \mathrm{minimum}(500,n)}{500*2}$.

Here, I trained \ac{mile-vice} models to perform the same tasks as those in the previous section described above this Chapter ---  disease detection, disease classification, \ac{sf3b1}mut \ac{mds} detection and anaemia classification, as well as a multi-objective scenario where the same set of cells is used to optimize different binary tasks. For clarity, I will also call the binary class tasks as "single objective" tasks. To reduce the search space for \ac{mile-vice}, I select the subset of features whose mean or variance had a coefficient with an absolute value greater than 0.01 in the elastic network regression model.

\subsection{MILe-ViCe predicts clinically-relevant conditions}

\ac{mile-vice} has to assume that cells can be clustered into a given number of virtual cells. I tested three different values for the number of virtual cells in the binary classification tasks (10, 25 and 50) and two different values for the multi-objective task (25 and 50). I also test whether including blood count data can help improve results \figref{fig:mile-vice-performance}. The number of virtual cells has an impact on the median cross-validation \ac{auroc}:

\begin{itemize}
    \item In the \textbf{single objective} setting using \textbf{morphology}, the best performing models are generally the ones with 25 virtual cells excluding the case of anaemia classification, for which the best performing model considers 10 virtual cells;
    \item In the \textbf{single objective} setting using \textbf{morphology and blood counts}, the best performing models have 50 virtual cells excluding the case of anaemia classification, for which the best performance is observed for the model considering 10 virtual cells;
    \item In the \textbf{multiple objective} setting using \textbf{morphology}, the best performing models have 50 virtual cells excluding the case of anaemia classification. In this case, the best performing model considers 25 virtual cells;
    \item In the \textbf{multiple objective} setting using \textbf{morphology and blood counts}, the best performing models have 50 virtual cells excluding the case of disease classification, for which the best performing model has 25 virtual cells.
\end{itemize}

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/mile-vice-cv-performance.pdf}
    \placecaption{\Ac{auroc} for \ac{mile-vice} models with different numbers of virtual cells and using different types of data --- "Morphology" (only using image features retrieved from cells in the slide) and "Morphology + B.C." (using image features from cells and blood counts). The models with the best performance for each set of virtual cell numbers are highlighted with black rectangles.}
    \label{fig:mile-vice-performance}
\end{figure}

While performance does not vary widely between different numbers of virtual cells, this analysis highlights that for some tasks a smaller number of virtual cells is beneficial --- this may be associated with the over-parametrization of models with higher numbers of virtual cells in specific tasks. Additionally, I also note that the multi-objective scenario leads to slightly worse performance (\figref{fig:mile-vice-performance} and \figref{fig:mile-vice-roc-curves}). However, the multi-objective has a clear advantage --- while the single-objective performance can be slightly better, it also offers a less compact representation, with a higher number of possible virtual cells.

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/mile-vice-roc-curve.pdf}
    \placecaption{\Ac{auroc} for \ac{mile-vice} models with different numbers of virtual cells and using different types of data --- "Morphology" (only using image features retrieved from cells in the slide) and "Morphology + B.C." (using image features from cells and blood counts). The models with the best performance for each set of virtual cell numbers are highlighted with black rectangles.}
    \label{fig:mile-vice-roc-curves}
\end{figure}

Comparing \ac{mile-vice} with the elastic linear regression models trained on morphometric distributions shows that \ac{mile-vice} offers better performance (\figref{fig:mile-vice-vs-glmnet} and \figref{fig:mile-vice-vs-glmnet-scatter}) in both the single and multiple objective settings. The only exception is for the task of disease classification, where using \ac{mile-vice} with morphology and blood counts leads to slightly worse performance in both single and multiple objective settings, and using using \ac{mile-vice} with morphology leads to slightly worse performance in the multiple objective setting. Overall, however, I show here that \ac{mile-vice} offers comparable performance while avoiding aggregating morphometric data in a class-agnostic way. 

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/mile-vice-vs-glmnet.pdf}
    \placecaption{Barplot comparing the cross-validated \ac{auroc} between \ac{mile-vice} and the elastic linear regression model trained on morphometric features (glmnet).}
    \label{fig:mile-vice-vs-glmnet}
\end{figure}

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/mile-vice-vs-glmnet-scatter.pdf}
    \placecaption{Scatterplot comparing the cross-validated \ac{auroc} between \ac{mile-vice} and the elastic linear regression model trained on morphometric features (glmnet).}
    \label{fig:mile-vice-vs-glmnet-scatter}
\end{figure}

\subsection{Virtual cell inspection and validation}

After showing that \ac{mile-vice} is capable of discriminating different clinically relevant conditions, performing almost always better than the elastic network regression model using the distribution parameters of differnt morphometric features, I inspect specific virtual cell groups to better understand which ones are contributing the most towards classification. To this effect, I retrieve the absolute coefficient values for the classification layer from the best performing \ac{mile-vice} model for each task and multiply it by the cell proportions in the MLLC cohort, thus obtaining an effect-size for each virtual cell type. I then use this effect size to rank the most and least important virtual cell types, taking the 5 most important cells for classification (the 5 cells with the highest absolute effect size). The final step is the careful observation of the more important virtual cell types (the virtual cell types with the highest effect size) in order to determine how these cell types relate to cytomorphological phenotypes and define cellular (\ac{rbc} and \ac{wbc}) archetypes. I refer here to "morphology-only model" as the models which consider only morphological aspects of cells, and "complete model" as the model which, apart from morphology, also considers \ac{cbc}. 

\subsubsection{Disease detection} 

In disease detection (normal vs. disease), it is evident that \ac{mile-vice} does a good job at descriminating between cell types which are more prevalent in each condition when using only morphology and when using both morphology and \ac{cbc} (B.C.; \figref{fig:mile-vice-vcq-so-disease-detection}). More particularly, when considering \ac{rbc}, relatively large \ac{rbc} (high perimeter) are one of the \ac{rbc} archetypes detected by \ac{mile-vice} as more prevalent in disease when compared with normal cases (virtual \ac{rbc} types 25, 42 and 5 in the morphology-only model and 44, 16, 35 and 19 in the complete model; \figref{fig:rbc-disease-detection-examples}). Additionally, rounder \ac{rbc} (low eccentricity) are \ac{rbc} archetypes of the normal classification (virtual \ac{rbc} types 45 in the morphology-on ly model and 37 in the complete model \figref{fig:rbc-disease-detection-examples}). However, it should be noted that \ac{rbc} homogeneity makes it complicated to assign specific and well-defined \ac{rbc} archetypes. These \ac{rbc} archetypes make sense --- a hallmark of \ac{mds} and megaloblastic anaemia is indeed enlarged \ac{rbc}, while the presence of non-circular \ac{rbc} (poikilocytosis) is a general hallmark of aberrant \ac{rbc} production as observed in both \ac{mds} and anaemia \cite{Ford2013-nd}. 

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/mile-vice-vcq-so-disease-detection.pdf}
    \placecaption{The effect size and average proportion ratio for the five most relevant virtual cell types for both red blood cells (RBC) and white blood cells (WBC) for disease detection (disease vs. normal) when using only morphological features (Morphology) and when using both morphological features and complete blood counts (Morphology + B.C.). Positive effect values/higher proportion ratio are associated with normal cases (controls), whereas negative effect values/lower proportion ratio are associated with disease cases.}
    \label{fig:mile-vice-vcq-so-disease-detection}
\end{figure}

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/rbc-disease-detection-examples.pdf}
    \placecaption{Virtual red blood cell (RBC) type examples relevant for disease detection for the morphology-only model and the complete model.}
    \label{fig:rbc-disease-detection-examples}
\end{figure}

A relevant \ac{wbc} archetype can also be pointed out (\figref{fig:mile-vice-vcq-so-disease-detection}) --- virtual \ac{wbc} type 20 in the morphology-only model and 4 in the complete model contain neutrophils featuring nucelar abnormalities (pseudo-Pelger-Huët anomalies/band cells, characterized by either U- or dumbbell-shaped nuclei \cite{Bain2005-zg} and nuclear oversegmentation), whereas virtual cell type features, among other types of cells, a relatively high proportion of blasts \figref{fig:wbc-disease-detection-examples}. It is worth highlighting that, specifically for this task, the identified \ac{wbc} archetypes are somewhat varied, probably due to the considerable variability of \ac{wbc} in anaemias and, more particularly, in \ac{mds} \cite{Bain2005-zg,Bain2014-oc}. I must note here that distinguishing between some specific \ac{wbc} can be complicated --- particularly, band cells (immature and hypolobulated neutrophils, typically with U-shaped nuclei) and pseudo-Pelger-Huët anomalies (granulocytes with U- or dumbbell-shaped nuclei) are remarkably similar and their distinction is non-trivial --- as mentioned, both can be loosely characterized as granulocytes with U-shaped nuclei, with band cells having slightly larger nuclei and less densely packed chromatin (i.e. the staining appears lighter under the microscope) \cite{Colella2012-so}. These fine-grained differences are likely to go undifferentiated within \ac{mile-vice} in the task of discriminating \ac{mds} and anaemia, particularly since both cell types have been observed in cases of \ac{mds} \cite{Davey1988-zn,Cunningham1995-pc}.

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/wbc-disease-detection-examples.pdf}
    \placecaption{Virtual white blood cell (WBC) type examples relevant for disease detection for the morphology-only model and the complete model. Examples of hyperlobulated neutrophils and pseudo-Pelger-Huët anomalies/band cells are circled in red and green, respectively.}
    \label{fig:wbc-disease-detection-examples}
\end{figure}

Henceforth, I will focus on the virtual cell types important for the classification while using \ac{cbc} --- this allows me to focus aspects of condition prediction which are not captured by \ac{cbc} as these are relatively simple to obtain. 

\subsubsection{Disease classification} 

In disease classification (anaemia vs. \ac{mds}), an \ac{rbc} archetype typical of anaemia is the presence of dacrocytes (tear-shaped \ac{rbc}) and eliptocytes (elongated \ac{rbc}; 25 in the complete model), whereas larger \ac{rbc} are expected in \ac{mds} (15 and 23 in the complete model; \figref{fig:mile-vice-vcq-so-disease-classification} and \figref{fig:rbc-disease-classification-examples}). Regarding \ac{wbc} archetypes, the presence of normal lymphocytes is more prevalent in anaemia (18 in the complete model), whereas pseudo-Pelger-Huët anomalies/band cells and immature myeloid lineage \ac{wbc} (blasts) are more typical of \ac{mds} (9 and 24 in complete model; \figref{fig:wbc-disease-classification-examples}). Here it is important to highlight the fact that blasts are not a group of cells defined by a single set of characteristics --- indeed, a blast can any of the following: a deformable nucleus (in contrast with lymphocytes, were the nuclei are generally round), immature chromatin (textural variation within the nuclei), a relatively small cytoplasm-to-nucleus ratio and relatively large sizes (typically larger than lymphocytes) \cite{Bain2004-uq,Bain2005-zg,Bain2014-oc}.

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/mile-vice-vcq-so-disease-classification.pdf}
    \placecaption{The effect size and average proportion ratio for the five most relevant virtual cell types for both red blood cells (RBC) and white blood cells (WBC) for disease classification (anaemia vs. \ac{mds}) when using only morphological features (Morphology) and when using both morphological features and complete blood counts (Morphology + B.C.). Positive effect values/higher proportion ratio are associated with \ac{mds} cases, whereas negative effect values/lower proportion ratio are associated with anaemia cases.}
    \label{fig:mile-vice-vcq-so-disease-classification}
\end{figure}

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/rbc-disease-classification-examples.pdf}
    \placecaption{Virtual red blood cell (RBC) type examples relevant for disease classification for the complete model.}
    \label{fig:rbc-disease-classification-examples}
\end{figure}

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/wbc-disease-classification-examples.pdf}
    \placecaption{Virtual white blood cell (WBC) type examples relevant for disease classification for the complete model. Examples of pseudo-Pelger-Huët anomalies/band cells and immature myeloid lineage cells are circled in green and blue, respectively.}
    \label{fig:wbc-disease-classification-examples}
\end{figure}

\subsubsection{\ac{sf3b1}-mutation detection} 

The task of \ac{sf3b1}-mutation detection focuses on detecting which individuals with \ac{mds} have an \ac{sf3b1} mutation. \Ac{sf3b1} mutations are generally recognized as having a better prognosis and a fairly well determined cytomorphological phenotype in the bone marrow \cite{Malcovati2020-no,Hellstrom_Lindberg2015-zs,Malcovati2015-tz}. To the best of my knowledge, no cytomorphogical phenotype has been previously identified for \ac{sf3b1}-mutant \ac{mds} --- here I outline such a phenotype. 

Using \ac{mile-vice}, it is possible to detect both \ac{rbc} and \ac{wbc} archetypes in \ac{sf3b1}-mutant \ac{mds} (\figref{fig:mile-vice-vcq-so-disease-classification}) --- the most easily identifiable \ac{rbc} archetype is relatively large \ac{rbc}, which are more prevalent in \ac{sf3b1}-mutant \ac{mds}, while poikilocytic \ac{rbc} are more prevalent in other \ac{mds} subtypes \ac{mds} (15 and 2 in the complete model; \figref{fig:rbc-mds-classification-examples}). As for \ac{wbc}, the prevalence of pseudo-Pelger-Huët anomalies is higher in cases of \ac{sf3b1}-mutant \ac{mds} (virtual cell types 2 in the complete model \figref{fig:wbc-mds-classification-examples}). These are unlikely to be band cells due to the lack of granules in the cytoplasm, a hallmark of pseudo-Pelger-Huët anomalies.

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/mile-vice-vcq-so-mds-classification.pdf}
    \placecaption{The effect size and average proportion ratio for the five most relevant virtual cell types for both red blood cells (RBC) and white blood cells (WBC) for \ac{mds} classification (\ac{sf3b1}-mutant vs. non-\ac{sf3b1}-mutant \ac{mds}) when using only morphological features (Morphology) and when using both morphological features and complete blood counts (Morphology + B.C.). Positive effect values/higher proportion ratio are associated with non-\ac{sf3b1}-mutant \ac{mds} cases, whereas negative effect values/lower proportion ratio are associated with \ac{sf3b1}-mutant \ac{mds} cases.}
    \label{fig:mile-vice-vcq-so-mds-classification}
\end{figure}

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/rbc-mds-classification-examples.pdf}
    \placecaption{Virtual red blood cell (RBC) type examples relevant for \ac{mds} classification for the complete model.}
    \label{fig:rbc-mds-classification-examples}
\end{figure}

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/wbc-mds-classification-examples.pdf}
    \placecaption{Virtual white blood cell (WBC) type examples relevant for \ac{mds} classification for the complete model. Examples of pseudo-Pelger-Huët anomalies/band cells are circled in green.}
    \label{fig:wbc-mds-classification-examples}
\end{figure}

\subsubsection{Anaemia classification} 

The final task I used to evaluate the performance of \ac{mile-vice} was that of anaemia classification (iron deficiency vs. megaloblastic anaemia). Here, the \ac{rbc} archetypes are relatively simple as they all capture the expected aspects of both of these anaemias --- iron deficiency anaemias exhibit \ac{rbc} which are smaller, whereas \ac{rbc} in megaloblastic anaemia are considerably larger (virtual \ac{rbc} types 4/23 and 19, respectively, in the complete model; \figref{fig:rbc-anaemia-classification-examples}) and there is a larger fraction of poikilocytic \ac{rbc} in megaloblastic anaemia (virtual \ac{rbc} type 23; \figref{fig:rbc-anaemia-classification-examples}). As for \ac{wbc}, \ac{mile-vice} captures a fairly well known \ac{wbc} archetype in both megaloblastic anaemia and iron deficiency anemia --- hyperlobulated neutrophils (virtual \ac{wbc} type 14 and 12, respectively, in the complete model; \figref{fig:wbc-anaemia-classification-examples}) \cite{Lindenbaum1980-ux,Westerman1999-gs}. The main difference between both of these groups is that in megaloblastic anaemia hyperlobulated neutrophils (virtual cell type 14) are considerably larger than in iron deficiency (virtual cell type 12). While being a hallmark of both iron deficiency and megaloblastic anaemia, hyperlobulated neutrophils are larger in megaloblastic anaemia.

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/mile-vice-vcq-so-anaemia-classification.pdf}
    \placecaption{The effect size and average proportion ratio for the five most relevant virtual cell types for both red blood cells (RBC) and white blood cells (WBC) for anaemia classification (iron deficiency vs. megaloblastic anaemia) when using only morphological features (Morphology) and when using both morphological features and complete blood counts (Morphology + B.C.). Positive effect values/higher proportion ratio are associated with iron deficiency anaemia cases, whereas negative effect values/lower proportion ratio are associated with megaloblastic anaemia cases.}
    \label{fig:mile-vice-vcq-so-anaemia-classification}
\end{figure}

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/rbc-anaemia-classification-examples.pdf}
    \placecaption{Virtual red blood cell (RBC) type examples relevant for anaeamia classification for the complete model.}
    \label{fig:rbc-anaemia-classification-examples}
\end{figure}

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/wbc-anaemia-classification-examples.pdf}
    \placecaption{Virtual white blood cell (WBC) type examples relevant for anamemia classification for the complete model. Examples of pseudo-Pelger-Huët anomalies/band cells are circled in green.}
    \label{fig:wbc-anaemia-classification-examples}
\end{figure}

\subsubsection{Cross-objective importance} 

To better understand if there were virtual cell types which were transversally important, I used the multi-objective models described above and calculated the absolute median effect size for each virtual cell type, proceeding to filter all values smaller than $0.05$ (\figref{fig:multi-objective-importance}). Unsurprisingly, some of the aforementioned \ac{wbc} archetypes are once again evidenced --- hyperlobulated neutrophils (virtual cell type 1 in the complete model), and pseudo-Pelger-Huët anomalies (virtual cell type 19 in the complete model) come up once again as being among the most relevant for classification \figref{fig:multi-objective-cell-examples}. Regarding \ac{rbc}, some cellular archetypes are also picked up by the multi-objective \ac{mile-vice} model --- poikilocytic \ac{rbc} (with a relatively high prevalence of echinocytes --- \ac{rbc} with small and frequent membrane projections; virtual \ac{rbc} type 20) and small \ac{rbc} (virtual \ac{rbc} type 18 \figref{fig:multi-objective-cell-examples}).

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/multi-objective-importance.pdf}
    \placecaption{Average median effect size for white and red blood cells (WBC and RBC, respectively) in the complete multi-objective model.}
    \label{fig:multi-objective-importance}
\end{figure}

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/cell-mo-examples.pdf}
    \placecaption{Virtual white and blood cell (WBC and RBC, respectively) type examples relevant for anaeamia classification for the complete multi-objective model.}
    \label{fig:multi-objective-cell-examples}
\end{figure}

\subsection{Virtual cell expert annotation}

I asked 3 haematologists to annotate 1746 \ac{rbc} and 1600 \ac{wbc} automatically detected using the protocol described above from MLLC to assess how cohesive each virtual cell type was. For the annotation, I implemented an online platform --- \ac{tbca} --- where haematologists could register and annotate cells according to previously defined cell types discussed with my co-advisor George Vassiliou. More particularly, \ac{wbc} could be classified as neutrophils, basophils, eosinophils, band cells, lymphocytes, monocytes, blasts or artefacts (poor resolution, nucleated \ac{rbc}, reticulocytes, fragmented or dead \ac{wbc}, multiple \ac{wbc}, incomplete or no \ac{wbc} and platelet clumps). \Ac{rbc} could be classified as normal \ac{rbc}, nucelated \ac{rbc}, spherocytes, target cells, irregularly contracted, dacrocytes, keratocytes, echinocytes, eliptocytes, acanthocytes and artefacts (poor resolution, multiple \ac{rbc}, incomplete or no \ac{rbc} and platelet clumps, platelets or oversegmented \ac{rbc} with platelets). Whenever a cell had more than one annotation, I used a majority rule to decide which annotation to keep. To address ties, I selected a label at random.

The Blood Cell Atlas --- was deployed in the \ac{ebi} Embassy cloud using Kubernetes, with a backend in Flask, Python and PostGreSQL and a frontend with Flask and Bootstrap. In \ac{tbca}, each \ac{wbc} and \ac{rbc} is labelled using a dropdown menu by experts (\figref{fig:tbca}; available in \url{http://45.88.80.93:5000/}).

\begin{figure}[!ht]
    \missing{This figure is missing because the Embassy cloud is down}
    \placecaption{Screenshot of cell labelling menu in The Blood Cell Atlas.}
    \label{fig:tbca}
\end{figure}

I start this analysis by noting that most annotations were done by a single expert which can be problematic as shown by previous studies \cite{De_Swart2017-wc,Howe2004-mn,Goasguen2009-dn,Foucar2020-uz}. This is further highlighted once I take a more dedicated look at the degree of agreeability between experts --- in \figref{fig:mile-vice-annotated-cells-wbc-concordance} I show, for cells which were annotated by two or more experts, the fraction of annotations with at least one disagreement. In essence, while some cell groups (neutrophils or lymphocytes) show good concordance, others are more subjective --- annotations of monocytes and band cells/Pelger-Huët anomalies by experts show very limited agreeability between experts. Taking this into account during this analysis is crucial as it renders some annotations (particularly those of monocytes and Pelger-Huët anomalies) considerably more questionable. Analysis of which annotations are picked by experts stratified by the majority label shows that Pelger-Huët anomalies were considered neutrophils in at least one third of the annotations (22/65), in line with previous reports showing that the annotation of nuclear hypolobulation in neutrophils showed the least inter-expert agreement \cite{Weinberg2015-ra}. Monocytes are considered neutrophils or lymphocytes in 20.6\% and 7.6\% of the annotations, respectively. I also note here that the "poor resolution" and the "fragmented/unrecognizable" categories also have a relatively high fraction of uncertainty. This may be due to the personal experience of each expert --- while some experts will be more confident in calling specific cell types, others will be more confident in not analysing ambiguous cell types since the diagnostic is generally performed by inspecting a sufficiently large number of peripheral blood cells. Very few \ac{rbc} were annotated by more than one expert (N=28) and, as such, no assessment of inter-expert agreement will be performed.

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/mile-vice-annotated-cells-wbc-concordance.pdf}
    \placecaption{Proportion of \ac{wbc} annotations where all experts agree.}
    \label{fig:mile-vice-annotated-cells-wbc-concordance}
\end{figure}

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/mile-vice-annotated-cells-wbc-concordance-heatmap.pdf}
    \placecaption{Distribution of annotations stratified by the majority label (the most consensual label).}
    \label{fig:mile-vice-annotated-cells-wbc-concordance-heatmap}
\end{figure}

To further understand whether these labels could be used reliably, I assessed whether annotated \ac{wbc} typically associated with pathogenic clinical conditions --- Perger-Huët/band cells and blasts --- were absent from normal individuals. It is worth noting here that some of these cell types may be observed in healthy individuals, but never to the same extent they are observed in individuals with medical conditions where these cell types are overrepresented such as \ac{mds}. There appears to be no statistically significant differences between either cell type and clinical conditions, with the exception of an overrepresentation of Perger-Huët/band cells in \ac{mds} when compared with anaemia (\figref{fig:annotated-wbc-rate} and \figref{fig:annotated-wbc-rate-test}). 

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/annotated-wbc-rate.pdf}
    \placecaption{Detection rates for white blood cell types stratified by condition.}
    \label{fig:annotated-wbc-rate}
\end{figure}

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/annotated-wbc-rate-test.pdf}
    \placecaption{Poisson rate ratio tests between different conditions for expert annotations for Pelger-Huët/band cells and blasts.}
    \label{fig:annotated-wbc-rate-test}
\end{figure}

Conclusively, for Pelger-Huët/band cells and blasts, agreement is limited and the differential presence of \ac{wbc} in different conditions is unlikely --- while a small fraction of band cells are expected in healthy individuals (under 10\% of the total granulocyte count \cite{Drees2012-sz}), the reported rates of congenital Pelger-Huët anomalies are too low to explan this similarity (1 in 6,000 in the United Kingdom) and acquired Pelger-Huët anomalies are expected to be elevated in \ac{mds} \cite{Colella2012-so}. Additionally, it is unreasonable to expect blasts to be equally prevalent in both healthy individuals and individuals with \ac{mds} --- while \ac{mds} features a relatively lower amount of blasts when compared with \ac{aml}, healthy individuals should present no blasts in peripheral blood \cite{Aster2020-cu}. Taking this into consideration, I do not consider these annotations any further.

For \ac{wbc} and in general, there does not appear to be any virtual cell types in which the majority of a specific cell type is contained --- the main exceptions being eosinophils in virtual cell type 18 for the \ac{sf3b1}-mutation detection in \ac{mds} task \figref{fig:mile-vice-annotated-cells-wbc}. This virtual cell type is more prevalent in \ac{sf3b1}-mutant \ac{mds}, hinting that hypereosinophilia (the overproduction of eosinophils) may be a hallmark of \ac{sf3b1}-mutant \ac{mds}. This is a phenotype which has been previously observed across several myeloid neoplasms but which is typically associated with fusion genes \cite{Reiter2017-pj}. As for \ac{rbc}, a similar picture can be observed --- no particular virtual cell type is particularly representative of a specific cell type except for spherocytes (spherical \ac{rbc}) in the disease detection task (virtual cell type 37). 

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/mile-vice-annotated-cells-wbc.pdf}
    \placecaption{Proportion of annotated white blood cell (WBC) types by each virtual cell type (left) and number of annotated \ac{wbc} (right). Only virtual cell types with at least 10\% of a specific \ac{wbc} type are shown.}
    \label{fig:mile-vice-annotated-cells-wbc}
\end{figure}

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/mile-vice-annotated-cells-rbc.pdf}
    \placecaption{Proportion of annotated red blood cell (RBC) types by each virtual cell type (left) and number of annotated \ac{wbc} (right). Only virtual cell types with at least 10\% of a specific \ac{rbc} type are shown.}
    \label{fig:mile-vice-annotated-cells-rbc}
\end{figure}

While it would be good to recapitulate specific cell types using a weakly supervised approach, it is worth noting that the tasks at hand are likely to not be the most useful for this --- if one had access to cases of hypereosinophilia or hyperbasophilia (excess of peripheral eosiniphils and basophils, respectively), perhaps this approach could be useful to create a virtual cell vocabulary that could be useful to label specific cell types. However, here I analyse conditions which are not known for having an excess of specific healthy cell types but rather specific deformations and dysplasias. Nonetheless, the analysis concerned with inter-expert agreement in \ac{wbc} annotation is a further confirmation of the difficulties experts in haematology face when analysing \ac{wbs}.

\section{External validation of models}

I used AC2, a cohort of individuals with either one of the conditions studied above in this chapter --- case controls (normal; N=10), iron deficiency anaemia (N=12), megaloblastic anaemia (N=3), \ac{sf3b1}-mutant \ac{mds} (N=13) and other \ac{mds} subtypes (specifically, \ac{srsf2}-mutant \ac{mds}; N=12). 

For the elastic regression model it is possible to state that there is a good generalization capability, with \ac{auroc} scores suffering no significant drop (\figref{fig:glmnet-auc-validation}). However, when inspecting the same scores for \ac{mile-vice}, a considerable drop in \ac{auroc} for the disease detection task is evident (\figref{fig:glmnet-auc-validation}), which led me to further investigate possible reasons for this drop.

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/glmnet-auc-validation.pdf}
    \placecaption{Cross-validated (bars) and external validation \ac{auroc} values (diamonds) for the elastic net regression model. The confidence intervals for the external validation \ac{auroc} are obtained as $\frac{1}{\sqrt{n}}$, where $n$ is the sample size used to obtain this value.}
    \label{fig:glmnet-auc-validation}
\end{figure}

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/mile-vice-auc-validation.pdf}
    \placecaption{Cross-validated (bars) and external validation \ac{auroc} values (diamonds) for \ac{mile-vice}. The confidence intervals for the external validation \ac{auroc} are obtained as $\frac{1}{\sqrt{n}}$, where $n$ is the sample size used to obtain this value.}
    \label{fig:mile-vice-auc-validation}
\end{figure}

By reducing the morphometric distributions to a two-dimensional space using a \ac{umap} and inspecting their distribution, a batch effect becomes considerably evident and relatively more apparent for \ac{rbc} when compared with \ac{wbc} (\figref{fig:u-map}). This highlights a problem of trying to generalize \ac{ml} algorithms trained on single-centre datasets --- often times one observes dataset shift. 

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/u-map.pdf}
    \placecaption{Uniform manifold approximation and projection (UMAP) for the morphometric distributions stratified by datasets for white blood cell features (left), red blood cell features (center) and for the features of both cell types combined (right). UMAP1/2 refer to the two dimensions of the \ac{umap} projection.}
    \label{fig:u-map}
\end{figure}

\subsection{Dataset shift} 

In \ac{ml}, the objective is to estimate the joint probability of the predictors $X$ and the labels $Y$ ($P(X,Y)=P(Y|X)P(X)$). However, the learning algorithm does not have access to $P(X)$, only to the probability of the training data $X_{train}$ ($P(X_{train})$). Consequently, if the distribution of the validation data ($P(X_{val}$) of the relationship between the validation data and the validation labels ($P(Y|X_{val})$) shifts, the joint probability will also change. This is known as a dataset shift and can be further classified into covariate shift ($P(X_{val})$ is shifted) or concept shift ($P(Y|X_{val})$ is shifted) \cite{Y2019-vc}. The fact that the feature distributions are considerably different between datasets (\figref{fig:u-map}) hints that this may be a covariate shift, where the feature extraction protocol produces different results between datasets.

\subsubsection{Possible reasons for dataset shift} 

Firstly, I considered in which ways the external validation cohort (AC2) differed from the training cohort (MLLC) --- AC2, while similar in composition to (MLLC), was digitalized using a different scanner --- an Aperio AT2 --- and slides were obtained in different ways depending on whether they were from an \ac{mds} patient. Whereas slides from normal individuals or individuals with anaemia were digitalized within 12 hours after being automatically smeared and stained using a Siemens Hematek 3000 system, the slides from individuals with \ac{mds} came from a different laboratory, dedicated to haematooncological diagnostic, and were prepared (smeared and stained) by hand 12-24 hours after blood collection. This difference in preparation time induces different levels of \ac{edta} artefacts --- as mentioned in the Introduction, \ac{edta} is an anticoagulant agent used to ensure that there is no cellular aggregates in the \ac{wbs}; however, if more than a few hours pass between the addition of \ac{edta} to blood and the smearing, artefacts affecting both \ac{wbc} and \ac{rbc} are to be expected even after 2 hours, with their severity and frequency increasing as time passes \cite{Bain2005-zg,Narasimha2008-fh}.

Furthermore, the Aperio AT2, as mentioned earlier in this Chapter, has a slightly different resolution from the Hamamatsu Nanozoomer 2.0 which was used to digitalize the other cohorts (0.2517 micrometers/pixel for Hamamatsu NanoZoomer 2.0; 0.2268 micrometers/pixel for Aperio AT2) and to correct this systematic shift I resize images digitalized using the Aperio AT2 prior to their analysis; however, this resizing step may introduce interpolation artefacts which can cause small changes to the values of each shape descriptor. Other artefacts, relating to differences in image compression and digitalization associated with the use of difference scanners, may also influence this.

\subsubsection{Fixing dataset shift by unbiased feature selection} 

Typical solutions to training a \ac{ml} algorithm that is resistant to dataset shift always rely on either having multiple sources for the data (i.e. multi-centre datasets) and allowing the algorithm to control for soures of noise associated with different datasets or in the estimation of the importance of each sample based on the empirical or estimated density of this sample in the validation dataset \cite{Y2019-vc}. In other words, the solutions for dataset shift implicate having access to more data, be it data coming from other sources or the actual validation data.

As a demonstration of the potential of this feature extraction protocol, I have excluded features which were predictive of each dataset --- either MLLC or AC2. To do this, I use the Mann-Whitney U-statistic between both datasets for all features and divide it by the product of both sample sizes, a measure known as the common language effect size which is equivalent to the \ac{auroc}. The common language effect size represents the proportion of pairs supporting a direction --- values close to 0 or 1 imply that the U-statistic is biased in a specific direction (towards a specific dataset in my case), whereas values close to 0.5 imply little biasing \cite{Kerby2014-fy}. Setting a hard cut-off for the common language effect size between $1/3$ and $2/3$, I regenerate the \ac{umap} representations, confirming that there no longer appears to be a a visible separation between datasets \figref{fig:u-map-unbiased}. Having confirmed this, I retrain both the elastic regression model and \ac{mile-vice} using only these unbiased features (20 \ac{wbc} features and 18 \ac{rbc} features). I note here that the point of this analysis is not to prove generalization, but rather to demonstrate the task-specific utility of the methodology here presented when the training protocol enables the elimination of dataset-specific noise.

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/u-map.pdf}
    \placecaption{Uniform manifold approximation and projection (UMAP) for the morphometric distributions stratified by datasets for white blood cell unbiased features (left), red blood cell unbiased features (center) and for the unbiased features of both cell types combined (right). UMAP1/2 refer to the two dimensions of the \ac{umap} projection.}
    \label{fig:u-map-unbiased}
\end{figure}

Excluding features which were considerably different between datasets led to a considerable drop in performance for the \ac{sf3b1}-mutant \ac{mds} detection task when considering the elastic net regression model, whereas other \ac{auroc} scores remained relatively similar (\figref{fig:glmnet-auc-validation-unbiased}). On the other hand, while using an dataset-unbiased set of features, \ac{mile-vice} consistently retains its predictive ability except for the task of disease classification using only morphological data, where it dramatically underperforms (\figref{fig:mile-vice-auc-validation-unbiased}). Both cases are likely to stem from underfitting, where the morphological description is not rich enough to enable the model to learn to discriminate between conditions.

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/glmnet-auc-validation-unbiased.pdf}
    \placecaption{Cross-validated (bars) and external validation \ac{auroc} values (diamonds) for the elastic net regression model using only dataset-unbiased features. The confidence intervals for the external validation \ac{auroc} are obtained as $\frac{1}{\sqrt{n}}$, where $n$ is the sample size used to obtain this value.}
    \label{fig:glmnet-auc-validation-unbiased}
\end{figure}

\begin{figure}[!ht]
    \placefigure{gfx/cytomorphology/mile-vice-auc-validation-unbiased.pdf}
    \placecaption{Cross-validated (bars) and external validation \ac{auroc} values (diamonds) for \ac{mile-vice} using only dataset-unbiased features. The confidence intervals for the external validation \ac{auroc} are obtained as $\frac{1}{\sqrt{n}}$, where $n$ is the sample size used to obtain this value.}
    \label{fig:mile-vice-auc-validation-unbiased}
\end{figure}

\FloatBarrier

\section{Discussion}

In this chapter I use one of my \ac{wbs} cohorts --- MLLC --- to show how the protocol outlined in Chapter 4 can be used to create individual and clinically-relevant cytomorphological profiles based on morphometric distributions, applying them successfully to the prediction of different haematological conditions. Additionally, using vocabulary-based \ac{mil}, a \ac{ml} paradigm focused on the discovery of latent terms which are relevant for the classification of bags of instances, can be used to discover virtual cells types, which can be used as cytomorphological hallmarks --- cellular archetypes --- of different haematological conditions while performing adequately in most predictive tasks.

\paragraph{Computational cytomorphology enables the prediction of clinically-relevant conditions and identification of cellular archetypes.} In this Chapter I show how computational cytomorphology can be used to accurately predict different clinical conditions using only \ac{wbs} and \ac{cbc}, noting that this generalizes to other datasets except when considering the task of disease detection using \ac{mile-vice}. I uncover specific cellular subtypes associated with each condition, showing how weak supervision in a \ac{mil} setting can be used to derive powerful conclusions regarding the specific cytomorphological signatures which characterize each condition. 

These relative prevalence of these cellular archetypes across different conditions was used to verify some previously stated conclusions such as the prevalence of hyperlobulated in both iron deficiency and megaloblastic anaemia \cite{Lindenbaum1980 -ux,Westerman1999-gs}, relatively large \ac{rbc} sizes in megaloblastic anaemia when compared with iron deficiency anaemia, and the presence of Pelger-Huët cells in \ac{mds} \cite{Colella2012-so}. Additionally, I uncover novel associations between cellular archetypes and specific conditions --- Perger-Huët anomalies are more common in \ac{sf3b1}-mutant \ac{mds} than in other \ac{mds} subtypes and, while hyperlobulated neutrophils are present in both iron deficiency anaemia and megaloblastic anaemia, those in the latter are typically larger. This suggests that the increase in cellular size observed in \ac{rbc} in megaloblastic anaemia also affects granulocytes. A further novel cellular archetype detected by both the elastic network regression and \ac{mile-vice} is the increase in \ac{rbc} size in \ac{sf3b1}-mutant \ac{mds} when compared with other \ac{mds} subtypes. The main objective of these cellular archetypes is to provide haematologists with novel types of cell that can allow them to better discriminate between conditions; considering this, the validation of cellular archetypes as valuable clinical tools will have to be done by providing them as guidelines for trained experts and determining whether they provide relevant information for condition discrimination.

Considering the biology of these tasks, little is known about the mechanisms which lead to alterations to blood cell maturation such as the ones described above. It is thought that neutrophil hyperlobulation can be caused by nutrient deficiency (as would be the case of megaloblastic anaemia, typically caused by folate or vitamin B12 deficiency) or iron deficiency anaemia but no known functional role has been identified thus far \cite{Manley2018-xo}. Hypolobulation, on the other hand, can be relatively normal if observed in band cells small proportions or benign if associated with congenital Pelger-Huët anomaly \cite{Colella2012-so}, but in sufficiently high numbers can be indicative of deficient cell maturation \cite{Aster2020-cu}. To further complicate these reflections, the literature shows no good agreement regarding the actual function of nuclear lobulation, with possible differences between the functional impacts of inherited and acquired conditions \cite{Manley2018-xo}. Similarly, enlarged \ac{rbc} can also be caused, as mentioned, by nutrient deficiency and \ac{mds} \cite{Aslinia2006-en}, but have no specific functional impact. 

\paragraph{Towards multi-centre blood cytomorphology.} External validation shows this methodology has some generalization abilities which are not sufficient for its application in other centres --- particularly for \ac{mile-vice}, the detection of diseases (\ac{mds} or anaemia) shows notably poor predictive performance when tested on an independent test set, an effect that likely stems from covariate shift --- a change in the probability distribution of the data. This highlights an important aspect of works in digital histopathology which has been stated in recent reviews --- multi-centre training sets, allowing \ac{ml} algorithms to learn how to control for dataset origin (which can affect, among other aspects, image compression, illumination, and stain intensity and distribution), are an increasing necessity if good generalization is to be expected \cite{Van_der_Laak2021-id}, especially in settings where images are obtained mainly for clinical, rather than scientific, use.

\section{Code availability and statistical analysis}

The code for different sections of this work is available across different repositories:

\begin{itemize}
    \item R scripts and code to train the elastic network regression model and the regularized random forest model, and generate figures is available in \url{https://github.com/josegcpa/wbs-prediction} under \texttt{analysis-plotting}
    \item The code to train, test and predict using \ac{mile-vice} is available in \url{https://github.com/josegcpa/wbs-prediction} under \texttt{mile-vice}
\end{itemize}

All statistical analyses were performed in R 3.6.3 \cite{R-core-team} --- \texttt{caret} was used to calculate cross-validated metrics \cite{Kuhn-2021-caret} and \texttt{pROC} was used to calculate \ac{auroc} \cite{Robin-2011-proc}. 
