\chapter{Discussion}

In this dissertation, I have studied the somatic evolution and cytomorphology of the haematopoietic system. To study the somatic evolution of haematopoiesis, I used longitudinal targeted sequencing data from over 300 elderly individuals over a median period of 13 years and 7 single-cell colony derived phylogenies to track \ac{ch} clones and quantify their driver-specific and lifelong growth patterns. Using simulations, I developed and validated statistical methods that quantify the clonal dynamics using these types of data and applied them, revealing lifelong clonal growth behaviours. Changes to blood cytomorphology were studied using a cohort of over 300 \ac{wbs} from individuals with either \ac{sf3b1}-mutant \ac{mds}, other subtypes of \ac{mds}, iron deficiency anaemia or megaloblastic anaemia and case controls using \ac{cv} methods. I developed methods to detect and characterize blood cells from \ac{wbs} and showed how the morphometric characterizations of cells obtained using these methods can be used not only to predict clinical conditions, but also to define cytomorphology-disease associations that reveal novel morphological cellular archetypes. Below I discuss the main conclusions and limitations of both works and how they can be further extended and improved to enable their clinical application. Finally, I discuss how both works could be combined to enable the study of somatic evolution through cytomorphology.

\section{The longitudinal dynamics and natural history of clonal haematopoiesis} 

\subsection{The biology of \ac{ch} dynamics}

In \ac{ch}, different drivers are associated with distinct growth rates during old age: while \ac{dnmt3a} clones grow on average at 5\% per year, others, such as those characterized by \ac{u2af1} mutations and \ac{srsf2}-P95H, grow 6 and 10 times faster, respectively. While constant during old age, this growth rate is not representative of the lifelong behaviour of clones: factors such as an increasingly competitive oligoclonal landscape lead to clonal deceleration, a behaviour detected in the vast majority of expansions detected in the single-cell colony-derived phylogenies of 7 individuals. While this phenomenon is likely to affect all clones, the driver-specific effect it may exert on clonal dynamics is hard to assert from this study --- the estimation of driver-specific dynamics was performed using longitudinal targeted sequencing during old age, which does not permit the study of lifelong alterations to clonal growth. Most mutations, including those in \ac{dnmt3a} and \ac{tet2} (the two most commonly mutated genes in \ac{ch}) appear consistently through life. However, mutations in \ac{u2af1} and \ac{srsf2}-P95H (two genes associated with splicing) are restricted to old age in \ac{ch}. Finally, there is a clear association between the gene-specific clonal growth rates and \ac{aml} onset risk, and between the site-specific clonal growth rates and positive selection in both \ac{mds} and \ac{aml}.

One of the most informative aspects of this work is tied to the different clonal dynamics conferred by different mutations and, consequently, the different levels of clonal deceleration. This is likely to be true for other tissues as demonstrated by selection analyses in different cancer types and normal tissues \cite{Martincorena2017-ii,Martincorena2018-yp,Yoshida2020-zi} and mutation timing analyses in cancer \cite{Gerstung2020-kf}. However, the mechanisms driving the associations between specific mutations and different fitness advantages is still hard to trace --- while for mutations in splicing genes this is probably related with the extent and specificity of splicing alterations \cite{Daubner2012-zp,Kim2015-qz}, those in \ac{dnmt3a}, one of the most commonly mutated genes in \ac{ch}, can be associated with loss of function largely due to protein instability \cite{Huang2021-dz}. Additionally, possible interactions with inflammation have been observed in \ac{dnmt3a} and \ac{tet2} \ac{ch} \cite{Hormaechea_Agulla2019-cd,Hormaechea-Agulla2021-kr,Cai2018-yi}, highlighting the role of external factors in clonal dynamics. It is also worth considering the age-associated and non-homogeneous depletion of the stem cell niche \cite{Crane2017-hl}, leading to decreased \ac{hsc} growth and repopulation capabilities \cite{Tanaka2017-pu,Wang2017-rg} --- this can have a considerable effect in clonal deceleration (and consequently in the fitness advantage of clones during old age) by reducing the available space for expansion in the bone marrow. While the field of \ac{ch} has expanded greatly in terms of understanding the prevalence of \ac{ch} and disease associations \cite{Jaiswal2014-rl,Genovese2014-eu,Abelson2018-wh}, the study of the biology of the mechanisms underlying the lifelong dynamics of different clones requires more than large human sequencing cohorts. Particularly, studies which combine animal models of \ac{ch} with expression and epigenetic analyses can be used to reveal how mutations lead to specific changes in haematopoiesis \cite{Asada2021-wd}. 

\subsection{\Ac{ch} and its potential role in the clinic} 

It is a well known fact that \ac{ch} is associated with progression to haematological malignancies and death \cite{Genovese2014-eu,Jaiswal2014-rl,Abelson2018-wh}. However, translating this knowledge to the clinic is complicated --- while there is a clear association between \ac{ch} growth dynamics and risk of progression to \ac{aml} and selection in \ac{aml} and \ac{mds}, most cases of \ac{ch} will not be associated with transformation to a myeloid malignancy. Beyond showing here how specific drivers in \ac{ch} are associated with relatively well-conserved dynamics, I also demonstrate that mutations associated with an increased risk of progression grow at a much faster rate in healthy individuals. This knowledge can potentially reveal which mutations should be of clinical concern and whether they should be tracked through time. However, several studies, including the work presented here, have highlighted how little is actually known about the prevalence and behaviour of \ac{ch}, with large clonal expansions having no known driver but similar dynamics to other \ac{ch} expansions with known drivers \cite{Poon2020-ek,Mitchell2021-zl,Fabre2021-uw}. Moreover, recent evidence shows the high prevalence of \ac{mca} and copy number alterations in \ac{ch} --- these can act as confounders, especially considering that some mutations are associated with copy number alterations that directly affect the gene \cite{Gao2021-ph,Saiki2021-sq}. 

\paragraph{Closing the gap on \ac{ch} knowledge.} Future studies into \ac{ch} need to focus much more on the discovery of novel variants, creating a more complete picture of \ac{ch} before effective and economically-feasible tracking of most \ac{ch} variants becomes a reality; for this, approaches such as those applied by \etal{Genovese} and \etal{Jaiswal}, where blood whole-exome sequences were obtained for very large cohorts and used to identify blood clones in healthy individuals \cite{Jaiswal2014-rl,Genovese2014-eu}, can be adapted by increasing the sequencing depth to further reveal recurring drivers of \ac{ch}. Additionally, studies which use single-cell colonies to explore \ac{ch}, such as the work presented here and that by \etal{Mitchell} can be extremely useful --- not only do they allow the practical exploration of the lifelong trajectories of different expansions, they can allow, given sufficient magnitude, for the tentative identification of novel drivers of \ac{ch} \cite{Fabre2021-uw,Mitchell2021-zl}. However, the cost of sequencing a sufficient number of single-cell colonies can act as a considerable obstacle. \etal{Poon} overcame the problem of data availability by using large cohorts where synonymous mutations could be used to track clonal expansions --- by doing so, and under the assumption that synonymous mutations can only expand by being passengers in clonal expansions, they were able to identify novel putative drivers. This knowledge can be used to create large panels of potential drivers which can then be further validated. With this knowledge, single-cell colonies could be used to establish the lifelong dynamics of putative drivers in clonal expansions. Moreover, through the long-term storage of viable \ac{dna} samples (which constitute relatively small and easily storable aliquots), it could be possible to retrospectively reconstruct the \ac{vaf} trajectories of clones, similarly to what was done in Chapters 2 and 3, and to better understand the oncogenic trajectories of different mutations and how they evolve as disease onset approached. 

Finally, it is worth noting that the collection of clinical data --- such as blood counts and behavioural patterns --- can help illuminate the mechanisms underlying specific expansions or even stratify individuals regarding the risk of developing \ac{ch} and of progression \cite{Dawoud2020-af,Abelson2018-wh}. Particularly, smoking has been shown to be associated with \ac{asxl1} \ac{ch} \cite{Dawoud2020-af}, while increased \ac{rdw} is associated with increased risk of \ac{aml} \cite{Abelson2018-wh}. Other blood counts have also shown specific associations with \ac{ch} \cite{Bick2019-uj,Dawoud2020-af,Cordua2019-mo,Abelson2018-wh,Zink2017-zi,Jaiswal2014-rl,Zehir2017-gh}, further showing how \ac{ch} changes haemaotopoiesis.

\section{The cytomorphology of myelodysplastic syndromes} 

\subsection{Computational methods reveal morphology-disease association}

The use of \ac{cv} methods to analyse \ac{wbs} can detect and characterize between thousands and hundreds of thousands of cells as demonstrated in Chapter 4. These vast collections of cells can then be used to predict clinical conditions with morphometric moments and through the use of \ac{mil} as shown in Chapter 5. Morphometric moments are individual-specific parametric characterizations of blood cell morphometry. While simple and useful, morphometric moments do not permit a visually intuitive characterization of blood cells. For this reason, I developed \ac{mile-vice}, a \ac{mil} method that clusters cells into visually coherent and clinically-relevant \ac{vct} and uses their relative proportions to predict diseases. These \ac{vct} reveal novel cellular archetypes of different conditions such as anaemias and \ac{mds} and its different subtypes. For instance, \ac{rbc} are larger in \ac{sf3b1}-mutant \ac{mds} than in other \ac{mds} subtypes, whereas poikilocytic \ac{rbc} are more prevalent in other \ac{mds} subtypes. Additionally, \ac{sf3b1}-mutant \ac{mds} has a larger relative prevalence of hypolobulated neutrophils when compared with other \ac{sf3b1} subtypes, whereas hyperlobulated neutrophils, prevalent in both megaloblastic anaemia and iron deficiency anaemia, are larger in the megaloblastic anaemia. Finally, I show how the predictive performance of most tasks generalizes to an external validation cohort. However, when considering the prediction of whether a \ac{wbs} belongs to an individual with either \ac{mds} or anaemia, \ac{mile-vice} underperforms. I posit that this is likely to be due to artefacts potentially introduced by the preparation and digitalisation of \ac{wbs} and show how a smaller subset of features, uncorrelated with dataset of origin, can alleviate the drop in the external validation performance.

\subsection{Generalization to other diseases} 

Different clinically-relevant conditions can be predicted from \ac{wbs} using only computational methods and, through the use of \ac{mile-vice}, new cellular archetypes may be discovered in situations where little expert annotation is required. However, clinical application requires these methods to be applicable to a much larger number of clinical conditions, implying the necessity for much larger cohorts of digitalised and diagnosed \ac{wbs}. While the work presented in Chapters 4 and 5 serves as a proof-of-concept for the possibilities of computational cytomorphology, confirming these associations more routinely --- in \ac{wbs} cohorts prepared in different centres but under similar circumstances and digitalised using the same slide scanner --- is crucial to understand whether or not they can be reliably used by trained experts. It is worth noting that other works have shown the theoretical potential of computational cytomorphology in haematology \cite{Eckardt2021-fb,Bruck2021-fx} but their lack of external validation makes it complicated to draw conclusions regarding widespread applicability. Finally, it is worth noting that generalization to other cohorts, digitalized with a different scanner and prepared under different standards, is still unsatisfactory --- the most logical path towards a possible solution for this lack of generalization is the use of multi-centre training cohorts. 

\subsection{Improving the characterization of blood cells}

Regarding the characterization of blood cells, improvements can be made --- large databases of annotated cells can enable the extraction of morphologically-meaningful features with no need for feature design. This can be achieved by training classification \ac{cnn} models and using the feature vectors produced by these models as morphological descriptors of different cells, or by training auto-encoder models to characterize the morphology of \ac{wbc} and \ac{rbc}. Other works have demonstrated the effectiveness of \ac{cnn} models in the classification of \ac{wbc} type \cite{Matek2019-ld,Matek2021-mp,Kouzehkanan2021-mz}, so a potential path forward would be to build cytomorphometric descriptors on top of the models suggested by these works, which has the additional advantage of not requiring fine-grained segmentation maps. Nonetheless, the work I present in Chapters 4 and 5, where segmentation maps are obtained for millions of cells, could potentially ellucidate how complex \ac{cnn}-derived cytomorphometric descriptors associate with simpler and more intuitive size, shape and texture descriptors.

\subsection{Computational cytomorphology --- a potential path to the clinic} 

\Ac{ml} and \ac{dl} systems enable the high-throughput and reproducible analysis of \ac{wbs} in a clinical context. However, particular hurdles are still in the way --- the absence of very large, multi-centre cohorts impedes the realistic assessment of predictive power. Moreover, most studies are retrospective (using available and previously stored data), which is not representative of point-of-care where predictions are expected on a regular basis for new slides \cite{Eckardt2020-fp}. Large multi-centre cohorts also enable a more realistic learning of the true disease signal, which should be independent of cohort-specific noise. An additional factor to consider is that diagnostic conventions change --- how these prediction algorithms change is also worthy of consideration: if an augmented intelligence setting is considered, where the predictive algorithm is merely an assistant to the expert which is expected to confirm the diagnoses, a reinforcement learning strategy should be considered, where the algorithm is iteratively refined by a team of experts. Federated learning, a decentralized \ac{ml} paradigm that permits guarantees data-privacy and better data governance \cite{Rieke2020-hl}, could be combined with large multi-centre databases and reinforcement learning to create a powerful semi-automated diagnostic system capable of constant updates. However, if diagnostic criteria change, new training cohorts are possibly necessary --- in the worst case scenario, a disease subtype may be stratified further, requiring expert annotation or confirmation (it may also be possible that a classification algorithm will already, implicitly, cluster disease subtypes within its feature space). This last circumstance would require the retraining of these models.

\paragraph{Digitaliser agnostic models?} Finally, it is worth considering whether generalization to other digitalisers should be an objective of these models. Companies working on the automated analysis of \ac{wbs} focus, for now, exclusively on the development of protocols which work exclusively with their digitalizers. These methods perform differential counting of \ac{wbc} and generic cellular detection in \ac{wbs} with some abnormality detection capabilities by leveraging large expert-annotated collections of \ac{wbc} \cite{cellavision,advia-120}. Combined with \ac{mile-vice}, which enables the definition of novel and clinically-relevant cellular archetypes, the robust cellular detection and characterization capability of these systems can be not only expanded but used for cell type discovery. It is also worth noting that the work presented in Chapters 4 and 5 shows how \ac{rbc} morphology, generally not considered by these systems, can be useful to establish novel cytomorphology-disease. 

Some automated systems for \ac{wbs} analysis incorporate the automatic preparation of \ac{wbs} from blood samples \cite{advia-120}. This eliminates most of the variability associated with slide preparation and digitalization. In realistic terms, a relatively simple solution --- training models which are optimized for detecting and characterizing cells in specific systems of slide preparation and digitalization and testing them as diagnosis and prognosis tools prospectively --- will provide the strongest evidence for their generalization as a clinically applicable and transferrable solution. Models for disease prediction from \ac{wbs} --- such as those developed and presented in Chapter 5, whose underperformance in an external validation cohort is likely to be associated with \ac{wbs} preparation and digitalisation --- become much more applicable in these circumstances. Nonetheless, it should be noted that specific sources of noise, associated with the specific composition of different populations, may still be a problem for these models --- it has been shown that incidence and 5-year survival of blood cancers, or the incidence of specific haemoglobinopathies (such as thalassemia, a condition characterized by decreased or defective haemoglobin production), differs by ethnicity and race \cite{Kirtane2017-dh,Lorey1996-yc}. The aetiology of such differences is unknown, with factors such as diet, socio-economic conditions and biology being possible causes, and the changes they elicit on the cytomorphology of blood are far from being studied. Approaches such as the one presented here, where morphometric trends and cell types are established in association to specific conditions, could help reveal the morphological signatures of different ethnicities and races, possibly creating better informed diagnostic practices by defining novel patient strata which are associated with clinical outcome in specific diseases.

\section{Somatic evolution and cytomorphology}

\paragraph{Determining the cytomorphology of \ac{ch}.} At the moment and to the best of my knowledge there are no cytomorphological markers of \ac{ch}. However, several blood count-\ac{ch} associations have been established, hinting that \ac{ch} leads to concrete changes to the haematopoietic process. Furthermore, through methods such as \ac{mile-vice} or other approaches, it is possible to determine specific associations between \ac{rbc} and \ac{wbc} morphology and clinical condition that standard expert analysis was unable to detect \cite{Eckardt2021-fb,Bruck2021-fx}. Additionally, it is worth noting that some of the changes in \ac{vct} prevalence associated with different conditions discovered in this study may be hard to detect by humans (hyperlobulated neutrophils are prevalent in both iron deficiency anaemia and in megaloblastic anaemia but are larger in the latter), or that experts struggle to concordantly identify some well known blood cell types (such as hypolobulated neutrophils) \cite{Weinberg2015-ra}. To assess the potential of \ac{mile-vice} to identify \ac{ch} cellular archetypes, a cohort of individuals with different types of \ac{ch} (in terms of mutated genes and clonality) and with information regarding the size of different clones would have to be assembled. The former would allow a classification algorithm similar to that presented in Chapter 5, using different mutations and/or clonality as classes in a multiclass and multi-objective setting. Additionally, assigning groups of \ac{vct} to \ac{vaf} for different mutations could possibly be done as \textit{post hoc} analysis or even be incorporated into the training process by incorporating a regression objective that matched different sums of \ac{vct} proportions to \ac{vaf}, thus allowing the investigation of \ac{vct} proportion-clonal expansion associations.

\paragraph{Investigating the progression from \ac{ch} to \ac{aml} through cytomorphology.} This type of protocol could be further expanded --- using the knowledge that \ac{ch}, \ac{mds} and \ac{aml} are likely to be different stages of dysplasia and neoplasia, training a model based on \ac{mile-vice} with \ac{wbs} from healthy individuals and individuals with \ac{ch} and different stages of \ac{mds} and \ac{aml} could enable a more unified understanding of the cytomorphological manifestations of these conditions. Different possibilities can be posited here:

\begin{itemize}
    \item \Ac{vct} can be enriched in specific conditions --- it is possible that \ac{aml} may have more severe levels of dysplasia which could be classified as a distinct \ac{vct}, or that a single, very large \ac{ch} clone will lead to slight and non-dysplastic changes which are prevalent only in \ac{ch}
    \item The prevalence of specific \ac{vct} may increase with the severity of the condition (with \ac{ch}, \ac{mds} and \ac{aml} being increasingly more severe) and with the \ac{vaf} of specific clones --- this would, to some extent, confirm that \ac{ch}, \ac{mds} and \ac{aml} indeed exist on a spectrum and open new doors to early detection of \ac{mds} and \ac{aml}.
\end{itemize}

Finally, clinical research in \ac{mds} has established that blast counts in the bone marrow or different cytopenias are associated with increased risk of death \cite{Malcovati2020-no,Greenberg2012-en}, and previous works have shown that the risk of \ac{aml} onset can be predicted using \ac{cbc} \cite{Abelson2018-wh}. This suggests that different proportions of \ac{vct}, together with information on \ac{cbc}, may be potentially helpful in predicting the severity different diseases and risk of myeloid progression in healthy individuals.