\chapter{Methods for inferring the evolutionary dynamics of haematopoietic clones}

\section{Introduction}

Specific mutations are responsible for characterising different haematopoietic clones - as mentioned in the Introduction, different studies have looked at how the dynamics of clonal haematopoiesis can be inferred from both longitudinal and cross-sectional studies. Here, I validate two methods - one for the quantification of clonal dynamics from longitudinal sequencing data, and another one for the quantification of clonal dynamics from single-cell phylogenies - using simulations under a Wright-Fisher model with mutations conferring fitness advantages. Using this relatively simple model, I show that a relatively simple parametrisation is capable of capturing clonal dynamics from longitudinal sequencing data, while phylodynamic estimation, combined with relatively simple statistical inference, allows not only the recapitulation of the true fitness of the clone, but also the study of changes to the relative fitness of clones as they progress in time.

\subsection{Simulating evolution with the Wright-Fisher model}

Under genetic drift, populations evolve practically at random, following a small but strict set of rules. While several models try to tackle this process of evolution, here I shall focus on the one which I used - the Wright-Fisher model. Two defining features of this model can be quickly enumerated:

\begin{enumerate}
	\item Generations do not overlap and all individuals give rise to their offspring at exactly the same time. In effect, the smallest unit of time in the Wright-Fisher model is the generation. This is the major difference from the Moran model, where individuals from the population are selected at random to produce offspring at any given time
	\item The allele counts in a generation are determined by their prevalence in the previous generation. Indeed, considering that two alleles (A and B) exist in a diploid population of $N$ individuals, the probability of allele A having $k$ counts in the next generation is given by $\binom{2N}{k}p^kq^{2N-k}$, where $p$ and $q$ are the fraction of alleles A and B in this generation, respectively. In other words, the counts $X$ of allele A are distributed according to a binomial distribution with probability $p$ and $2N$ trials, making $E[X] = np$ and $Var[X] = npq$
\end{enumerate}

The two principles stated above are enough to simulate the evolution of a population given an initial distribution of alleles. However, this is not necessarily useful if the focus is on somatic mutations - the amount of somatic mutations at birth is quite low \cite{SpencerChapman_2021_gjz4x4}, increasing linearly with age \cite{Abascal_2021_gjvqfm}. As such, the Wright-Fisher simulations can be adapted so that, at every generation, there is a small chance of a mutation occurring - this chance is generally known as the mutation rate or $u$. This allows us to simulate a process that is more similar to that of somatic mutation.

However, the presence of mutations does not render this description of evolution particularly complete: a key aspect of the evolution of somatic mutations is missing - selection. Indeed, the greater resistance to genetic drift mutations that is conferred by mutations - in other words, the ability of some clones to strive over other clones - is parameterised as a fitness advantage or $s$. Here I will only consider the case where $s \geq 0$, noting that $s = 0$ implies no fitness advantage and $s < 0$ leads to the clone disappearing. This fitness advantage, in terms of the Wright-Fisher model, is seen as a relative increase in $p$ such that $p' = (1+s)*p$, leading to an increase of the expected number of counts $X'$ of allele A - $E[X'] = Np*(1+s)$. This has quite simple consequences - by increasing the probability of the binomial sampling of A counts, the population of clones with A will increase at a greater rate than that of B. However, the effect of absolute fitness is affected by the average fitness of the population $\bar{w} = p^2(1+s) + 2pq(1+hs) + q^2(1) = p^2 + p^2s + 2pq + 2pqhs + q^2 = 1 + p^2s + 2pqhs$, following the Hardy-Weinberg law ($1 = p^2 + 2pq + q^2$) and where $h$ is the fraction heterozygous (carrying both A and B alleles) individuals in the population. Having the average population fitness $\bar{w}$ and the absolute fitness $s$, the relative allele frequency at time $t+1$ is $p(t+1) = p(t)\frac{1+s}{\bar{w}} = p(t)\frac{1+s}{1 + p^2s + 2pqhs}$. As $p$ becomes closer to $1$ (and $q$, consequently, approaches $0$), $p(t+1) = p(t)\frac{1+s}{1+s} = p(t)$. 

To better formalize the advantage of $A$ over $B$, assuming that B can confer a fitness advantage $r$ and remembering that $p(t+1) = p(t)\frac{1+s}{\bar{w}}$, the ratio between allele frequencies can be given by $\frac{p(t+1)}{q(t+1)} = \frac{p(t)\frac{1+s}{\bar{w}}}{q(t)\frac{1+s}{\bar{w}}} = \frac{p(t)(1+s)}{q(t)(1+r)}$ which becomes, by induction and for any time increment $t_i$, $\frac{p(t+t_1)}{q(t+t_i)}=\frac{p(t)}{q(t)}(\frac{1+s}{1+r})^{t_i}$. Taking the logarithm, $log(\frac{p(t+t_i)}{q(t+t_i)}) = log(\frac{1+s}{1+r})t + log(\frac{p(t)}{q(t)})$. This linear function of $t$ allows us to know that A is $e^{m} = \frac{1+s}{1_r}$ times fitter than B, with $m$ being the slope of this linear function of $t$. 

An important aspect of somatic mutation is still missing from this model - that of multiple sites. In terms of the simulation itself, a clone is characterised by a set of $n$ mutations $S_{mut} = {mut_1,mut_2,...,mut_n}$ and each of these mutations confers a fitness advantage $s_i \geq 0$, where $s_i$ is the fitness advantage of $mut_i$. To combine multiple fitness effects, one can assume that these are additive - for example, if a clone $c_1$ is characterised by mutations $mut_42$ and $mut_13$, its fitness advantage is $s_42 + s_13$ and its proportion in the next generation ($t+1$) is given by $p_{c_1}(t+1) = \frac{1+s_{c_1}}{\bar{w}}p_{c_1}(t) = \frac{1+s_{42}+s_{13}}{\bar{w}}p_{c_1}(t)$. Now, rather than having a binomial distribution with $2N$ trials for a population of $N$ diploid individuals, it is multinomial distribution parameterised by a set of probabilities $S_{p_c} = {p_{c_1},p_{c_2},...,p_{c_n}}$, each multiplied by the fitness advantage conferred by the mutations on their respective clones. One advantage of this approach is that the expected and the variance of the proportion of the counts of any given clone $c_i$ for the next generation is analogous to the binomial case above - $E[X_{c_i}] = np'_{c_i}$ and $Var[X_{c_i}] = np'_{c_i}(1-'_{c_i})$, where $p'_{c_i} = (1+s_{c_i})p_{c_i}$. 

It is thought that the vast majority of somatic mutations acquired during life confer no fitness advantage ($s \approx 0$), aligning with the core tenet of the neutral theory of evolution - that most of the mutation variation encountered confers no change in selection. These neutral mutations are otherwise known as passenger mutations, whereas mutations conferring a measurable amount of selective advantage are known as driver mutations. In the context of multiple driver mutations in the same individual, competition is responsible for the triumph of fitter clones, leading these to reach detectable frequencies and, in more extreme cases, sweep the entire population.

Over the course of this chapter, I will use Wright-Fisher simulations as a tool to validate the growth rate estimates obtained from longitudinal targetted sequencing. While these simulations are able to track the entire trajectories of clones and mutations, they represent a sceneario in the study of somatic mutations in individuals which has not been replicated in any piece of scientific literature - the consistent tracking of mutations through life using deep sequencing. However, the coalescent is sufficient to establish the lifelong trajectories of clones without requiring access to a longitudinal study design.

\subsection{The coalescent}

The coalescent is a phylogenetic reconstruction of the relations between a finite set of individuals (in most cases sequences) up to their \ac{mrca}. The simplest way to build an example of a coalescent is to take a sample of $n$ individuals from the present and fixed-size population of $N$ individuals and to "go back in time" - assuming a present generation $t$, to get to the parents in generation $t-1$ a sample with replacement of the $N$ individuals is drawn for every individual in the sample at generation $t$ with probability $p$. Repeating this iteratively, until only on parent is identified - in other words, until the \ac{mrca} for the sample is found. In the coalescent, it is the probability $p$ that will determine whether or not the population is evolving neutrally - if $p$ is the same (uniform) for all parents in generation $t-1$, any single one has the same chance of being a parent of any one of the children in generation $t$. If $p$ is not uniform (i.e. one of the possible parents has $5$ times the probability of being a parent) this will lead to more children in generation $t$ having the same parent, which would be the expected case when a sample is under positive selection. The reason for the coalescent being simulated "backwards" is largely due to computational reasons - it is much easier to simulate the previous generation of a sample with $n$ elements than the next generation of a population of size $N$ (here $N>>n$). From this description we can see how this is remarkably similar to the Wright-Fisher model assuming neutrality - the population is constant and well-mixed (or panmictic), each generation is discrete and non-overlapping and populations evolve under neutral evolution. 

The coalescent is helpful to study a number of properties of a population - for instance, it is possible to calculate the probability that a coalescence will not happen at a given generation $t-1$ for a sample of $n$ individuals from a (diploid) population of size $2N$ in generation $t$: $p_{coal=0} = \frac{2N-1}{2N}\frac{2N-2}{2N}...\frac{2N-(n-1)}{2N} = (1 - \frac{1}{2N})(1 - \frac{2}{2N})...(1 - \frac{n+1}{2N})$ and, by ignoring terms which are similar to $\frac{1}{4N^2}$ or smaller and considering that $\sum^A_{a=1}{i} = \frac{A(A-1)}{2} = \binom{A}{2}$, $p_{no\ coal} = 1 - \frac{1}{2N} - \frac{2}{2N} -...- \frac{n-1}{2N} = 1 - \binom{n}{2}\frac{1}{2N}$. This makes the probability of observing at least one coalescent event in generation $t-1$ $p_{coal>0} = \binom{n}{2}\frac{1}{2N}$. In other words, as $N$ increases, the probability of a coalescence on the previous generation decreases, leading to trees that span more generations. Another consequence of this is that, as the coalescent progresses through generations, the probability of a coalescent event decreases as $n$ will also decrease, leading to trees with short branches and many coalescent events near the initial sample and longer branches as the \ac{mrca} approaches. We can generalize this probability to the probability of no coalescence over $\Delta t$ generations - $P(T > \Delta t) = p_{coal=0}^{\Delta t} = (1-\binom{n}{2}\frac{1}{2N})^{\Delta t}$. Using a similar formulation, Kingman showed, in 1982, that the coalescent times are independent and exponentially distributed such that their probability density function is $f(t_i) = \binom{n}{2}exp^{-\binom{n}{2}t_i}$, where $t_i$ is the time during which there are $i$ branches in the coalescent. This allows the calculation of the expected value and variance of the times between coalescent events $T_i$ - $E[T_i] = \frac{2}{i(i-1)}$ and $Var[T_i] = (\frac{2}{i(i-1)})^2$ \cite{Kingman1982-tc}. At this point, it is also important to note that the coalescent can also approximate the Wright-Fisher model under neutrality for large enough $N$ and with mutations, highlighting how the coalescent is analogous to genetic drift progressing backwards in time \cite{Wakeley2008-wh}. The coalescent and some of its properties - particularly the distribution of times between coalescent events - are the foundation of the estimation of the evolution of the \ac{eps}.

\subsection{Phylodynamic estimation of populations}

Phylodynamics is, historically, the field concerned with how specific processes - immunological, evolutionary or epidemiological - affect viral phylogenies \cite{Volz2013-ey}. Particularly and at its inception, changes to the viral population were suggested as having a sizeable impact on viral phylogenies \cite{Grenfell2004-ip} - indeed and thinking about the coalescent, where the probability of a coalescent event is in part determined by the size of the population, this is quite a reasonable statement. Following this line of thought also allows us to draw a few more conceptual conclusions from the aspect of a phylogeny - if the population increases exponentially with time, then coalescent events would have been more probable early on; if, on the other hand, the population remains constant, most coalescent events would occur closer to the time of sampling. It should be noted that, for phylogenetics in general and especially for phylodynamics, the size of the population describes the \textit{effective} population size, rather than some arbitrary population size. In other words, the population sizes I will discuss in the remainder of this chapter refer specifically to the size of the population capable of generating offspring.

While phylodynamics is concerned with the causes of a specific phylogenetic tree structure, the starting point for phylodynamic analyses is often the phylogenetic tree itself, reconstructing a few of the aspects of the tree that are nascent to it. For instance, the time for two members of a phylogeny to share a coalescent event - directly observable from a phylogenetic tree - is exponentially distributed with rate $\lambda(n) = \binom{n}{2}\frac{1}{N} = p_{coal>0}$ - in other words, the expected time for a coalescence to occur between two members is the inverse of the probability of a coalescent event occurring and can be used to infer how the population evolved. From this, it naturally follows that the expected time until the \ac{mrca} of a phylogenetic tree can be calculated as $\frac{1}{\lambda(n)} + \frac{1}{\lambda(n-1)} + ... + \frac{1}{\lambda(2)}$, which can be stated as $2N(1-\frac{1}{n})$; a natural consequence of this is that, as $n \rightarrow \inf$, the expected time until the phylogenetic tree converges on its \ac{mrca} is $2N$ \cite{Volz2013-ey}. 

The previous paragraph worked under the assumption that $N$ was constant through time - that the population size did not change along the phylogeny. However, it would be unwise to assume this is a fair assumption - in my case, where the main interest is the evolutionary dynamics of fit clones, this would be especially useless. As such, one has to think of these models in terms of a changing population size, $N(t)$ - $\lambda(n,t) = \binom{n}{2}\frac{1}{N(t)}$. From this consideration, I can consider, for illustrative purposes, a relatively simple model of population dynamics: since I analyze fit clones independently of one another by looking at the specific part of the tree they exist in (a clade), I can assume that each individual clade characterized by a driver mutation is evolving under neutral pressure, provided it does not acquire any other driver mutation at some point. This is helpful since it allows us to assume that a constant growth rate drives the expansion - for instance, if this growth rate is exponential with initial population size $N_0$ and rate $r$, we can say that the distribution of times between coalescent events is $\lambda(n,t) = \binom{n}{2}\frac{1}{N_0 e^{-rt}}$ \cite{Volz2013-ey}. This example can then be expanded to include changes in rates and dynamics, making the growth of the \ac{eps} more flexible in less parametric; this can be useful to capture, for example, cases of competition or catastrophic events - in the case of competition, one expects growth to be faster when no other clones are present and competing for the same resources, while in the case of a catastrophic event we can expect the \ac{eps} to grow more slowly or even to decrease. Another, perhaps better known example in the field of phylodynamics, is that of flu seasonality - during colder months, the \ac{eps} of H1N1 circulates through the population much more effectively, leading to a temporary increase in the effective population size \cite{Karcher2020-bg}. 



\subsection{Contribution}

The work presented in this chapter is partly available as a preprint in \cite{Fabre2021-uw}. Here, I ran a large set of Wright-Fisher simulations and used them to simulate a longitudinal targeted sequencing experiment and inferred dynamic coefficients for each mutation and clone, as well as the age at onset for each clone. Next, using a separate set of simulations, I infer the phylogenetic tree for a single sample at old age and use phylodynamic methods to estimate the dynamic aspects of each clone. Through this, I show how recently available technologies - longitudinal targeted sequencing and single-cell haematopoietic phylogenies - can be used to infer aspects of clonal dynamics.

\section{Methodology}

\subsection{Validating the dynamic coefficient and age at onset inference with Wright-Fisher simulations}

\subsubsection{Bayesian inference of dynamic coefficients}

The quantification of clonal dynamics requires the specification of a model - here, we use a relatively simple parametrisation assuming that counts for mutation $i$ in individual $j$ for a given age $t$ can be modelled according to a Beta-binomial distribution with parameters $\alpha_{ij}(t)$ and $\beta$ for a given coverage $cov$ such that $counts_{ij}(t) \sim BB(cov,\alpha_{ij}(t),\beta)$. Here, $\beta$ refers to the technical overdispersion and $alpha_{ij}(t) = \frac{p_{ij}(t)\beta}{1-p_{ij}(t)}$. Here, $p_{ij}(t) = \frac{sigmoid((b_{mut_i} + b_{clone_{ij}}) * t + u_{ij})}{2}$, with $b_{mut_i}$ corresponding to the growth advantage conferred by mutation $i$, $b_{clone_{ij}}$ corresponding to changes in growth from $b_{mut_i}$ in mutation $i$ for individual $j$ (in other words, the growth advantage attributable specifically to $clone_{ij}$) and $u_{ij}$ as an offset to account for the time at which the clone appeared. This parametrisation implicitly assumes that $counts_{ij}(t)$ are, at most, half of $cov$, while having the fortunate consequence that $E[counts_{ij}(t)] = cov.p_{ij}(t)$. We assume that $b_{mut_i} \sim N(0,0.2)$, $b_{clone_{ij}} \sim N(0,0.05)$, $u_{ij} \ sim Uniform(-50,0)$ and $\beta \sim N(\mu_{od},\sigma_{od})$. The inference of hyperparameters $(\mu_{od}$ and $\sigma_{od}$ is detailed ahead in this subsection. To validate this model, we sample the parameters $b_{mut_i}$, $b_{clone_{ij}}$, $u_{ij}$ and $\beta$ using Hamiltonian Monte Carlo on a set of simulated longitudinal sequencing data under a Wright-Fisher model. 

\subsubsection{Simulating a longitudinal targeted sequencing setup}

I simulate populations of \ac{hsc} under a Wright-Fisher model \cite{Beerenwinkel_undated-up} over 1,300 generations and with 1,000 passenger and 50 driver sites. I use a constant passenger mutation rate of $2*10^{-6}$ and driver mutation rates between $1.0*10^{-10}$ and $4.0*10^{-9}$ per generation and fitness advantages between $0.001$ and $0.030$. I use a fixed population size of 200,000 \ac{hsc} and assume that every year there are 13 cell divisions, in line with previous estimates \cite{Lee-Six2018-lp,Watson2020-pz}. For every combination of mutation rates and fitness advantages, I conduct three distinct simulations. In the end, I had a total of 2,250 simulations for which the true simulated fitness effect and the generation at clone onset were known.

Having the general parameters for the simulations, I first simulate the conditions under which our data were acquired - over a median period of 13 years, specific driver mutations were sequenced between 3-5 times in older individuals. To simulate these conditions, I fit Gamma distributions to the observed coverage and observed age at first timepoint and truncate these at the minimum and maximum values for both coverage and age. After this, I follow a simple algorithm for each simulation:

\begin{enumerate}
    \item Sample an age from the age at first timepoint distribution;
    \item Sample the number of subsequent timepoints (between 2 and 4) which are separated by approximately 3 years (40 generations);
    \item Verify that a driver mutation is present in two or more timepoints;
    \item Sample the coverage for each of these timepoints;
    \item Using the technical overdispersion and the true allele frequency of the driver as $p$ to derive $\alpha = \frac{\beta p}{1-p}$ as in the previous subsection, sample assuming that the counts follow a Beta-binomial distribution such that $counts \sim BB(\alpha,\beta,cov)$;
\end{enumerate}

\subsubsection{Coefficient inference and age at onset estimation}

Using the aforementioned simulation settings and this sampling algorithm I derive a comprehensive set of simulated data that spans different mutation rates and fitness effects with a similar age and coverage distribution to the one observed in the real data. The inference of coefficients was then done after transforming the generation at sampling into years (assuming that each year contains 13 generations) and using the model specified above with Hamiltonian Monte Carlo with a number of leapfrog steps between 100 and 200 and drawing 2,000 samples (out of which the first 1,000 were discarded as burn-in samples). 

To determine the age at onset of a clone - in other words, the age at which there was a single cell with a mutation - I consider that growth for a given clone with mutation $i$ in individual $j$ with fitness advantage $s_{ij}$ (per generation) under a Wright-Fisher model can be split into two distinct regimes (\figref{fig:wf-example}):

\begin{enumerate}
    \item When a clone exists in relatively small numbers, the stochastic nature of genetic drift overpowers its growth trajectory, making this regime (hereby referred to as the stochastic regime) largely stochastic, with the clone growing linearly ($n(g) = g$, where $n$ is the size of the clone and $g$ is the number of generations);
    \item After reaching a certain size - when the number of cells constituting a clone becomes larger than $1/s_{ij}$ - a clone starts growing at a rate that is approximately deterministic and determined by $s_{ij}$.
\end{enumerate}

\begin{figure}
	\placefigure{gfx/wf-sim-example.pdf}
	\placecaption{Example of a Wright-Fisher simulation.}
	\label{fig:wf-example}
\end{figure}

Taking this and the fact that growth under our longitudinal model is approximately exponential, it is necessary to first determine when the clone reached $1/s_{ij}$ cells and started growing deterministically - in other words, the equation $\frac{s_{ij}}{gN} = e^{s_{ij} * t + u_{ij}}$ must be solved for $t$, where $N$ is the maximum population size. Next, knowing that the clone is expected to grow linearly during the stochastic regime it is possible to calculate the expected time spent by the clone in this phase of growth as $t_{stoch} = \frac{1}{sg}$. Solving the first equation for $t$ and subtracting $t_{stoch}$, the result is $t_{onset,ij} = \frac{log(g/s_{ij}/N)-1-s_{ij}}{s_{ij}}$. I note here that different assumptions for $N$ and $g$ yield different results - as such, I confirm that our results are still consistent despite small errors in our assumptions by running two additional sets of simulations - assuming $g=1$ and $N=50,000$, as well as $g=5$ and $N=100,000$ (I base these combinations on the work done by \etal{Lee-Six} \cite{Lee-Six2018-lp}). Additionally, I also tested different values for $g$ (between $1$ and $20$) and $N$ (between $10,000$ and $600,000$) to provide a thorough confirmation that these do not influence our conclusions in terms of the age at onset.

Finally, I calculated the coefficient of determination between the simulated and inferred coefficients to assess the validity of our estimates, repeating this process for the simulated and inferred ages at onset. To assess the validity of our \ac{mcmc} samples, I use two distinct diagnostic statistics - the effective sample size and the Gelman-Rubin statistic \cite{Gelman1992-zo} - as well as the visual inspection of \ac{mcmc} chains.

\subsubsection{Inferring the technical overdispersion of targeted sequencing using technical replicates}

To infer the parameters $\mu_{od}$ and $\sigma_{od}$, I had access to two sets of experimental data:

\begin{enumerate}
	\item Between two and three different solutions of known \ac{vaf} concentrations (0, 0.005, 0.01, 0.02, 0.05) for 4 different genes (\ac{idh1}, \ac{jak2}, \ac{kras} and \textit{NRAS}) were sequenced at a median depth of 415x;
	
	\item For 5 individuals, technical replicates were elaborated by sequencing two additional samples at all timepoints. These individuals had at least one mutation across 15 different genes (\ac{asxl1}, \textit{BCOR}, \ac{cbl}, \ac{dnmt3a}, \textit{EZH2}, \textit{GNAS}, \ac{idh1}, \ac{jak2}, \textit{NOTCH1}, \textit{PHF6}, \ac{sf3b1}, \textit{SMC1A}, \ac{srsf2}, \ac{tet2}, \textit{ZRSR2}) with \ac{vaf} between 0 and 0.394. 
\end{enumerate}

For 1., I model the distribution over the expected \ac{vaf} as a beta distribution such that $VAF \sim Beta(\alpha,\beta)$, where $\alpha=\frac{p\alpha}{1-p}$ and $p$ is the observed \ac{vaf} and for 2. I adopt a model identical to the one described in the previous section using only gene growth effects. Here, I model $\beta \sim exp(r)$ with $r$ as a variable with no prior. I use an identical process as in the previous section - \ac{mcmc} with \ac{hmc} - with 400-500 leapfrog-steps as implemented in greta \cite{Golding2019-wh} to estimate the mean and standard deviation of  using both models for 1. and 2. in parallel. For this estimate I use 1,000 samples from the posterior distribution.

\subsection{Validating phylodynamic estimation with Wright-Fisher simulations}

\subsubsection{Simulations and tree construction}

Similarly to the previous section, I start by simulating a set of scenarios to assess the validity of the fitness estimation protocol I applied to real data. Here, I simulated a population of 200,000 \ac{hsc}, each having 7,500 passenger mutations and 50 drivers and selected 6 distinct combinations of fitness effect and mutation rate for the 50 possible driver mutations - $[0.005,200*10^{-9}; 0.01,50*10^{-9}; 0.015,20*10^{-9}; 0.02,15*10^{-9}; 0.025,8*10^{-9}; 0.03,5*10^{-9}]$ - and each simulation was repeated 20 times for 800 generations, yielding a total of 120 simulations. For each of these simulations, a phylogenetic tree was built by following a relatively simple protocol:

\begin{enumerate}
    \item For each simulation I take a representative sample of 100 clones from the last timepoint ($g = 800$). To do so, the probability of each clone being sampled is given by its relative frequency;
    \item Using the complete mutational landscape of each clone, I build trees using a neighbour-joininig algorithm. This algorithm is relatively simple and iteratively builds the tree by joining the two most similar tips/nodes. 
\end{enumerate}

For each tree, I detect clades by defining them as a group of tips and branches whose \ac{mrca} has the earliest instance of the driver they share. I also extracted the trajectories for the drivers out of each simulation considering the possibility that two different clones could have the same driver - for instance, if a clone $a$ acquired mutation $i$ at time $t_a$ and clone $b$ acquired mutation $i$ at time $t_a$ these are considered two distinct drivers rather than a single one - this enables the comparison between driver trajectories and trajectories inferred from each clade. In practical terms, I consider that a clade and a driver trajectory correspond to the same clone if the mutations characterising them have the smallest Hamming distance out of all clade-driver trajectory combinations. 

\subsubsection{Phylodynamic inference of clonal trajectories from phylogenetic trees}

To determine the clonal trajectory of each expansion using the information contained in its corresponding clade - particularly, the distribution of times between coalescent events - we use \ac{bnpr}, a method implemented in the \texttt{phylodyn} \cite{Lan2015-sw,Karcher2017-kt} package for \texttt{R}. \ac{bnpr} estimates effective population sizes on a grid of sampled points at resolution $\tau$ between the tips and the first observable coalescence of each tree (or clade). In effect, this approach seeks to estimate the posterior $P[f,\tau|g]\alpha P[g|f]P[f|\tau]P[\tau]$, where $f$ is the piecewise linear estimate in the \ac{eps} estimator $N(t)$ for a grid of points along the tree between $x_d$ and $x_D=t_2$ such that $N(t) = \sum^{D-1}_{d-1}exp(f)1_{(x_d,x_{d+1}]}$ and $g$ is the inferred genealogy (our phylogenetic tree or clade). $f$ is parameterized as a Gaussian process and describes the population trajectory between coalescent events. In practical terms, the estimation process for $f$ is approximated using the integrated nested Laplacian approximation, with more concrete details in \cite{Lan2015-sw} (theoretical foundation) and \cite{Karcher2017-kt} (practical application and implementation). In other words, \ac{bnpr} estimates the \ac{eps} trajectory from the tips of a tree to its first coalescence, which can then be used to determine the clonal dynamics of specific expansions. I chose this method because it had been already used previously in the landmark paper on haematopoietic dynamics by \etal{Lee-Six}, is reasonably fast and, as I will be showing further ahead, provides good estimates for the \ac{eps} and allows the calculation of a consistent estimate of the true fitness of clones.

Calculating the \ac{bnpr} \ac{eps} estimates for all the trees in the clade assumes no fixed fitness effect. However, it is expected that each mutation confers a specific fitness effect, which should be determinant in these trajectories. As such, to determine the best way to infer fitness from \ac{bnpr} trajectories I fit three different models to all \ac{eps} trajectories $Neff(t)$, namely:

\begin{enumerate}
    \item A log-linear fit which assumes that clonal growth is exponential - $log(Neff'(t)) = t * b + a$, where $b$ and $a$ are determined using least squares;
    \item A scaled and shifted sigmoid fit which assumes exponential but saturating growth - $Neff'(t) = \frac{scale}{1+e^{\frac{xmid-input}{b}}}$, where $scale$, $xmid$ and $b$ are determined using a non-linear optimization routine (\texttt{nl2sol}) from the Port library \cite{noauthor_undated-gs}. I initialize variables in such a way that guarantees convergence - $scale=max(Neff(t))$, $xmid=max(t)$ and $b=10$;
    \item A biphasic log-linear fit which assumes there are two distinct phases of growth separated by a breakpoint - $log(Neff(t)) = t * b_1 + a, if\ t < bp; log(Neff'(t)) = t * b_2 + a, if\ t \geq bp$, where $b_1$, $b_2$, $a$ and $bp$ are determined using \ac{l-bfgs-b}, a quasi-Newthonian method which uses a limited amount of computer memory. Similarly to 2., I initialize variables to facilitate convergence - $b_1=b$, $b_2=b$, $a=a$, $bp=mean(t)$, where $a$ and $b$ are the estimates from 1. I also constrain estimates of $bp$ to be between $min(t) + 0.25*range(t)$ and $max(t) + 0.75 * range(t)$.
\end{enumerate}

For all three estimates described above, I weigh data according to its uncertainty from the \ac{eps} estimates - \ac{bnpr} produces estimates for the 95\% credible interval, which I use to derive an approximate estimate of the logarithm of the variance as $log(var(t))' = (log(Neff(t)_{97.5\%}) - log(Neff(t)_{2.5\%}))^2/16$ and we use its inverse. Finally, I compare all three types of estimate by assessing how closely they are able to recapitulate the simulated fitness. To do so, I calculate their coefficient of determination and root mean squared error. I also visually assess how similar these trajectories are to the true driver trajectories as reconstructed from simulations.

To confirm the impact of this particular method of phylodynamic estimation - \ac{bnpr} - on the \ac{eps} trajectories, I use two additional methods for phylodynamic estimation as implement in the \texttt{ape} package for \texttt{R} \cite{Paradis2019-na}: \texttt{skyline} (an implementation of the generalised skyline plot, a non-parametric method for \ac{eps} estimation that assumes that the \ac{eps} for a given interval is proportional to the accumulated waiting time between coalescent events and the number of lineages while inversely proportional to previous coalescent events) and \texttt{mcmc.popsize} (a method with similar assumptions to those of \texttt{skyline} but uses reversible jump \ac{mcmc} to derive a smoother version of the \ac{eps} estimate) \cite{Opgen-Rhein2005-pi}. I use both additional methods to verify that estimates from \ac{bnpr} are consistent with those from other methods while enabling a more accurate recapitulation of the true fitness effect.

\section{Results}

\subsection{A simple model to recapitulate Wright-Fisher dynamics and infer age at clonal onset}

Using the simulations described above, I infer coefficients for the growth attributable to mutation $i$ - $b_{mut_i}$ - and for the growth attributable to clone $ij$ - $b_{clone_{ij}}$ and do this jointly for all the simulation experiments. By doing so, it is possible to faithfully estimate the part of growth that is attributable specifically to the mutation and disentangle it from clone-specific effects. First, I verify that the \ac{mcmc} chains converge adequately by visually inspecting them (\figref{fig:mcmc-chains-sim}) and assessing the effective sample size and Gelman-Rubin statistic for all coefficients (\figref{fig:mcmc-ess} and \figref{fig:mcmc-gelman-rubin}). Observing that values for \ac{ess} are routinely above 500 and that the values for the Gelman-Rubin statistic are close to the unit, I can confirm that the simulation has converged.

\begin{figure}
	\missing{figure}
	\placecaption{\Ac{mcmc} chains for the mutation coefficients inferred using data simulated under a Wright-Fisher model.}
	\label{fig:mcmc-chains-sim}
\end{figure}

\begin{figure}
	\missing{figure}
	\placecaption{Effective sample size for all dynamic parameters inferred using data simulated under a Wright-Fisher model.}
	\label{fig:mcmc-ess}
\end{figure}

\begin{figure}
	\missing{figure}
	\placecaption{The Gelman-Rubin statistic calculated for all dynamic parameters using data simulated under a Wright-Fisher model.}
	\label{fig:mcmc-gelman-rubin}
\end{figure}

Knowing that the \ac{mcmc} chains converge adequately, I use them to show that using this model it is possible to calculate the fitness of drivers with a $R^2 = 91.5\%$ (\figref{fig:sim-vs-inf-driver}). Visual inspection of these trajectories while comparing them with the original trajectories provides further confirmation of the effectiveness of this model in recapitulating clonal growth (\figref{fig:trajectory-examples-sim}). Here, I use the average of all chains to estimate the expected value of each coefficient and calculate the 90\% \ac{hdpi} for all coefficients. In practical terms, the 90\% \ac{hdpi} is the shortest-spanning interval containing 90\% of the samples in our simulation and it is observable that the \ac{hdpi} for each coefficient contains - or is close to containing - the true value of the coefficient. I also measure the likelihood of the simulated data under our model and define an outlier as a value not contained within the 95\% confidence interval defined by our model and consider, from here on, all trajectories with outliers as invalid and do not proceed with their analysis. Taking these standards, 87\% of all trajectories are explained by our model and we observe that this remains relatively constant for different true fitness intervals (\figref{fig:explained-traj-sim}).

\begin{figure}
	\placefigure{gfx/simulated-vs-inferred-fitness.pdf}
	\placecaption{Inferred driver effect as a function of the simulated driver effect.}
	\label{fig:sim-vs-inf-driver}
\end{figure}

\begin{figure}
	\placecaption{Examples of simulated and inferred trajectories for different true fitness levels (s).}
	\label{fig:trajectory-examples-sim}
\end{figure}

\begin{figure}
	\placecaption{Explained trajectories for different true fitness ranges.}
	\label{fig:explained-traj-sim}
\end{figure}

To obtain an age, rather than a generation, at clone onset, I divide the generations by 13 under our assumption of 13 generations per year. Using this value, I am able to compare the inferred and simulated age at onset to assess whether the method stated above to calculate the age at onset - $t_{onset,ij} = \frac{log(g/s_{ij}/N)-1-s_{ij}}{s_{ij}}$ - allows us to recapitulate the true age at onset. To generate each estimate for the age at onset, I calculate for each sample of $b_{mut_i}$, $b_{clone_{ij}}$ and $u_{ij}$ their respective $t_{onset,ij}$ and, for each clone (each having $1,000$ samples), I estimate the median and the 90\% confidence interval for the age at onset. I note here that a relatively low number of clones - 10.3\% of all clones - had a median age at onset that preceded birth. This is due to an underestimation of the true fitness of the clone, especially considering that competition or clones with smaller fitness advantages are present in these simulations. To address this problem and knowing that it is highly unlikely that clones appeared before birth (especially in the context of these simulations) I set a lower bound for the age at onset estimates that precede birth - in other words, if $t_{onset,ij} < 0$ then I correct this to $t_{onset,ij}=0$. Taking these corrections into consideration, I now compare the true with the inferred age at onset for these simulated clones and show that it is possible to estimate the true age at onset using a relatively simple longitudinal sequencing protocol and statistical modelling ($R^2 = 65.8\%$, \figref{fig:age-at-onset-sim}). Additionally, while these estimates can have a relatively high spread, their aggregation shows they appear to be a consistent estimator of the age at onset as represented with the boxplots in \figref{fig:age-at-onset-sim}. 

\begin{figure}
	\placecaption{Inferred age at onset as a function of the simulated age at onset. Each boxplot represents the distribution of ages at onset for 10 year intervals.}
	\label{fig:age-at-onset-sim}
\end{figure}

An additional and relatively intuitive finding from this analysis is that the \ac{mae} for the age at onset estimates is considerably higher for smaller fitness advantages ($MAE_{fitness < a} = $ \missing{value}, $MAE_{fitness < b} = $ \missing{value}, \figref{fig:age-at-onset-vs-true-fitness-sim}) - this conclusion highlights the fact that lower fitness advantages make clones more likely to be affected by competition or genetic drift, making the estimation of their ages at onset more biased due to greater relative errors in the inference of their fitness effect. 

\begin{figure}
	\placecaption{Inferred age at onset as a function of the simulated age at onset. Each boxplot represents the distribution of ages at onset for 10 year intervals.}
	\label{fig:age-at-onset-vs-true-fitness-sim}
\end{figure}

As made clear by the formula for $t_{onset,ij}$, there is some dependency on the assumption of values for $N$ and $g$. Taking this into consideration, I investigate the effect of wrong values for $N$ and $g$ on the age at onset estimates. This shows that there is a set of values for $N$ and $g$ for which the age at onset estimates are relatively similar in their \ac{mae} when compared with the simulated age at onset as shown in \figref{fig:age-at-onset-mae}. Using the correct values, we can be sure that our solution is, on average, within 8.72 years of the real value. Nonetheless, it should also be noted that incorrect assumptions for $N$ and $g$ do not render incorrect values due to their dependence - in the formula for $t_{onset,ij}$, a single term has both $N$ and $g$ ($log(g/s_{ij}/N)$) and, as such, it is reasonable that feasible solutions for $g$ and $N$ occur, as long as the $N/g$ remains close to that of the true value - in our case, $N/g = \frac{200,000}{13} \approx 15384$.

\begin{figure}
	\placecaption{Inferred age at onset as a function of the simulated age at onset. Each boxplot represents the distribution of ages at onset for 10 year intervals.}
	\label{fig:age-at-onset-mae}
\end{figure}

Finally, I was interested in understanding if there was a reasonable way of estimating a likely solution for the real values of $N$ and $g$ given the distribution of ages at onset for different clones. To do this, an additional set of clones were simulated under a Wright-Fisher model with two different combinations of $N$ and $g$ - $N=100,000;g=5$ and $N=50,000;g=1$. I note here that these constitute slightly different values for $N/g$, particularly $20,000$ and $50,000$, respectively. 

The motivation for this investigation was relatively simple - assuming identical mutation rates, we would expect clones to appear in a more or less uniform manner over life. Under the simulated experimental conditions here stated, however, this expectation may not hold for the detected clones - it would not be possible to detect very small clones or clones that did not manage to fixate in the population. As such, I first assess whether the distribution of simulated ages at onset could be considered uniform by any measure. 

\subsection{}

\subsection{Early growth is a consistent estimator of clone fitness using phylodynamic estimation}

\section{Conclusion}