\chapter{Methods for inferring the evolutionary dynamics of haematopoietic clones}

\section{Studying the haematopoietic evolution through longitudinal sequencing}

\subsection{A Wright-Fisher model of haematopoietic stem cell evolution}

Previous results from \etal{Lee-Six} suggest that the \ac{hsc} population is likely to have an \ac{eps} between 50,000 and 200,000 \ac{hsc}, with anything between 2 and 13 divisions --- or generations --- per year ($g$) \cite{Lee-Six2018-lp}, while those from \etal{Watson} suggest similar values for \ac{eps} and $g$ while describing the range of possible fitness advantages ($s$) and mutation rates ($u$) in this population. I simulate populations of \ac{hsc} under a Wright-Fisher model (\figref{fig:wf-example}) \cite{Beerenwinkel_undated-up} over 1,300 generations and with 1,000 passenger and 50 driver sites. I use a constant passenger mutation rate of $2 \times 10^{-6}$ and driver mutation rates between $1.0 \times 10^{-10}$ and $4.0 \times 10^{-9}$ per generation and fitness advantages between $0.001$ and $0.030$. I use a fixed population size of $N=200,000$ \ac{hsc} and assume that every year there are $g=13$ cell divisions per year, in line with previous estimates \cite{Lee-Six2018-lp,Watson2020-pz} --- this means that each \ac{hsc} population was simulated for 100 years (1,300 generations). For every combination of $u$ and $s$, I conduct three distinct simulations. In the end, I simulated a total of 2,250 \ac{hsc} populations for which the true simulated fitness effect and the generation at clone onset were known. 

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/wf-sim-example.pdf}
	\placecaption{Example of a Wright-Fisher simulation.}
	\label{fig:wf-example}
\end{figure}

Under the Wright-Fisher model, growth for clones with driver mutations can be divided into two distinct phases:

\begin{enumerate}
    \item When a clone exists in relatively small numbers, the stochastic nature of genetic drift overpowers its growth trajectory, making this regime (hereby referred to as the stochastic or drift regime) largely stochastic, with the clone growing linearly ($f(g) = g/N$, where $f$ is the relative size of the clone and $g$ is the number of generations since the clone's inception);
    \item After reaching a certain size --- when the number of cells constituting a clone becomes larger than the inverse of its fitness $1/s$ --- a clone starts growing at a rate that is approximately deterministic and determined by $s$.
\end{enumerate}

\subsection{Sampling in a longitudinal sequencing experiment}

Having access to the population trajectories of different driver mutations, I am able to calculate not only when each mutation was acquired but also its allele frequency $f$ at a given point in time (how many clones from the total population of $N=200,000$ \ac{hsc} harbour a specific mutation). Using this information, it is possible to sample the allele frequency of specific mutations of interest. To do so, the simplest would be to assume that the counts $c$ follow a binomial distribution parameterized by a given coverage $n$ and a probability $p$, identical to $f$ (or, in the case of heterozygous mutations, $\frac{f}{2}$ such that $c \sim \mathrm{Binom}(n,\frac{f}{2})$). However, technical overdispersion will have an impact on $p$; more concretely, slight variations in sampling will lead to values of $p$ which follow an overdispersed distribution. Here, I assume that this overdispersion ($\beta$) is quantifiable and that a Beta distribution is an accurate model of the distribution of $p$ such that $p \sim \mathrm{Beta}(\frac{f}{2},\beta)$. Reparameterizing for the usual Beta distribution parameters ($\alpha$ and $\beta$) and ensuring that $E[p] = \frac{f}{2}$, I derive $\alpha$ as $\alpha=\frac{\frac{f}{2}\beta}{1-\frac{f}{2}}$ and consequently $p \sim \mathrm{Beta}(\frac{\frac{f}{2}\beta}{1-\frac{f}{2}},\beta)$. This leads to the counts $c$ following a Beta-binomial distribution $c \sim \mathrm{BB(n,\frac{\frac{f}{2}\beta}{1-\frac{f}{2}},\beta)}$. 

Using the relatively large set of simulations described above, I simulate the conditions under which the real data, described in Chapter 3, were acquired --- over a median period of 13 years in older individuals, driver mutations were sampled between 3-5 times as described below. To better match the age and coverage distributions under which the data was acquired, I fit Gamma distributions to the observed coverage and observed age at first timepoint and truncate these at the minimum and maximum values for both quantities. After this, I follow a simple algorithm for each simulation:

\begin{enumerate}
    \item Sample an age from the age at first timepoint distribution;
    \item Sample the number of subsequent timepoints (between 2 and 4) which are separated by approximately 3 years (40 generations);
    \item Verify that a driver mutation is present in two or more timepoints;
    \item Sample the coverage for each of these timepoints;
    \item Using the technical overdispersion (more details on this parameter will be provided below) and the true allele frequency of the driver as $\frac{f}{2}$ to derive $\alpha = \frac{\beta p}{1-p}$ as previously described, sample assuming that the counts follow a Beta-binomial distribution such that $c \sim \mathrm{BB}(\alpha,\beta,n)$;
\end{enumerate}

In the data used in Chapter 3, where I will apply the models here developed, the maximum \ac{vaf} is 43.8\%, I exclude all trajectories with any point greater than 45\%.

\subsection{Bayesian inference of fitness coefficients}

The quantification of clonal dynamics requires the specification of a model --- here, I use a relatively simple parametrization assuming that counts for mutation $i$ in individual $j$ for a given age $t$ can be modelled according to a Beta-binomial distribution with parameters $\alpha_{ij}(t)$ and $\beta$ for a given coverage $n$ such that $c_{ij}(t) \sim \mathrm{BB}(n,\alpha_{ij}(t),\beta)$. Here, $\beta$ refers once again to the technical overdispersion and $\alpha_{ij}(t) = \frac{p_{ij}(t)\beta}{1-p_{ij}(t)}$. Here, $p_{ij}(t) = \frac{\mathrm{sigmoid}((b_{\mathrm{mut}_i} + b_{\mathrm{clone}_{ij}}) \times t + u_{ij})}{2}$, with $b_{\mathrm{mut}_i}$ corresponding to the growth advantage conferred by mutation $i$, $b_{clone_{ij}}$ corresponding to changes in growth from $b_{\mathrm{mut}_i}$ in mutation $i$ for individual $j$ (in other words, the growth advantage attributable specifically to $\mathrm{clone}_{ij}$) and $u_{ij}$ as an offset to account for the time at which the clone appeared. This parametrization implicitly assumes that the counts $c_{ij}(t)$ are expected to be, at most, half of $n$ while, similarly to the previous subsection, $E[c_{ij}(t)] = np_{ij}(t)$. I assume that $b_{mut_i} \sim N(0,0.2)$, $b_{\mathrm{clone}_{ij}} \sim N(0,0.05)$, $u_{ij} \sim \mathrm{Uniform}(-50,0)$ and $\beta \sim \mathrm{N}(\mu_{\mathrm{od}},\sigma_{\mathrm{od}})$. The inference of hyperparameters $(\mu_{\mathrm{od}}$ and $\sigma_{\mathrm{od}}$ is detailed ahead in this subsection. To infer under this model, I sample the parameters $b_{\mathrm{mut}_i}$, $b_{\mathrm{clone}_{ij}}$, $u_{ij}$ and $\beta$ using \ac{hmc} on a set of simulated longitudinal sequencing data under a Wright-Fisher model. \Ac{hmc} is an algorithm built on top of standard \ac{mcmc} that uses Hamiltonian dynamics to guarantee that the new sample is guaranteed to not be rejected and uncorrelated with the current sample, making it much more efficient. To do this, \ac{hmc} draws several samples at each step and estimates the direction (velocity) of the next sample based on the current sample through numerical integration, a process known as "leapfrog integration" \cite{Neal2011-lj}. Using \ac{hmc} with between 100 and 200 leapfrog steps, I draw 2,000 samples and discard the first 1,000 to estimate the posterior distribution of the parameters in my model.

As a quality-control step I calculate the probability of each true mutant count value under this model - $p(c_{ij}|b_{\mathrm{mut}_j},b_{\mathrm{clone}_{ij}},u_{ij},\beta,t)$ - and anything with a tail probability below 2.5\% is considered to be an outlier. Consequently, I discard all trajectories with outliers, using only those with no outliers for any subsequent analysis.

\subsubsection{Coefficient inference and age at onset estimation}

Using the aforementioned simulation settings and this sampling algorithm I derive a comprehensive set of simulated data that spans different mutation rates and fitness effects with a similar age and coverage distribution to the one observed in the real data. The inference of coefficients was then done after transforming the generation at sampling into years (assuming that each year contains 13 generations) and using 1,000 samples from the posterior described above to calculate the distribution parameters of each coefficient. 

To determine the age at onset of a clone --- in other words, the age at which there was a single cell with a mutation --- I consider that, as mentioned, growth under the Wright-Fisher model can be split into two distinct regimes. Taking this and the fact that growth under the stated longitudinal model is approximately exponential into consideration, it is necessary to first determine when the clone reached $1/s_{ij}$ cells and started growing deterministically --- in other words, the equation $\frac{s_{ij}}{gN} = e^{s_{ij} \times t + u_{ij}}$ must be solved for $t$, where $N$ is the maximum population size. Next, knowing that the clone is expected to grow linearly during the stochastic regime it is possible to calculate the expected time spent by the clone in this phase of growth as $t_{\mathrm{stoch}} = \frac{1}{sg}$. Solving the first equation for $t$ and subtracting $t_{\mathrm{stoch}}$, the result is \equationref{eq:t-onset}. I note here that different assumptions for $N$ and $g$ yield different results and confirm that these results are still consistent in the case of small errors in my assumptions by running two additional sets of simulations --- assuming $g=1$ and $N=50,000$, as well as $g=5$ and $N=100,000$ (I base these combinations on the work done by \etal{Lee-Six} \cite{Lee-Six2018-lp}). Additionally, I also tested different values for $g$ (between $1$ and $20$) and $N$ (between $10,000$ and $600,000$) to provide a thorough confirmation that these do not influence the conclusions drawn in terms of the age at onset.

\begin{equation}\label{eq:t-onset}
	t_{\mathrm{onset},ij} = \frac{\log(g/s_{ij}/N)-1-s_{ij}}{s_{ij}}
\end{equation}

Finally, I calculated the coefficient of determination between the simulated and inferred coefficients to assess the validity of my estimates, repeating this process for the simulated and inferred ages at onset. To assess the validity of the \ac{mcmc} samples, I use two distinct diagnostic statistics --- the \ac{ess} and the Gelman-Rubin statistic \cite{Gelman1992-zo} --- as well as the visual inspection of \ac{mcmc} chains. The Gelman-Rubin statistic (also known as the Gelman-Rubin convergence diagnostic) provides a quantity to determine whether all the chains in the model converged to relatively similar values by calculating the variances between and within chains. Ideally, this value will be, inferior to 1.2 or 1.1 depending on less or more stringent criteria, respectively \cite{Brooks1998-jx}. The \ac{ess} is a concept burrowed from the statistical analysis of surveys and quantifies the sample size for which the estimates of the mean would be that of a completely uncorrelated sample \cite{Kish1965-ei}. In other words, when the sample size is very low the estimation of the parameters --- mean, standard deviation, quantiles, etc. --- of a distribution is unlikely to be accurate.

\subsubsection{Inferring the technical overdispersion of targeted sequencing using technical replicates}

To infer the parameters $\mu_{\mathrm{od}}$ and $\sigma_{\mathrm{od}}$, I had access to two sets of experimental data:

\begin{enumerate}
	\item Between two and three different solutions of known \ac{vaf} concentrations (0, 0.005, 0.01, 0.02, 0.05) for 4 different genes (\ac{idh1}, \ac{jak2}, \ac{kras} and \textit{NRAS}) were sequenced at a median depth of 415x. I refer to this set of data as \textbf{known dilution}. These were obtained using Tru-Q DNA Reference Standards, a quality control material generally used to benchmark sequencing protocols;
	
	\item For 5 individuals, technical replicates were elaborated by sequencing two additional samples at all timepoints. These individuals had at least one mutation across 15 different genes (\ac{asxl1}, \textit{BCOR}, \ac{cbl}, \ac{dnmt3a}, \textit{EZH2}, \textit{GNAS}, \ac{idh1}, \ac{jak2}, \textit{NOTCH1}, \textit{PHF6}, \ac{sf3b1}, \textit{SMC1A}, \ac{srsf2}, \ac{tet2}, \textit{ZRSR2}) with \ac{vaf} between 0 and 0.394. I refer to this set of data as \textbf{technical replicates}.
\end{enumerate}

For the \textbf{known dilutions}, I model the distribution over the expected \ac{vaf} as a beta distribution such that $\mathrm{VAF} \sim \mathrm{Beta}(\alpha,\beta)$, where $\alpha=\frac{p\alpha}{1-p}$ and $p$ is the observed \ac{vaf} and for the \textbf{technical replicates} I adopt a model identical to the one described in the previous section using only gene growth effects. Here, I model $\beta \sim \mathrm{Exp}(r)$ with $r$ as a variable with no prior. I use an identical process as in the previous section --- \ac{mcmc} with \ac{hmc} --- with 400-500 leapfrog-steps as implemented in \texttt{greta} \cite{Golding2019-wh} to estimate the mean and standard deviation of using both models for \textbf{known dilution} and \textbf{technical replicates} in parallel. For this estimate I use 1,000 samples from the posterior distribution.

\subsection{Simulation results}

\subsubsection{Estimation of the technical overdispersion coefficient}

To simulate and model targeted sequencing experiments I started by estimating the technical overdispersion associated with sequencing. To do this I had two datasets --- \textbf{known dilution}, which consisted of prepared sequencing read solutions with known concentrations (\figref{fig:truq-data}) and \textbf{technical replicates}, which consisted on technical triplicate data for 4 individuals (\figref{fig:replicate-data}; more details on the cohort to which these individuals belonged will be made available in Chapter 3). For the Replicate dataset, it should be highlighted that the \textbf{technical replicates} were obtained from the same collection; in other words, the variation is expected to be due to technical aspects of sequencing rather than stochastic sampling of blood cells. 

I model both datasets and the associated technical overdispersion as described above using \ac{hmc} and obtain an estimate for the mean and standard deviation of the overdispersion by calculating both directly using posterior samples (218.72 and 29.50,respectively). I observed good convergence of the chain for the overdispersion parameters (\figref{fig:overdispersion-chains}) and good estimation of the apparent overdispersion by visually inspecting the fits in the longitudinal data (\figref{fig:overdispersion-longitudinal}). It is worth noting that 74732959 (G>A) in individual 260, while not being captured by this model, follows a fairly atypical trajectory: between the third and fourth timepoints there is a sharp decrease in \ac{vaf}, which can be an outcome of competition in the same individual caused by 74732959 (G>T) (this will be further discussed in the following chapter).

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/overdispersion-truq.pdf}
	\placecaption{The \textbf{known dilutions} dataset, containing technical sequencing replicates of prepared sequencing read solutions with known concentration. 90\% confidence intervals are represented assuming \ac{vaf} values follow a beta distribution.}
	\label{fig:truq-data}
\end{figure}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/overdispersion-data.pdf}
	\placecaption{The \textbf{technical replicates} dataset, which tracks mutations in 53 different genes across 4 individuals over 5 sampling phases. 90\% confidence intervals are represented assuming \ac{vaf} values follow a beta distribution.}
	\label{fig:replicate-data}
\end{figure}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/beta-chain.pdf}
	\placecaption{Trace plots for the technical overdispersion.}
	\label{fig:overdispersion-chains}
\end{figure}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/overdispersion-examples.pdf}
	\placecaption{Fitted values for the longitudinal modelling and comparison with real triplicate data from the \textbf{technical replicates} dataset. Dotted lines represent the trajectories defined by longitudinal sequencing and full lines represent the inferred trajectories. Vertical lines represent 90\% credible intervals.}
	\label{fig:overdispersion-longitudinal}
\end{figure}

\subsubsection{Estimation of clone fitness using longitudinal sequencing}

Using the simulations described in the methods above, I infer coefficients for the growth attributable to mutation $i$ ($b_{\mathrm{mut}_i}$) and for the growth attributable to the clone harbouring mutation $i$ in individual $j$ ($b_{\mathrm{clone}_{ij}}$) and do this jointly for all the simulation experiments. By doing so, it is possible to faithfully estimate the part of growth that is attributable specifically to the mutation and disentangle it from clone-specific effects. First, I verify that the \ac{mcmc} chains converge adequately by visually inspecting them (\figref{fig:mcmc-chains-sim}) and assessing the effective sample size and Gelman-Rubin statistic for all coefficients (\figref{fig:mcmc-ess}). Combining this visual inspection with the fact that values for \ac{ess} were routinely above 500 and that the values for the Gelman-Rubin statistic were close to the unit, I could confirm that the simulation has converged and that the chains can be further analysed. 

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/chains-genetic.pdf}
	\placecaption{\Ac{mcmc} chains for the mutation coefficients inferred using data simulated under a Wright-Fisher model.}
	\label{fig:mcmc-chains-sim}
\end{figure}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/rg-ess-sim.pdf}
	\placecaption{The Gelman-Rubin statistic (left) and effective sample size (right) for all modelled parameters inferred using data simulated under a Wright-Fisher model.}
	\label{fig:mcmc-ess}
\end{figure}

Knowing that the \ac{mcmc} chains converge adequately, I use them to show that using this model it is possible to calculate the fitness of drivers with a $R^2 = 91.5\%$ (\figref{fig:sim-vs-inf-driver}). Visual inspection of these trajectories while comparing them with the original trajectories provides further confirmation of the effectiveness of this model in recapitulating clonal growth (\figref{fig:trajectory-examples-sim}). Here, I use the average of all chains to estimate the expected value of each coefficient and calculate the 90\% \ac{hpdi} for all coefficients. In practical terms, the 90\% \ac{hpdi} is the shortest-spanning interval containing 90\% of the samples in these simulations and it is observable that the \ac{hpdi} for each coefficient contains --- or is close to containing --- the true value of the coefficient. I also measure the likelihood of the simulated data under the model and define an outlier as a value not contained within the 95\% confidence interval defined by the model and consider, from here on, all trajectories with outliers as invalid and do not proceed with their analysis. Taking these standards, 87\% of all trajectories are explained by the model. Assuming that each point had a 5\% chance of being an outlier and that each point belonging to the same trajectory is independent from the others, it is possible to calculate a theoretical probability that a trajectory contains an outlier for a different number of timepoints and compare these with the results from these simulations. As shown in \tableref{table:outlier-comparison}, this model explains a sizeable amount of the explained trajectories, considerably larger than what would be expected if each sample had a 5\% chance of being an outlier. Additionally, it is clear that this remains relatively constant for different true fitness intervals (\figref{fig:explained-traj-sim}).

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/simulated-vs-inferred-fitness.pdf}
	\placecaption{Inferred fitness effect vs. simulated fitness.}
	\label{fig:sim-vs-inf-driver}
\end{figure}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/example-trajectories-simulated.pdf}
	\placecaption{Examples of simulated and inferred trajectories for different true fitness levels (s).}
	\label{fig:trajectory-examples-sim}
\end{figure}

\begin{table}[!h]
	\centering
	\caption{Parameters for beta distributions fitted to the relative ages at onset.}
	\pgfplotstabletypeset[
	font=\footnotesize,
	string type,
	columns/tp/.style={
		column name=Number of timepoints,
		column type={C{.2\textwidth}}},
	columns/t/.style={
		column name=Expected fraction of trajectories with outliers,
		postproc cell content/.append code={\pgfkeysalso{@cell content/.add={}{\%}}},
		column type={C{.3\textwidth}}},
	columns/o/.style={
		column name=Observed fraction of trajectories with outliers,
		postproc cell content/.append code={\pgfkeysalso{@cell content/.add={}{\%}}},
		column type={C{.3\textwidth}}},
	every head row/.style={before row={\toprule},after row=\midrule},
	every last row/.style={after row={\toprule}},
	every odd row/.style={before row={\rowcolor[gray]{0.9}}}
	]\betaDistFit
\label{table:outlier-comparison}
\end{table}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/explained-trajectories.pdf}
	\placecaption{Explained trajectories for different true fitness ranges.}
	\label{fig:explained-traj-sim}
\end{figure}

\subsubsection{Estimation of the age at clone onset using longitudinal sequencing data}

To obtain an age, rather than a generation, at clone onset, I divide the generations by 13 under the assumption that there are 13 generations (cell divisions) per year. Using this value, I am able to compare the inferred and simulated age at onset to assess whether the method stated above to calculate the age at onset (\equationref{eq:t-onset}) allowing the recapitulation of the true age at onset. To generate each estimate for the age at onset, I calculate for each sample of $b_{\mathrm{mut}_i}$, $b_{\mathrm{clone}_{ij}}$ and $u_{ij}$ their respective $t_{\mathrm{onset},ij}$ and, for each clone (each having $1,000$ samples), I estimate the median and the 90\% confidence interval for the age at onset. I note here that a relatively low number of clones --- 10.3\% of all clones --- had a median age at onset that preceded birth. This is due to an underestimation of the true fitness of the clone, especially considering that competition or clones with smaller fitness advantages are present in these simulations. To address this problem and knowing that it is highly unlikely that clones appeared before birth (especially in the context of these simulations) I set a lower bound for the age at onset estimates that precede birth --- in other words, if $t_{\mathrm{onset},ij} < 0$ then I correct this to $t_{\mathrm{onset},ij}=0$. Taking these corrections into consideration, I now compare the true with the inferred age at onset for these simulated clones and show that it is possible to estimate the true age at onset using a relatively simple longitudinal sequencing protocol and statistical modelling ($R^2 = 65.8\%$, \figref{fig:age-at-onset-sim}). Additionally, while these estimates can have a relatively high spread, their across age intervals shows they appear to be a consistent estimator of the age at onset as represented with the boxplots in \figref{fig:age-at-onset-sim}. 

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/ages-at-onset.pdf}
	\placecaption{Inferred age at onset as a function of the simulated age at onset. Each boxplot represents the distribution of expected ages at onset for 10 year intervals.}
	\label{fig:age-at-onset-sim}
\end{figure}

An additional and relatively intuitive finding from this analysis is that the \ac{mae} for the age at onset estimates is considerably higher for smaller fitness advantages ($\mathrm{MAE}_{s \leq 0.01} = 12.49$, $\mathrm{MAE}_{s > 0.01} = 6.34$, \figref{fig:age-at-onset-vs-true-fitness-sim}) --- this conclusion highlights the fact that lower fitness advantages make clones more likely to be affected by competition or genetic drift, making the estimation of their ages at onset more biased due to greater relative errors in the inference of their fitness effect. 

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/fitness-vs-inferred-age-residuals.pdf}
	\placecaption{Residuals for the inferred age at onset as a function of simulated fitness.}
	\label{fig:age-at-onset-vs-true-fitness-sim}
\end{figure}

Having access to the original clonal trajectories also allows me to inspect the cases where age at onset estimation fails. As visible in \figref{fig:examples-bad-inference}, containing a few examples of trajectories where the absolute error for the age at onset was superior to 20 years. Specifically for the low fitness cases (such as the first panel in \figref{fig:examples-bad-inference}), this can also be due to the difficulty of estimating the age at onset for slow clones. In other cases, I can reasonably postulate that the large errors in the inference of the age at onset are mostly attributable to stochastic regimes deviating from the expected value or due to competition between different clones, leading to the underestimation of the growth rate. Regarding the longer than expected stochastic regime, it is worth restating that the expected value is $E[t_{\mathrm{stoch}}] = \frac{1}{sg}$. In my case, a clone with $s=0.001$ would have $E[t_{\mathrm{stoch}}] = 77$, whereas one with $s=0.01$ would have $E[t_{\mathrm{stoch}}] = 7.7$. For example, in \figref{fig:examples-bad-inference}, the top-left trajectory (s=0.002; mu=1.2e−07), $E[t_{\mathrm{stoch}} = 38.5$ but the observed $t_{\mathrm{stoch}}$ was much smaller than that ($3.1$). Conversely, for the top-right trajectory in \figref{fig:examples-bad-inference}, $E[t_{\mathrm{stoch}} = 19.2$ but the observed was much larger ($50.4$). Additionally, the effect of competition is that of causing clones to grow at a rate that is lower than their real growth rate due to an increase of the relative fitness of the population, i.e. other fit clones also reaching considerable sizes --- more concretely, the observable clone fitness will be the fitness of that clone minus the average population fitness. Importantly, while the effect of genetic drift will be considerably higher on smaller clones, that of competition can impact clones of any size.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/examples-bad-inference.pdf}
	\placecaption{Examples of trajectories for which the absolute error for the age at onset inference is at least 20 years.}
	\label{fig:examples-bad-inference}
\end{figure}

\paragraph{Effect of $N$ and $g$.} As made clear in \equationref{eq:t-onset}, there is some dependency on the assumption of values for $N$ and $g$. Taking this into consideration, I investigate the effect of wrong values for $N$ and $g$ on the age at onset estimates. This shows that there is a set of values for $N$ and $g$ for which the age at onset estimates are relatively similar in their \ac{mae} when compared with the simulated age at onset as shown in \figref{fig:age-at-onset-mae}. Using the correct values, one can be sure that the solution is, on average, within $\approx 9$ years of the real value. Nonetheless, it should also be noted that incorrect assumptions for $N$ and $g$ do not render incorrect values due to their dependence --- in the formula for $t_{\mathrm{onset},ij}$, a single term has both $N$ and $g$ ($\log(g/s_{ij}/N)$) and, as such, it is reasonable that feasible solutions for $g$ and $N$ occur, as long as the $N/g$ remains close to that of the true value --- in this case, $N/g = \frac{200,000}{13} \approx 15384$.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/age-at-onset-mae.pdf}
	\placecaption{Mean absolute error for the age at onset estimation for different population sizes and number of generations per year. The tile with a black outline represents the best (lowest) performing combination of parameters.}
	\label{fig:age-at-onset-mae}
\end{figure}

\paragraph{Estimating $N$ and $g$.} Finally, I was interested in understanding if there was a reasonable way of estimating a likely solution for the real values of $N$ and $g$ given the distribution of ages at onset for different clones. To do this, an additional set of clones were simulated under a Wright-Fisher model with two different combinations of $N$ and $g$: $N=100,000;g=5$ and $N=50,000;g=1$. I note here that these constitute slightly different values for $N/g$, particularly $20,000$ and $50,000$, respectively. 

The motivation for this investigation was relatively simple --- assuming identical mutation rates, one would expect clones to appear in a more or less uniform manner over life. Under the simulated experimental conditions here stated, however, this expectation may not hold for the detected clones --- it would not be possible to detect very small clones or clones that did not manage to fixate in the population. Indeed, as observable in \figref{fig:real-age-at-onset-distribution}, slower clones will appear mostly over the first half of life (otherwise their detection would be impossible) while faster clones are more likely to appear later in life. Nonetheless, the distribution of ages at onset for all clones appears to be generally characterized by a slow accumulation of clones after which a steady decay is observable.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/real-age-at-onset-distribution.pdf}
	\placecaption{Distribution of the simulated ages at onset for different fitness ranges.}
	\label{fig:real-age-at-onset-distribution}
\end{figure}

This suggests that an assumption of uniformity would not be the most adequate, making the choice of a more flexible distribution a necessity. For this reason, I chose the beta distribution, which requires some manipulation of the data. I transformed the ages at onset $a$ for each of the groups of simulations ($N=200,000;g=13$, $N=100,000;g=5$ and $N=50,000;g=1$) to have range $[0,1]$ as $a' = \frac{a-\min(a)}{\max(a)-\min(a)}$ (henceforth, I will be referring to this quantity as the relative age at onset). Having done so, I fitted three distinct Beta distributions to each group of simulations using moment matching (calculating the moments for the data and using them to estimate the parameters of the beta distribution), verifying relatively similar estimates for both shape parameters and their respective distributions (\tableref{table:beta-dist}; \figref{fig:beta-distance-fit}). The purpose of fitting these distributions is to have a more compact description of the empirical distribution and to better understand how much this deviates from the uniform ($\mathrm{Beta}(1,1) \sim \mathrm{Uniform}(0,1)$). Additionally, I fitted these distributions using a simple moment matching approach (i.e. the mean and variance are calculated from the real data and used to calculate the parameters of a Beta distribution using their closed form), which generally yields consistent estimators of the distribution it approximates.

\begin{table}[!h]
	\centering
	\caption{Parameters for beta distributions fitted to the relative ages at onset.}
	\pgfplotstabletypeset[
	font=\footnotesize,
	string type,
	columns/n/.style={
		column name=Population size,
		column type={C{.2\textwidth}}},
	columns/g/.style={
		column name=Generations/year,
		column type={C{.2\textwidth}}},
	columns/a/.style={
		column name=$\alpha$,
		column type={C{.05\textwidth}}},
	columns/b/.style={
		column name=$\beta$,
		column type={C{.05\textwidth}}},
	every head row/.style={before row={\toprule},after row=\midrule},
	every last row/.style={after row={\toprule}},
	every odd row/.style={before row={\rowcolor[gray]{0.9}}}
	]\betaDistFit
\label{table:beta-dist}
\end{table}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/beta-distance-fits.pdf}
	\placecaption{Empirical and theoretical densities from beta distribution fits to the relative ages at onset.}
	\label{fig:beta-distance-fit}
\end{figure}

For all three combinations of parameter values, I note that the expectation of the distribution shape may be that of a relatively flat and constant mutation rate as this remains constant through life. However, it is quite visible in \figref{fig:beta-distance-fit} that the number of mutations is closer to a left-shifted bell shape. The reason for this is two-fold --- to create a scenario that is closer to the one in Chapter 3, I remove mutations that achieve very high \ac{vaf} (>45\%). While this filter can affect any clone, it is more likely to affect those which appear earlier and that are more likely to reach larger clone sizes. On the other hand, the dip later in life is due to the fact that only sampled clones (i.e. clones reaching a \ac{vaf} of $\approx 0.5\%$) are being considered. Both these factors --- the filtering of large clones and the technical aspects of clonal detection --- make the distribution of clone onset take on a left-shifted bell-shaped curve. It is also worth noting that this left-shift is larger in smaller population sizes. The explanation for this is tied to the drift threshold --- for example, a clone growing at an annual fitness of 0.1 in a population where $N=50,000$ will reach the deterministic regime ($1/s$) when its size is $0.02\%$; however, that same clone in a population where $N=200,000$, will reach the deterministic regime when its size is $0.005\%$, implicating that clones appearing later in life will be more likely to grow to larger sizes. Given that I filter out large clones, this impacts smaller populations by shifting the distribution density towards early life. 

Having an adequate description of an empirical distribution, it is possible to assess whether the distribution of inferred relative ages at onset can be considered similar. To assess this, I calculate the Kolmogorov–Smirnov statistic between the fitted Beta distribution and the distribution of the inferred relative ages at clone onset. I present the results for this analysis in \figref{fig:heatmap-ks-d}, showing that the KS statistic is relatively small when the correct values are used. However, it should be noted that these results do not allow me to correctly assert the correct population size or number of generations per year. As such, I avoid using this sort of analysis to estimate the correct population size and number of generations per year, relying instead on the estimates offered in other works \cite{Lee-Six2018-lp,Mitchell2021-zl}.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/heatmap-ks-d.pdf}
	\placecaption{Heatmap representing the Kolmogorov-Smirnov (KS) statistic between the estimated theoretical distribution of relative ages at onset and the distribution of inferred relative ages at onset. The numbers on the heatmaps represent the ranking of each tile, with lower numbers and lighter colours representing higher similarity between distributions.}
	\label{fig:heatmap-ks-d}
\end{figure}

\section{Studying the haematopoietic evolution through single cell colonies}

\subsection{Simulations and tree construction}

Similarly to the previous section, I start by simulating a set of scenarios to assess the validity of the fitness estimation protocol I applied to real data. Here, I simulated a population of 200,000 \ac{hsc}, each having 7,500 passenger mutations and 50 drivers and selected 6 distinct combinations of fitness effect and mutation rate for the 50 possible driver mutations --- $[0.005,200 \times 10^{-9}; 0.01,50 \times 10^{-9}; 0.015,20 \times 10^{-9}; 0.02,15 \times 10^{-9}; 0.025,8 \times 10^{-9}; 0.03,5 \times 10^{-9}]$ --- and each simulation was repeated 20 times for 800 generations, yielding a total of 120 simulations. For each of these simulations, a phylogenetic tree was built by following a relatively simple protocol:

\begin{enumerate}
    \item For each simulation I take a representative sample of 100 clones from the last timepoint ($g = 800$). To do so, the probability of each clone being sampled is given by its relative frequency. I used 100 clones here to approximate the scenario in Chapter 3 where 96 colonies are used to construct each phylogenetic tree;
    \item Using the complete set of mutations describing each clone, I build trees using a neighbour-joininig algorithm \cite{Saitou1987-uq}. This relatively simple algorithm iteratively builds the tree by joining the two most similar tips/nodes. 
\end{enumerate}

For each tree, I detect clades by defining them as a group of tips and branches whose \ac{mrca} has the earliest instance of the driver they share. I also extracted the trajectories for the drivers out of each simulation considering the possibility that two different clones could have the same driver --- for instance, if a clone $a$ acquired mutation $i$ at time $t_a$ and clone $b$ acquired mutation $i$ at time $t_a$ these are considered two distinct drivers rather than a single one --- this enables the comparison between driver trajectories and trajectories inferred from each clade. In practical terms, I consider that a clade and a driver trajectory correspond to the same clone if the mutations characterizing them have the smallest Hamming distance out of all clade-driver trajectory combinations. 

\subsection{Phylodynamic inference of clonal trajectories from phylogenetic trees}

To determine the clonal trajectory of each expansion using the information contained in its corresponding clade --- particularly, the distribution of times between coalescent events --- I use \ac{bnpr}, a method implemented in the \texttt{phylodyn} \cite{Lan2015-sw,Karcher2017-kt} package for \texttt{R}. \ac{bnpr} estimates effective population sizes on a grid of sampled points at resolution $\tau$ between the tips and the first observable coalescence of each tree (or clade). In effect, this approach seeks to estimate the posterior $\mathrm{P}[f,\tau|g] \propto \mathrm{P}[g|f]\mathrm{P}[f|\tau]\mathrm{P}[\tau]$, where $f$ is the piecewise linear estimate in the \ac{eps} estimator $N(t)$ for a grid of points along the tree between $x_d$ and $x_D=t_2$ such that $N(t) = \sum^{D-1}_{d-1}\exp(f)1_{(x_d,x_{d+1}]}$ and $g$ is the inferred genealogy (our phylogenetic tree or clade). $f$ is parameterized as a Gaussian process and describes the population trajectory between coalescent events. In practical terms, the estimation process for $f$ is approximated using the integrated nested Laplacian approximation, with more concrete details in \cite{Lan2015-sw} (theoretical foundation) and \cite{Karcher2017-kt} (practical application and implementation). In other words, \ac{bnpr} estimates the \ac{eps} trajectory from the tips of a tree to its first coalescence, which can then be used to determine the clonal dynamics of specific expansions. I chose this method because it had been already used previously in the landmark paper on haematopoietic dynamics by \etal{Lee-Six}, is reasonably fast and, as I will be showing further ahead, provides good estimates for the \ac{eps} and allows the calculation of a consistent estimate of the true fitness of clones. In \figref{fig:models-bnpr-example} I present an example of different clades and their phylodynamic trajectories fitted using \ac{bnpr} (blue full lines and bands), where it is evident that the existence of coalescences is what allows \ac{bnpr} to infer relatively narrow intervals for \ac{eps} estimates.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/examples-bnpr-fit.pdf}
	\placecaption{Illustrative representation of clades and their respective \ac{bnpr} trajectories, along with the three parametric fits used to describe them.}
	\label{fig:models-bnpr-example}
\end{figure}

\subsection{Phylodynamic fitness}

Calculating the \ac{bnpr} \ac{eps} estimates for all the trees in the clade assumes no fixed fitness effect. However, it is expected that each mutation confers a specific fitness effect, which should be determinant in these trajectories. To determine the best way to infer fitness from \ac{bnpr} trajectories I fit three different models to all \ac{eps} trajectories $N_e(t)$, namely:

\begin{enumerate}
    \item A log-linear fit which assumes that clonal growth is exponential: $\log(N_e'(t)) = t \times b + a$, where $b$ and $a$ are determined using least squares (orange dashed line in \figref{fig:models-bnpr-example});
    \item A scaled and shifted sigmoid fit which assumes exponential but saturating growth: $N_e'(t) = \frac{b_{\mathrm{scale}}}{1+e^{\frac{b_{\mathrm{mid}}-t}{b}}}$, where $b_{\mathrm{scale}}$, $b_{\mathrm{mid}}$ and $b$ are determined using a non-linear optimization routine (\texttt{nl2sol}) from the Port library \cite{noauthor_undated-gs}. I initialize variables in such a way that guarantees convergence: $b_{\mathrm{scale}}=\max(N_e(t))$, $b_{\mathrm{mid}}=\max(t)$ and $b=10$ (green dashed line in \figref{fig:models-bnpr-example});
    \item A biphasic log-linear fit which assumes there are two distinct phases of growth separated by a breakpoint $\mathrm{bp}$: $\log(N_e(t)) = t \times b_1 + a, \mathrm{if}\ t < \mathrm{bp}$ and $\log(N_e'(t)) = t \times b_2 + a, \mathrm{if}\ t \geq \mathrm{bp}$, where $b_1$, $b_2$, $a$ and $\mathrm{bp}$ are determined using \ac{l-bfgs-b}, a quasi-Newthonian method which uses a limited amount of computer memory. Similarly to 2., I initialize variables to facilitate convergence: $b_1=b$, $b_2=b$, $a=a$, $\mathrm{bp}=\mathrm{mean}(t)$, where $a$ and $b$ are the estimates from 1. I also constrain estimates of $\mathrm{bp}$ to be between $\min(t) + 0.25 \times \mathrm{range}(t)$ and $\max(t) + 0.75 \times \mathrm{range}(t)$ (red dashed line in \figref{fig:models-bnpr-example}).
\end{enumerate}

For all three estimates described above, I weigh data according to its uncertainty from the \ac{eps} estimates --- \ac{bnpr} produces estimates for the 95\% credible interval, which I use to derive an approximate estimate of the logarithm of the variance as $\log(V(t))' = (\log(N_e(t)_{97.5\%}) - \log(N_e(t)_{2.5\%}))^2/16$ and I use its inverse. Finally, I compare all three types of estimate by assessing how closely they are able to recapitulate the simulated fitness. To do so, I calculate their coefficient of determination and root mean squared error. I also visually assess how similar these trajectories are to the true driver trajectories as reconstructed from simulations.

To confirm the impact of this particular method of phylodynamic estimation --- \ac{bnpr} --- on the \ac{eps} trajectories, I use two additional methods for phylodynamic estimation as implement in the \texttt{ape} package for \texttt{R} \cite{Paradis2019-na}: \texttt{skyline} (an implementation of the generalized skyline plot, a non-parametric method for \ac{eps} estimation that assumes that the \ac{eps} for a given interval is proportional to the accumulated waiting time between coalescent events and the number of lineages while inversely proportional to previous coalescent events) and \texttt{mcmc.popsize} (a method with similar assumptions to those of \texttt{skyline} but which uses reversible jump \ac{mcmc} to derive a smoother version of the \ac{eps} estimate) \cite{Opgen-Rhein2005-pi} (\figref{fig:examples-phylo-traj}). \texttt{mcmc.popsize} fits a set of splines to obtain a continuous representation of the \ac{eps} and the use of reversible jump \ac{mcmc} enables this method to also infer the correct number of necessary splines. I use both additional methods to verify that estimates from \ac{bnpr} are consistent with those from other methods while enabling a more accurate recapitulation of the true fitness effect.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/examples-phylo-traj.pdf}
	\placecaption{Three phylodynamic trajectories inferred using three different methods --- \ac{bnpr}, \texttt{mcmc.popsize} and \texttt{skyline}.}
	\label{fig:examples-phylo-traj}
\end{figure}

\subsection{Simulation results}

\subsubsection{Early growth is a consistent estimator of clone fitness using phylodynamic estimation}

Using a simulation scenario similar to the one described for the previous section of the results, I inferred phylogenetic trees for 120 Wright-Fisher simulations after 800 generations. I avoided using the same simulations as before for 3 reasons: firstly, I wanted to have access to a set of simulations that could be used to generate a variety of phylogenetic trees that included trees similar to those in Chapter 3, a process that was achieved by iteratively trying different parameter combinations; secondly, phylodynamic estimation is time-consuming and having a smaller set of simulated \ac{hsc} populations was preferable.

The structure of each of these trees alone highlights the expected diversity when clones with distinct fitness effects are present (\figref{fig:trees-simulated-examples}) --- while smaller fitness advantages do not lead to particularly evident expansions in the tree (a large branch preceding a clade with more than 5 tips), larger fitness advantages often have the capacity to sweep the entire clonal landscape and take over the whole \ac{hsc} population. Sections of the tree which have no driver and are not contained in an expansion have a comb-like structure, characterized by tips with long branches.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/tree-examples.pdf}
	\placecaption{Examples of the trees generated for different, increasing fitness effects.}
	\label{fig:trees-simulated-examples}
\end{figure}

I then defined clones in the trees as a clade described by a unique combination of drivers and correspond them to their respective clones in the simulation. I isolated the clades and for each of them used \ac{bnpr}, a phylodynamic estimation method, to determine the lifelong trajectories of each of these clones. A first assessment, comparing the original trajectories in the simulations with those inferred using phylodynamic estimation from the trees shows good concordance (\figref{fig:wf-vs-bnpr-traj}). This highlights the usefulness of single-cell colony experiments in determining the lifelong trajectory of clones. This comparison also shows that phylodynamic estimates are sensible to the carrying capacity of the \ac{hsc} population, saturating as the clone slows down. \figref{fig:wf-vs-bnpr-traj} also shows that some \ac{bnpr} trajectories show considerably more variance as their trajectory approaches the time of sampling. This is largely due these clades being relatively small --- indeed, when clades are smaller, it is less likely for them to display a sufficient number of coalescent events (the structure of which is used to infer the \ac{eps}) that would allow smaller confidence intervals. As such, when analyzing these clades I use the average log variance $V_{\mathrm{mean}} = \frac{1}{n}\sum_{i=1}^{n}{\log(V(N_e(t_i))})$ to define low variance clades ($V_{mean} < 5$; \figref{fig:low-variance-examples}) and high variance clades ($V_{\mathrm{mean}} \geq 5$; \figref{fig:high-variance-examples}). It is visible from \figref{fig:low-variance-examples} and \figref{fig:high-variance-examples} that the main difference in the data (clades) underlying low and high variance trajectories is the density of coalescent events --- in high variance trajectories, sizeable portions of the clade have no coalescent events. Finally, I note here that high variance clades have a median of 6 tips (range: [5,13]), while low variance clades have a median of 21 tips (range: [5-99]). 

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/wf-vs-bnpr-traj.pdf}
	\placecaption{Comparison of Wright-Fisher trajectories with their \ac{bnpr}-estimated trajectories.}
	\label{fig:wf-vs-bnpr-traj}
\end{figure}

\begin{figure}[!hb]
	\placefigure{gfx/dynamics-1/low-variance-examples.pdf}
	\placecaption{Examples of clades (top) and their respective trajectories estimated with \ac{bnpr} (bottom) for low variance trajectories.}
	\label{fig:low-variance-examples}
\end{figure}

\begin{figure}[!hb]
	\placefigure{gfx/dynamics-1/high-variance-examples.pdf}
	\placecaption{Examples of clades (top) and their respective trajectories estimated with \ac{bnpr} (bottom) for high variance trajectories.}
	\label{fig:high-variance-examples}
\end{figure}

Having observed the concordance between the Wright-Fisher trajectories of clones and their phylodynamic estimates, I then proceeded to quantify the growth from the phylodynamic estimates. It is worth noting that \ac{bnpr}, while providing a trustworthy recapitulation of the lifelong clonal trajectory, is a non-parametric method and, as such, does not provide \textit{per se} an estimate of the fitness advantage for any given clone. I fitted three different models to the phylodynamic trajectories --- a log-linear model, a sigmoid model and a biphasic log-linear model as represented in \figref{fig:models-bnpr-example}. Each has particularly different assumptions --- growth is constant and exponential, growth saturates as the clone size approximates the carrying capacity and growth is exponential and can be divided into two distinct phases, respectively. The inspection of \ac{bnpr} trajectories on their own already reveals the fact that it is the initial phase of \ac{eps} that is the most different between different fitness advantages (\figref{fig:all-simulated-bnpr}).

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/all-simulated-bnpr.pdf}
	\placecaption{All inferred \ac{bnpr} trajectories for clones simulated under a Wright-Fisher model (black). In red I show a log-linear fit to the first 400 generations of the data.}
	\label{fig:all-simulated-bnpr}
\end{figure}

To assess which fit is the best suited to capture the true clone fitness, I measure the coefficient of correlation and the root mean squared error between the fit coefficient and the true clone fitness (\figref{fig:benchmark-bnpr-fits}). This shows that early growth --- the first phase of growth as capture by the biphasic model --- is a consistent estimator of the fitness. On the other hand, assuming a single growth coefficient appears to dramatically underestimate the growth effect of fitter clones, whereas sigmoid growth overestimates lesser fit clones. It is important to note that the growth observed at a later phase is, in general, quite close to 0 as the clone approaches carrying capacity. Additionally, the clustering of trajectories as having low or high variance (red and blue in \ref{fig:benchmark-bnpr-fits}, respectively) allows a higher degree of confidence in my inference --- indeed, whereas the root mean squared error between the early growth and the true fitness advantage is 0.0098 for high variance trajectories, it is lower by 0.002 (0.008) for low variance trajectories. 

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/benchmark-bnpr-fits.pdf}
	\placecaption{Benchmarking different fits to the phylodynamic trajectories inferred using \ac{bnpr}.}
	\label{fig:benchmark-bnpr-fits}
\end{figure}

Finally, the intervals for the \ac{bnpr} trajectory inference were wider as fewer coalescent events were available (as visible in \figref{fig:models-bnpr-example} and mentioned earlier). This is particularly prevalent as the time of sampling approaches. This could be of potential concern as it could bias my estimates for growth. I evaluated the consistency of these estimates, i.e. how different would the inferred coefficients be if the trees were trimmed by 50 or 100 generations. As shown in \figref{fig:}, these trajectories are relatively similar to one another. Quantifying their consistency --- the mean absolute error between \ac{bnpr} trajectories obtained from the original tree and from trimmed trees --- leads to the results presented in \tableref{table:trimmed-fits}. The error is relatively small --- on average, between 0.002 and 0.003 for the estimation of the early growth and between 0.002 and 0.005 for the late growth --- showing that these estimates are relatively consistent. 

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/examples-bnpr-fit-trimmed.pdf}
	\placecaption{Examples of \ac{bnpr} trajectories obtained from the original trees, trees trimmed by 50 generations (5 years) and trees trimmed by 100 generations (10 years).}
	\label{fig:examples-bnpr-fit-trimmed}
\end{figure}

\begin{table}[!ht]
	\centering
	\caption{Mean absolute error between the dynamic parameters inferred from the original trees and trees trimmed by 5 or 10 years.}
	\pgfplotstabletypeset[
	font=\footnotesize,
	string type,
	columns/g/.style={
		column name=Growth,
		column type={C{.2\textwidth}}},
	columns/t5/.style={
		column name=Original-trimmed by 5 years,
		column type={C{.35\textwidth}}},
	columns/t10/.style={
		column name=Original-trimmed by 10 years,
		column type={C{.35\textwidth}}},
	every head row/.style={before row={\toprule},after row=\midrule},
	every last row/.style={after row={\toprule}},
	every odd row/.style={before row={\rowcolor[gray]{0.9}}}
	]\trimmedFits
\label{table:trimmed-fits}
\end{table}

\subsection{BNPR outperforms other methods for phylodynamic inference}

The last section of this chapter is dedicated to the evaluation of \ac{bnpr} against other \texttt{mcmc.popsize} and \texttt{skyline}, two other phylodynamic estimation tools with similar assumptions present in the \texttt{ape} package in R \cite{Paradis2019-na}. Visually, all three methods appear to capture the same general trends, with \texttt{mcmc.popsize} showing little success in estimating the earlier phase of growth for a few clades as represented in \figref{fig:compare-phylo-traj}. While it is hard to determine a cause for this poor estimation of earlier growth, it is possible that the fact that \texttt{mcmc.popsize} initializes \ac{eps} as a heuristic constant value \cite{Opgen-Rhein2005-pi}, coupled with a relatively low number of very early coalescent events, may cause this.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/compare-phylo-traj.pdf}
	\placecaption{Comparison of phylodynamic trajectories obtained using \ac{bnpr} and \texttt{mcmc.popsize} and \texttt{skyline}.}
	\label{fig:compare-phylo-traj}
\end{figure}

I then used two of the fits stated above --- log-linear and biphasic --- to assess which method was capable of best capturing the true fitness advantage of clones, showing these results in \figref{fig:compare-phylo-traj-heatmap}. Once again, it is evident that early growth is much more representative of the true fitness advantage when compared with lifelong growth. Finally, while \texttt{skyline} performs reasonably well, it is \ac{bnpr} that shows the best performance at estimating the true fitness advantage of the clone. The weaker performance of \texttt{skyline} when compared with \ac{bnpr} is possibly due to the way that \texttt{skyline} interpolates between point estimates --- while \ac{bnpr} uses a Gaussian process as the prior for the \ac{eps} variation between coalescences, \texttt{skyline} assumes that population is relatively constant between each point estimate, leading to more abrupt changes in \ac{eps}. Taking this into consideration, I will be using \ac{bnpr} over the following chapter to estimate the population trajectories of different expansions in phylogenetic tree.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-1/compare-phylo-traj-heatmap.pdf}
	\placecaption{Benchmarking of the fitness advantage estimates obtained using \ac{bnpr} and \texttt{mcmc.popsize} and \texttt{skyline}.}
	\label{fig:compare-phylo-traj-heatmap}
\end{figure}

\section{Summary}

In this chapter, I show that it is possible to recapitulate two important aspects of clonal dynamics --- age at onset and fitness advantage --- using longitudinal targeted sequencing and phylogenetic trees combined with phylodynamic modelling. Additionally, I provide evidence that phylodynamic modelling using \ac{bnpr} recapitulates the lifelong behaviour of clones.

Using a simple hierarchical Bayesian model, I show that it is possible to accurately infer the growth rate associated with specific driver mutations in a simulated longitudinal sequencing setting ($R^2 = 91.5\%$ between the simulated and inferred coefficients). Additionally, using assumptions derived from the Wright-Fisher model, together with accurate growth coefficient inference, it is possible to estimate the age at clonal onset ($R^2 = 65.8\%$ between the simulated and inferred ages at clonal onset). While in many cases the age at clonal onset inferences were correct, lifelong changes in growth rate --- motivated by the stochastic nature of genetic drift or competition --- play a key role in clonal dynamics that longitudinal studies during old age are unable to capture.

Using simulation experiments that replicate the process of deriving and sequencing single colonies, I show that these lifelong changes can be captured using phylodynamic estimation, particularly through the use of \ac{bnpr}. Additionally, I show how growth during earlier clone expansion provides a better representation of the true fitness of the clone, with competition playing a determining role in these alterations.

In these simulations, determined by the Wright-Fisher model, competition causes the deceleration observed in the simulated trajectories of clones and their respective phylodynamic inferences (\figref{fig:wf-vs-bnpr-traj}) --- by increasing the average population fitness $\bar{s}$, the onset of other fit clones leads to a generalized inability of clones to grow at their true fitness rate $s$, growing instead at a relatively slower growth rate $s' = s - \bar{s}$. Conversely, when clones are smaller their growth is mostly determined by genetic drift and when clones appear earlier in life, when fewer fit clones are present in the \ac{hsc} population, they are able to grow at a rate that is more representative of their true fitness rate. These aspects of competition in the \ac{hsc} population explain the fact that early growth better represents $s$. However, it is worth noting that other factors --- for which evidence is becoming increasingly available and which will be discussed in the Discussion of Chapter 3 --- can play a role in deceleration. 

\section{Code availability}

The code for this analysis and the associated notebooks are available in \url{https://gerstung-lab.github.io/ch-dynamics/}, more specifically in "Estimating the overdispersion", "Validation of hierarchical Bayes model", "Validating the estimation of growth rates and evidence for saturation from phylogenetic trees" and "Comparing different methods for the estimation of population size trajectory".