\chapter{The longitudinal dynamics and natural history of clonal haematopoiesis}

\section{Introduction}

\ac{ch} has been widely studied regarding its likely associations with specific blood indices, smoking history, chemotherapy, inflammation and age \cite{Dawoud2020-af,Coombs2017-ph,McKerrell2015-rl,Cai2018-yi,Bolton2020-ct}, and has been shown to be associated with an increased risk of cardiovascular disease and haematological cancer \cite{Jaiswal2014-rl,Genovese2014-eu, Young2016-du, Xie2014-np, Desai2018-pj, Midic2020-zh, Zink2017-zi, Acuna-Hidalgo2017-ng}. However, the study of the dynamics and evolution of \ac{ch} is still in its infancy --- indeed, while it is now known that fitness advantages are gene- and site-dependent \cite{Watson2020-pz,Robertson2021-sw}, previous work on this topic have only been able to clarify either the average behaviour of clones or the clonal dynamics at an older age. While both sound relatively complimentary, apparent inconsistencies have arisen. Indeed, while \cite{Watson2020-pz} used the distribution of clone sizes to infer that \ac{dnmt3a} mutations can lead to some of the fastest growing clones, the work in \cite{Robertson2021-sw}, which uses longitudinal targeted sequencing, reaches different conclusions: \ac{dnmt3a} actually leads to some of the slowest growing clones in old age, a period during which mutations in splicing genes (particularly in \ac{u2af1} and \ac{sf3b1}). As such, there seems to be an apparent paradox --- while the distribution of clone sizes would imply that \ac{dnmt3a} clones expand quickly over life, these fast dynamics are not observable during old age. 

The work here presented contributes to this body of knowledge and, to some extent, resolves the aforementioned contradiction. Here, I have analysed data from longitudinal targeted sequencing for over 300 older individuals and confirm several of the findings presented by the previous by Robertson \textit{et al.} using longitudinal sequencing \cite{Robertson2021-sw}. Additionally, we are able to disentangle clone-specific (or "unknown-cause") effects from those attributable to specific genes and sites. This revealed that mutations in splicing genes, particularly in \ac{u2af1} and \ac{srsf2}-P95H, confer the greatest fitness advantage in older age, while mutations in \ac{dnmt3a} conferred a relatively small fitness advantage, similarly to \cite{Robertson2021-sw}. To complement this evolutionary snapshot of clonal dynamics during older age, I also used phylogenetic data from single-cell colonies derived for three of the individuals for which we had longitudinal targeted sequencing data and from four additional older individuals from a publication by Mitchell \textit{et al.} \cite{Mitchell2021-zl} and infer the phylodynamic trajectories for all expansions, with and without known drivers. Together with estimates from the longitudinal targeted sequencing data, this helped us clarify the apparent contradiction between \cite{Robertson2021-sw} and \cite{Watson2020-pz} --- most clones decelerate through life, an effect that is likely to be motivated by competition driven by an oligoclonal \ac{hsc} population and affects lesser fit clones the most. Additionally, I use the fitness estimates from longitudinal targeted sequencing to establish associations between both selection in \ac{aml} and \ac{mds} and risk of \ac{aml} onset as calculated in \cite{Abelson2018-wh}.

\subsection{Contribution and disclaimer}

The work presented in this chapter is also available as a preprint in \cite{Fabre2021-uw} and some parts may coincide. My contribution to this work, consisted on the analysis of longitudinal data and generation of all figures associated with it, as well as all the analyses associated with phylodynamic estimation of growth coefficients and phylogenetic estimation of ages at onset.

\section{Methodology}

\subsection{Cohort composition}

I worked exclusively with variant calls and, as such, will not go into detail about the variant calling process; nonetheless, for reference, these are available in \cite{Fabre2021-uw}. Regarding the cohort here studied, the data I used consisted of targeted sequencing data for 384 individuals. Notably, 80 (20.8\%) of these individuals had no mutations on any of the genes under study and were disregarded. For each individual, blood was collected at between 2 and 5 timepoints (Table \ref{table:age-dist}) across a median period of 12.9 years (3.21-16.1) and 5 phases (Phase 1-5), with 96.4\% of all individuals having more than two timepoints. Each blood sample was sequenced for a panel of 52 genes at a median coverage of 1350X. Only 17 of these genes --- as represented in \ref{table:genes-sites-modelling} --- were used after assessing which ones were under positive selection \cite{Fabre2021-uw,Martincorena2017-ii}. For 13 individuals, blood was collected at an additional timepoint --- Phase 6 --- for validation and for three of these individuals (SardIDs 2259, 3877 and 500) single-cell colonies were derived from the sixth timepoint and sequenced (85 colonies for 2259, 93 colonies for 3877 and 92 colonies for 500). This latter process was used to build single-cell phylogenies, a process that will be detailed further ahead. A schematic representation of the design of this study is made available in \figref{fig:study-design-ch}. The specific selection of these individuals was informed in a particular way --- mutations in the spliceosome, due to their age-dependency, are of high interest and motivated the derivation and sequencing of 96 single-cell colonies for three individuals with large (>5\% \ac{vaf}) spliceosome mutations. Particularly, the mutations of interest were \ac{sf3b1}-K666N, \ac{u2af1}-Q157R and \ac{srsf2}-P95H.

\begin{table}[!ht]
\centering
\caption{Distribution of individuals with a given number of timepoints.}
\pgfplotstabletypeset[
font=\footnotesize,
string type,
columns/ntp/.style={
    column name=Number of timepoints,
    column type={C{.2\textwidth}}},
columns/n/.style={
    column name=Number of individuals,
    column type={C{.2\textwidth}}},
columns/p/.style={
    column name=Proportion,
    postproc cell content/.append code={\pgfkeysalso{@cell content/.add={}{\%}}},
    column type={C{.2\textwidth}}},
columns/cp/.style={
    column name=Cumulative proportion,
    postproc cell content/.append code={\pgfkeysalso{@cell content/.add={}{\%}}},
    column type={C{.3\textwidth}}},
every head row/.style={before row={\toprule},after row=\midrule},
every last row/.style={after row={\toprule}},
every odd row/.style={before row={\rowcolor[gray]{0.9}}}
]\ageDist
\label{table:age-dist}
\end{table}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/study-design.pdf}
	\placecaption{Schematic representation of the study design for this work --- 384 individuals were followed over a median period of 13 years, during which 2-5 blood samples were collected and sequenced for a panel of \ac{ch} genes. This data was used to infer the annual growth and age at onset of 17 genes and 39 sites. For three of these individuals, single cell colonies were derived and whole-genome sequenced. These data were then used to build phylogenetic trees that recapitulate the haematopoietic process and allow the inspection of lifelong clonal behaviour.}
	\label{fig:study-design-ch}
\end{figure}

\subsection{Modelling clonal trajectories from longitudinal targeted sequencing data}

I assume that the \ac{vaf} at each timepoint is a proxy for clone size in the bone marrow and assume that each mutation is monoclonal. Both these assumptions are supported by previous studies \cite{Miles2020-fz,Lee2020-yp,Hwang2018-tp} and allow us to model each clonal trajectory --- the collection of \ac{vaf} values for a single driver mutation in a single individual --- as a time-dependent scaled sigmoid with an asymptotic value at 0.5. The reason why 0.5, rather than 1, is used is to accommodate the fact that most mutations are likely to be haploid. While phasing these mutations would be the ideal case scenario this is not possible using this targeted sequencing panel.

As mentioned, each clonal trajectory is characterized by a time-dependent sigmoid curve. To parametrize it, I have, for mutation $j$ in individual $i$, four distinct parameters: $b_{gene_j}$ characterizing the fitness advantage conferred by gene identity, $b_{site_j}$ characterizing the fitness advantage conferred by site identity for missense mutations in sites occurring in two or more individuals, $b_{unknown_{ij}}$ characterizing the fraction of growth that depends on clone-specific factors and $u_{ij}$ which is an offset for the sigmoid curve to account for clones appearing at different times. As such, the expected proportion of mutant counts on site $i$ in individual $j$ at time $t$ is given by the following model: $p_{ij}(t)=sigmoid((b_{gene_j}+b_{site_j}+b_{unknown_{ij}})*t + u_{ij})*0.5$. In our case and to speed up convergence, $t$ is calculated as the age of each individual after subtracting the youngest age in the cohort. With effect, the hierarchical model used here is quite similar to the one used in the previous chapter, with the main exception being the decomposition of $b_{mut_j}$ in the previous chapter into $b_{gene_j}$ and $b_{site_j}$ when a site is present in two or more individuals or the replacement of $b_{mut_j}$ with $b_{gene_j}$ when the mutation is present in a single individual.

Using this model I fit each parameter through a \ac{mcmc} approach. I assume as priors for $b_{gene_j}$ and $b_{site_j}$ a normal distribution with mean 0 and standard deviation 0.1, for $b_{unknown_{ij}}$ a normal distribution with mean 0 and standard deviation 0.05 and for $u_{ij}$ a uniform distribution between -50 and 0. The prior for $\beta$ follows a truncated normal distribution whose parameters were inferred as described in the previous chapter. For the sampling, I use \ac{hmc} as implemented in greta \cite{Golding2018-zp} with a number of leapfrog steps between 150 and 300. I draw 5000 samples and discard the initial 2500 samples, thus estimating the mean, median, credible intervals and \ac{hpdi} of each parameter using the last 2500 samples. The quality control step described in the previous section --- calculating the likelihood of each sample under our model and discarding all trajectories with at least one point whose likelihood has a tail probability of over $97.5\%$. I calculate the age at onset as described in the previous chapter --- $t_{onset,ij} = \frac{log(g/s_{ij}/N)-1-s_{ij}}{s_{ij}}$.

\subsubsection{Preventing identifiability issues and reducing uninformed estimates.}

To address possible identifiability issues in this model, when a gene has a single mutation (\ac{jak2} with V617F and \ac{idh2} with R140Q), the effect is considered to occur at the site level. To avoid estimating the dynamics of a site from a single individual, I only model sites explicitly when two or more individuals have a missense mutation on that site --- I refer to these sites as “recurrent sites”. For all other sites where a single mutation is observed across the entire cohort their differential dynamics will be contained in the unknown cause effect. Ultimately, I consider a total of 17 genes and 39 recurrent sites (Table \ref{table:genes-sites-modelling}).

\begin{table}[!ht]
\centering
\caption{Summary of different effects in the model. “X” indicates the utilization of that specific effect (Truncating and Missense) for each gene and in Sites I detail the sites that were explicitly modelled.}
\pgfplotstabletypeset[
font=\footnotesize,
string type,
columns/g/.style={
    column name=Gene,
    postproc cell content/.style={@cell content=\textit{##1}},
    column type={C{.1\textwidth}}},
columns/t/.style={
    column name=Truncating effect,
    column type={C{.20\textwidth}}},
columns/nt/.style={
    column name=Missense effect,
    column type={C{.20\textwidth}}},
columns/sites/.style={
    column name=Explicitly modelled sites (absolute prevalence),
    column type={C{.4\textwidth}}},
every head row/.style={before row={\toprule},after row=\midrule},
every last row/.style={after row={\toprule}},
every odd row/.style={before row={\rowcolor[gray]{0.9}}}
]\genesSites
\label{table:genes-sites-modelling}
\end{table}

\subsubsection{Assessing the predictive performance of clonal growth predictions}
Using an additional time-point (phase 6) available for 11 individuals with mutations in \ac{cbl} (c.2434+1G>A), \ac{dnmt3a} (P385fs, R882H, W330X), \ac{gnb1} (K57E), \ac{jak2} (V617F), \ac{ppm1d} (Q524X), \ac{sf3b1} (K666N, K700E, R625L), \ac{srsf2} (P95H, P95L), \ac{tet2} (Q1542X) and \ac{u2af1} (Q157P, Q157R). Using the model described in this section of the Methods and conditioning on the previous timepoints, we predict the additional time-point and assess the predictive performance through the \ac{mae} to the true \ac{vaf} value.

\subsection{Analysing the impact of smoking history, sex, age and smoking experience on the clonal growth rate}

The association between unknown-cause growth and age was calculated using the Pearson correlation considering all genes, both together and separately while controlling for multiple testing. For the association between unknown-cause growth and sex and smoking history, a multivariate regression was used, with unknown-cause growth as the dependent variable and sex and previous smoking experience as the covariates while controlling for age.

\subsection{Phylogenetic and phylodynamic estimation of clonal trajectories from single-cell derived phylogenetic trees}

Single-cell colonies were derived and sequenced for three individuals and their phylogenetic trees inferred as detailed in \cite{Fabre2021-uw}. Each tree was made ultrametric by iteratively extending branches until the distance between any two tips is no bigger than the distance between either tip and any third tip (also known as the three point condition) and scaled in such a way that they span the age of the individual, with an 0.75 years accounting for gestational development. In these trees, we define and detect expansions as either clades with one known driver mutation at their \ac{mrca} or as clades whose \ac{mrca} branch length is at least 10\% of the depth of the tree. This allows us to detect expansions with no known driver. The determination of clonal trajectories from each expansion, as well as the quantification of growth from these trajectories, is described in the previous chapter.

The motivation to scale the ultrametric trees according to the age of the individual comes from recent work by \etal{Abascal} showing that mutation rates are constant through life \cite{Abascal_2021_gjvqfm} and, as such, mutations should accumulate linearly with age from birth. Present evidence also shows that a relatively small but significant number of mutations happen between conception and birth, with studies showing this to be close to 55 mutations \cite{SpencerChapman_2021_gjz4x4,Mitchell2021-zl}; as such, we consider the first 55 mutations of each tree to have happened before birth, scaling the rest of the tree linearly with time.

\subsection{Detecting deceleration in single-cell phylogenies and longitudinal data}

We infer the presence of deceleration in both single-cell phylogenies and longitudinal data. To do this, we use two distinct methods: calculating the ratio between the expected and observed \ac{vaf} and calculating deceleration using growth rates. 

For the first method --- calculating the ratio between expected and observed \ac{vaf} --- we use the value for the early growth from the biphasic log-linear fit described in the previous chapter and extrapolate the \ac{eps} to the age at sampling. By doing so we get the expected clone fraction if growth had not changed during the \ac{eps} trajectory. We also calculated the observed clone fraction as the fraction of tips in the clade. To get the expected clone fraction from \ac{eps} we divide \ac{eps} by the inferred population size in \etal{Lee-Six} (200,000 HSC) \cite{Lee-Six2018-lp}. We then calculate the ratio between the expected and observed clone size --- if this ratio is close to 1 this implies little to no changes in dynamics, whereas a ratio above 1 implies deceleration and a ratio below 1 implies acceleration.

For the second method --- calculating deceleration using growth rates --- we define two distinct quantities for both single-cell phylogenies/longitudinal data --- expected/observed growth, corresponding to the growth rate of each clone during observation at old age, and early/minimal historical growth, corresponding to the growth rate of each clone at an earlier stage of clonal dynamics --- and calculate the ratio between them.

As such, for phylogenies we first calculate the \ac{eps} trajectory for each clade using \ac{bnpr}. Next, and using their \ac{eps} trajectory, we calculate their expected growth rate by assuming a sigmoid growth. We additionally assume that the final \ac{eps} (\ac{eps} at sampling) estimate corresponds to the fraction of tips in the clade and we scale our data accordingly such that the unit corresponds to the maximum \ac{eps} and the fraction of tips in the clade corresponds to \ac{eps} at sampling. Thirdly and using the biphasic log-linear fit described in “Validating annual growth rate inferences from single-cell phylogenies with Wright-Fisher simulations” we derive the value for early growth. Finally, as a measure of deceleration, we calculate the ratio between expected and early growth --- a value close to 1 for this ratio implies an absence of deceleration whereas smaller values imply deceleration. 

For the longitudinal data we use the observed growth for each clone, calculated as described above. Next, we calculate the (minimal) historical growth as the growth that excludes all posterior samples that would lead to age at onset estimates exceeding lifetime (ages at onset for clones appearing below the age of -1, a heuristic value chosen to represent developmental onset of clones). Finally and as a measure of deceleration, we calculate the ratio between observed and historical growth. The interpretation for this ratio is similar to that defined in the previous paragraph for phylogenetic data --- a value of 1 implies an absence of detectable deceleration, whereas smaller values represent the minimal amount of deceleration. This method has, however a caveat --- due to the nature of this calculation (excluding posterior samples which are too slow to provide solutions within lifetime), values above 1 (indicating acceleration) are technically impossible. 

\section{Results}

\subsection{The age dependent mutational landscape of clonal haematopoiesis}

First, using this cohort --- 384 individuals aged between 54 and 93 at study entry whose blood was sampled and sequenced for a panel of 53 genes between 2 to 5 times (median: 4) over a median period of 13 years --- in a gene-agnostic way I was able to confirm a well known aspect of \ac{ch} that it increases with age in both magnitude, clonality and prevalence \cite{Jaiswal2014-rl,McKerrell2015-rl}, as illustrated in \figref{fig:ch-magnitude-increase} and \figref{fig:ch-prevalence-increase}, with 305 individuals harbouring at least one \ac{ch} mutation. Additionally, I could also verify that not only are some genes much more prevalent than others, with both \ac{dnmt3a} and \ac{tet2} being present in more than 30\% of individuals, but also that the prevalence of some genes is differentially affected by age (\figref{fig:ch-prevalence-genes}) --- particularly, while the prevalence of mutations in spliceosome genes and \ac{tet2} increases consistently with age (5.4\% increase per year (p-value=0.00124) and 6.8\% increase per year (p-value=0.000371), respectively, for a binomial regression controlling for gender), that of \ac{dnmt3a} appears to show no association with age. 

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/ch-magnitude-increase.pdf}
	\placecaption{Average \ac{ch} clone size stratified by age.}
	\label{fig:ch-magnitude-increase}
\end{figure}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/ch-prevalence-increase.pdf}
	\placecaption{\ac{ch} prevalence and clonality stratified by age.}
	\label{fig:ch-prevalence-increase}
\end{figure}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/ch-prevalence-genes.pdf}
	\placecaption{Prevalence of mutations in different genes in \ac{ch} (top). Age-stratified prevalence of mutations \ac{dnmt3a}, \ac{tet2} and spliceosome (\ac{sf3b1}, \ac{srsf2} and \ac{u2af1}) genes (bottom).}
	\label{fig:ch-prevalence-genes}
\end{figure}

Inspecting the prevalence of multiple mutations across different genes also revealed distinct patterns. Using a series of permutations to derive empirical distributions and Monte Carlo p-values, I observe that individuals with multiple \ac{tet2} mutations are overrepresented, with a single individual harbouring 8 \ac{tet2} mutations \figref{fig:ch-mutations-per-gene}. Individuals with multiple \ac{dnmt3a} mutations are also overrepresented to a smaller extent. Together with the analysis presented in \figref{fig:ch-prevalence-genes}, this can suggest that the increasing prevalence of \ac{tet2} ac{ch} is linked with a higher likelihood of mutational acquisition as age progresses. 

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/ch-mutations-per-gene.pdf}
	\placecaption{Prevalence of mutations in different genes in \ac{ch} (top). Age-stratified prevalence of mutations \ac{dnmt3a}, \ac{tet2} and spliceosome (\ac{sf3b1}, \ac{srsf2} and \ac{u2af1}) genes (bottom).}
	\label{fig:ch-mutations-per-gene}
\end{figure}

\subsection{Clones expand steadily during old age}

Having validated my clonal dynamic model as illustrated the previous chapter, I applied it to all the trajectories in this cohort. The most immediate finding is that trajectories are remarkably consistent over old age (\figref{fig:ch-trajectories-examples}), with the vast majority of clones (92.4\%) having no outliers (defined as samples falling outside the 95\% confidence interval defined by our model). For simplicity, I will also be referring to trajectories without outliers as "explained trajectories". However, this scenario is not the same for all genes --- while trajectories for clones harbouring mutations in genes such as \ac{u2af1}, \ac{dnmt3a} and \ac{tet2} have practically no outliers, those with mutations in \ac{jak2}, \ac{idh1} and \ac{srsf2} show a higher proportion of unexplained trajectories, hinting that these mutations may make clones more unpredictable (\figref{fig:ch-trajectories-explained-gene}). Given the number of mutations or available timepoints for each individual could affect the number of outliers in trajectories, I confirmed that this was not the case as per \figref{fig:ch-trajectories-explained-mut-tp}. 

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/ch-trajectories-examples.pdf}
	\placecaption{Examples of trajectories overlapped with real data. Lines represent inference while points/crosses represent real data. Grey bands represent the 90\% \ac{hpdi}.}
	\label{fig:ch-trajectories-examples}
\end{figure}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/ch-trajectories-explained-gene.pdf}
	\placecaption{Fraction of explained trajectories by gene with 90\% beta-distributed confidence intervals.}
	\label{fig:ch-trajectories-explained-gene}
\end{figure}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/ch-trajectories-explained-mut-tp.pdf}
	\placecaption{Fraction of explained trajectories by number of mutations (left) and number of available timepoints (right) with 90\% beta-distributed confidence intervals.}
	\label{fig:ch-trajectories-explained-mut-tp}
\end{figure}

To better understand whether unexplained trajectories could be attributed to the modelling assumptions or to other biological factors, I further assessed whether the proportion of explained trajectories was different between different modelling conditions --- between truncating and missense mutations, and between clones with and without explicitly modelled sites. Additionally, I checked whether sex (male vs. female) and smoking status (has smoked vs. has never smoked) could impact the proportion of explained trajectories. However, I found no statistically significant association between either of the groups according to a Chi-square test (for the model-intrinsic factors: recurring vs. non-recurring and truncating vs. missense; for other biological factors: male vs. female and has smoked vs. has never smoked) \figref{fig:ch-unexplained-trunc-site-sex-smoke}. As such, to try to better understand whether a more illustrative cause for unexplained trajectories could be found, I inspected all unexplained trajectories and found that they could be tentatively categorized as belonging to one of three main groups --- competition (one clone outcompetes a second clone, causing it to decelerate or even decrease in size), sequencing artefacts or consequences of genetic drift (a mutation randomly fluctuates to detectable levels for a single timepoint) and size-dependent deceleration (a clone grows too big and, due to the increase in the relative fitness of the population, grows more slowly), with examples presented in \figref{fig:ch-bad-trajectories-examples}. It is important to note that these explanations are not absolute, but rather possible. To avoid using poor inferences and unless otherwise noted, I only used explained trajectories and their respective coefficients in the remaining analyses of this chapter.

\begin{figure}[!ht]
    \label{gfx/dynamics-2/ch-unexplained-trunc-site-sex-smoke.pdf}
	\placecaption{Fraction of explained trajectories by modelling conditions (left) and by sex and smoking status (right).}
	\label{fig:ch-unexplained-trunc-site-sex-smoke}
\end{figure}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/ch-bad-trajectories-examples.pdf}
	\placecaption{Examples of trajectories which are not explained by the model.}
	\label{fig:ch-bad-trajectories-examples}
\end{figure}

Despite the few unexplained trajectories and thanks to an additional timepoint for 10 individuals, I was able to test the predictive performance of this model and the consistency of the trajectories. As visible in \figref{fig:trajectories-6th-tp}, most trajectories do not deviate greatly from a 6th timepoint which was not used in the inference of the coefficients for the model. Further quantifying this, as represented in \figref{fig:performance-6th-tp}, it becomes clear that the predicted changes in \ac{vaf} correspond nicely to those observed ($MAE = 3.4\%$, $R^2 = 89\%$). This further confirms that trajectories remain consistent over relatively long periods of time during old age.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/trajectories-6th-tp.pdf}
	\placecaption{Predicted trajectories for an additional timepoint for 10 individuals. Circles represent datapoints that were used to infer the coefficients in the model, whereas crosses (+) represent datapoints that were not used by the model for inference. Inferred trajectories, as well as their 90\% credible intervals, are represented by arbitrary colours.}
	\label{fig:trajectories-6th-tp}
\end{figure}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/performance-6th-tp.pdf}
	\placecaption{Observed and predicted change in \ac{vaf} for a 6th timepoint.}
	\label{fig:performance-6th-tp}
\end{figure}

\subsection{Identifying the drivers of clonal growth rate}

Considering the relative stability of the trajectories, it is now possible to consider the effect of specific sites (whose mutations are present in two or more individuals) and genes on the growth per year (\figref{fig:ch-gene-site-coefficients}; \tableref{table:ch-gene-coefficients}). I show that the growth advantage conferred by most sites is relatively small with that conferred by the gene, with \ac{srsf2}-P95H being the single exception. It is also clear that \ac{dnmt3a} and \ac{tet2}, the two most commonly mutated genes in \ac{ch}, are by no means the genes conferring the greatest fitness advantage; in fact, it is mutations in \ac{u2af1}, \ac{ptpn11} and \ac{srsf2} that are responsible for the fastest growing clones (\tableref{table:ch-gene-coefficients}). Importantly, both \ac{u2af1} and \ac{srsf2} are, together with another relatively fast gene, \ac{sf3b1}, splicing genes, making their relatively late detection in \ac{ch} more intriguing \cite{McKerrell2015}. I also quantified the fitness effect of truncating and missense separately, showing that, for \ac{cbl}, truncating mutations confer a greater fitness advantage than missense ones (the annual growth rate is 10.9\% ($CI_{90\%}=[3.3\%,18.7\%]$) higher for truncating \ac{cbl} mutations), while the reverse is observable in \ac{tp53} (the annual growth rate is 10.4\% ($CI_{90\%}=[2.6\%,18.0\%]$) higher for missense \ac{tp53} mutations; \figref{fig:ch-trunc}), while other genes for which both effects were modelled show no significant difference between truncating and missense mutations.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/ch-gene-site-coefficients.pdf}
	\placecaption{Driver growth per year for different genes and sites. Individual points and vertical lines represent the expected value and 90\% \ac{hpdi} for sites and horizontal lines and rectangles represent the expected value and 90\% \ac{hpdi} for genes.}
	\label{fig:ch-gene-site-coefficients}
\end{figure}

\begin{table}[!ht]
    \centering
    \caption{Distribution genetic coefficients.}
    \pgfplotstabletypeset[
	font=\footnotesize,
    string type,
    columns/g/.style={
        column name=Gene,
        postproc cell content/.style={@cell content=\textit{##1}},
        column type={C{.1\textwidth}}},
    columns/t/.style={
        column name=Type,
        column type={C{.1\textwidth}}},
    columns/v/.style={
        column name=Annual growth,
        column type={C{.4\textwidth}}},
    every head row/.style={before row={\toprule},after row=\midrule},
    every last row/.style={after row={\toprule}},
    every odd row/.style={before row={\rowcolor[gray]{0.9}}}
    ]\geneCoefficients
    \label{table:ch-gene-coefficients}
\end{table}    

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/ch-trunc.pdf}
	\placecaption{Difference between truncating and missense effect for genes where both effects were modelled.}
	\label{fig:ch-trunc}
\end{figure}

By including in the model the unknown cause effect, an additional parameter to the model which quantifies how different the observed growth is from conferred by gene and/or site, it is possible to see that growth is largely determined by gene and site identity (\figref{fig:ch-genetic-vs-unknown}), highlighting genetics as a strong driver of clonal growth. Nonetheless, it is worth noting that the extension of this difference depends largely on the gene --- whereas genes such as \ac{u2af1} or \ac{ptpn11} show a relatively small spread of unknown cause effects, those of \ac{asxl1}, \ac{jak2} and \ac{tet2} are considerably more dispersed (\figref{fig:unknown-by-gene}). This highlights a peculiarity of these trajectories --- whereas the wider spread of unknown cause effects in \ac{jak2} mutants is expectable due to its relatively low proportion of explained trajectories, such is not the case of \ac{tet2} or \ac{asxl1}; indeed, while more variable, trajectories in both these genes may be affected by relatively stable and extra-genetic effects as will be discussed further ahead in the Discussion. However, it should also be highlighted that a wider spread in the unknown cause effect may not necessarily imply a considerable shift in growth rate --- comparing the mean of the annual growth and the standard deviation of the unknown cause growth for different drivers, it becomes clear that, for most genes, there appears to be a clear positive trend between both \figref{fig:ch-mean-std-growth}. However, drivers such as mutations in \ac{ptpn11} or \ac{u2af1} and \ac{srsf2}-P95H have distinctly low standard deviations for the unknown cause effect when compared with their mean, suggesting these clones are particularly unaffected by non-driver effects. Conversely, \ac{jak2} clones have considerably higher standard deviations for the unknown cause effect when compared with its mean total annual growth, suggesting other factors which are not capture by this model can explain the high dynamic variability of \ac{jak2} clones.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/ch-genetic-vs-unknown.pdf}
	\placecaption{Comparison of the effects determined by gene and site (predicted by driver) with those observed.}
	\label{fig:ch-genetic-vs-unknown}
\end{figure}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/unknown-by-gene.pdf}
	\placecaption{Unknown cause effects stratified by gene.}
	\label{fig:unknown-by-gene}
\end{figure}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/ch-mean-std-growth.pdf}
	\placecaption{Comparison between the mean annual growth effect and the standard deviation of the unknown cause effect for different drivers. The 90\% credible intervals for the mean are derived from the posterior distribution, whereas the 80\% confidence intervals for the standard deviation are calculated as $\frac{\sigma\sqrt{N-1}}{\sqrt{q_{\chi^2}(0.1,N-1)}}$, with $N$ as the total number of mutations for each driver, $q_{\chi^2}(x,f)$ as the quantile function for the Chi-squared distribution for probability $x$ and $f$ degrees of freedom, and $\sigma$ is the standard deviation.}
	\label{fig:ch-mean-std-growth}
\end{figure}

Similarly to the analysis performed with the unexplained trajectories, I also investigated if any model-intrinsic or other biological factors could be underlying the differences in the unknown cause growth effect:

\begin{enumerate}
    \item Regarding \textbf{model-intrinsic} factors, I investigated whether one could expect the intra-individual variance to be smaller than the inter-individual variance, suggesting that individual-specific, rather than clone-specific, effects were at play, finding no significant difference between either group. Additionally, I assessed whether recurrence or truncation status could be associated with the unknown cause effect, but found no association between either \figref{fig:ch-variance-uc-technical}. This, together with the fact that these factors do not affect the number of explained trajectories, provides confirmation that the parameterization of our model is not biasing the results here presented;

    \item Regarding \textbf{biological} factors, I started by investigating whether age could have an impact on the unknown cause effect by measuring the Spearman correlation between the age at which the clones were first detected and the unknown cause effect. There is an association between the unknown cause effect for all genes and age ($\rho = 0.14; p = 2.56*10^{-4}$) and between \ac{tet2} clones and age ($\rho = 0.31; p = 2.33*10^{-6}$) after correcting for false discoveries. However, if \ac{tet2} clones are excluded from the all gene analysis, the result no longer is statistically significant ($\rho = 0.10; p = 3.76*10^{-4}$), pointing towards the impact of \ac{tet2} in this association (\figref{fig:ch-variance-biological} (left)). Next, I investigated the association between smoking status and sex, and the unknown cause effect, finding no association between either (\figref{fig:ch-variance-biological} (right)). Given its particular association with smoking history in \ac{ch} \cite{Dawoud2020-af} and wide spread of unknown cause factors (\figref{fig:unknown-cause-gene}), I assess whether there is a specific association between \ac{asxl1} and smoking status, but find no particular association, implying that the effect of smoking on \ac{asxl1} \ac{ch} may be associated with an increased mutation rate rather than with an increase in clone fitness. I also investigated whether the number of clones in a single individual --- the individual's clonality --- could have an impact on the unknown cause effect to assess whether competition could impact growth during old age, finding no statistically significant association for a linear regression while controlling for age ($b_{unknown_{ij}} \sim no.clones_j + AgeAtStudyEntry_j$) both when considering all genes and when performing this analysis stratified by gene.
\end{enumerate}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/ch-variance-uc-technical.pdf}
	\placecaption{Association between technical aspects of modelling and the estimated unknown cause effect: comparison between and within individual variance (left) and between truncating and recurrence status (right).}
	\label{fig:ch-variance-uc-technical}
\end{figure}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/ch-variance-uc-biological.pdf}
	\placecaption{Association between biological factors and the estimated unknown cause effect: comparison of between and within individual variance (left) and between truncating and recurrence status (right).}
	\label{fig:ch-variance-uc-biological}
\end{figure}

\subsection{Single-cell phylogenetic trees reveal lifelong clonal behaviour}

As demonstrated in the previous section, clonal dynamics can be captured over decade-long periods. However, given that mutation acquisition is relatively constant through life \cite{Abascal_2021_gjvqfm}, it is important to have a perspective into the lifelong behaviour of clones. With this in mind, single-cell colonies were derived and sequenced by collaborators for three individuals characterized by the presence of splicing gene mutations --- PD42176 (\ac{sf3b1}; 79 years old), PD34493 (\ac{sf3b1} and \ac{u2af1}; 83 years old) and PD41305 (\ac{srsf2}; 73 years old) --- which are associated with older ages \cite{McKerrell2015-rl}. Margarete Fabre, Michael Spencer Chapman and Emily Mitchell were then responsible for the generation of the phylogenetic trees represented in this section, assignment of mutations to branches and tree linearization (making the trees ultrametric); I further analysed these trees, scaling them considering the age at sampling and assigning the first 55 mutations as having happened before birth as shown by previous studies \cite{Mitchell2021-zl}, to determine the ages at onset for different mutations and to quantify the annual growth through the analysis of phylodynamic trajectories. Additionally, I extrapolated the clonal trajectories obtained from the longitudinal sequencing data and according to Wright-Fisher dynamics and compared these with those inferred using phylodynamic methods.

The first tree (PD42176; \figref{fig:tree-pd42176}) is characterized by a single, massive \ac{sf3b1}-K666N clone, which appeared between the age of 36.5 and 45.4 years old. Other mutations are presented at relatively low proportions (in only a few clades), while a chromosome Y deletion preceded all other mutations in this individual, suggesting that this must have happened during development or very early in life. Using \ac{bnpr}, a phylodynamic method, to estimate the lifelong trajectory of the \ac{sf3b1}-K666N clone, it becomes clear that this clone had an initial period during which its relative fitness was much higher than that of the other clones, progressively slowing down as it approached the time of sampling (\figref{fig:tree-pd42176}). Comparing this lifelong trajectory with the one inferred using longitudinal data, we can observe good concordance between both, with the phylodynamic estimation using the single-cell colonies highlighting an initial, more aggressive expansion which the longitudinal data fails to capture.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/trees-pd42176.pdf}
	\placecaption{Ultrametric phylogenetic tree for individual PD42176, with all studied driver mutations annotated and relevant structural rearrangements (left). Lifelong trajectory for the \ac{sf3b1}-K666N, inferred using both longitudinal targeted sequencing data and phylodynamic methods.}
	\label{fig:tree-pd42176}
\end{figure}

The tree of individual PD34493 shows four fairly large expansions (\figref{fig:tree-pd34493}), two of which are characterized as being driven by no immediately recognizable driver. The expansions that are characterized by known drivers, clones 1 and 2 (\ac{sf3b1}-K666N and \ac{u2af1}-Q157R, respectively), exhibited fairly distinct behaviour over the lifetime of this individual --- while clone 1 expanded steadily until the time of sampling, having first appeared anywhere between conception and the age of 33, clone 2 expanded rapidly, having first appeared between the ages of 39.1 and 60.1 on the back of a previous expansion with an unidentified driver, highlighting distinct behaviours from two genes frequently clustered together as functionally similar. Once again and in both cases, it is visible that the trajectories inferred from longitudinal targeted sequencing and phylodynamic methods are in good agreement. A final fascinating aspect of this phylogenetic tree is that its largest expansion has no known driver, and until it took over more than half of the carrying capacity (\figref{fig:tree-pd34493}). 

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/trees-pd34493.pdf}
	\placecaption{Ultrametric phylogenetic tree for individual PD34493, with all studied driver mutations annotated and relevant structural rearrangements (left). Lifelong trajectory for the \ac{sf3b1}-K666N and \ac{u2af1}-Q157R, inferred using both longitudinal targeted sequencing data and phylodynamic methods (right).}
	\label{fig:tree-pd34493}
\end{figure}

The final tree derived from the Sardinian cohort is that of individual PD41305, characterized as having no expansions with identifiable drivers (\figref{fig:tree-pd41305}). This individual was selected because we were interested in assessing the behaviour of its \ac{srsf2}-P95H clone; however, due to possible technical complications associated with \ac{srsf2}-P95H clones not being particularly well suited for cultivation or colony selection bias before choosing the colonies which were to be sequenced. As such and unfortunately, we were unable to capture the lifelong behaviour of this clone. Nonetheless, this tree further highlights the dimension and relevance of expansions with no known drivers that were already visible in the tree of individual PD34493 (\figref{fig:tree-pd34493}). Their lifelong behaviour, similarly to the expansions in the tree of individual PD34493 excluding the \ac{u2af1}-Q157R, is characterized by appearing early in life (any time between conception and 22 years old). Particularly for individual PD41305, expansions with no known driver expand quickly early in life, stabilizing during later years.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/trees-pd41305.pdf}
	\placecaption{Ultrametric phylogenetic tree for individual PD34493, with all studied driver mutations annotated and relevant structural rearrangements (left). Lifelong trajectory for two expansions with no known drivers, inferred using phylodynamic methods (right).}
	\label{fig:tree-pd41305}
\end{figure}

Considering all three phylogenetic trees, it is also worth noting that chromosome Y deletion is prevalent during old age, with all three (male) individuals presented here harbouring at least an occurrence of chromosome Y deletion and with the tree for individual PD41305 harbouring three independent losses of chromosome Y. Additionally, the prevalence and dynamics of expansions with no known drivers --- 5 out of 8 --- highlights the unknown dimensions of \ac{ch}. Finally, by comparing trajectories inferred from two separate methods it is possible to observe fairly good concordance, as observable in \figref{fig:trees-coefficients-ages-at-onset}, presenting the quantification of the growth dynamics during the longitudinal sampling and on the ages at onset. This, together with the validation experiments in previous chapter using simulations, provides further orthogonal validation of both approaches. Interestingly, expansions with no known drivers appear to be remarkable slow during later stages of life, having largely stagnated despite not having reached the carrying capacity of the \ac{hsc} population.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/trees-coefficients-ages-at-onset.pdf}
	\placecaption{Comparison of the growth dynamic coefficients calculated using longitudinal targeted sequencing and phylodynamic modelling over the study period (left) and the ages at clone onset using longitudinal targeted sequencing and phylogenetic trees.}
	\label{fig:trees-coefficients-ages-at-onset}
\end{figure}

\subsection{Clonal deceleration is ubiquitous in old age}

Upon calculating the ages at clone onset for all the clones in our cohort, I this estimate exceeded lifetime (preceded birth) for large fraction of clones (34.4\%; \figref{}). Together with the fact that the trees in the previous section revealed that several expansions had a period during which their growth was considerably faster than the one observed, I set out to quantify this apparent phenomenon of deceleration. Thanks to four additional phylogenetic trees provided by collaborators (Emily Mitchell and Peter Campbell) from their 2021 publication investigating changes to haematopoiesis \cite{Mitchell2021-zl}, I was able systematically quantify the effect of deceleration on clonal trajectories.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/exceeds-within-lifetime-representation.pdf}
	\placecaption{Representation of the lifelong extrapolation of two longitudinal trajectories, one of which exceeds lifetime (age at onset precedes the birth of the individual) and the other is fully contained within lifetime.}
	\label{fig:exceeds-within-lifetime-representation}
\end{figure}

To do this, I first inferred the phylodynamic trajectories for all expansions in all trees across both cohorts (\figref{fig:trajectories-fabre-mitchell}), highlighting the ubiquitousness of this effect --- even if only qualitatively, it is possible to see that several of the clones show some form of deceleration later in their life. Two particular aspects of this representation is the much higher frequency of detected expansions in the trees from \etal{Mitchell}, attributable to the fact that phylogenetic trees in this publication were constructed using approximately 4 times more colonies than ours, and the prevalence of \ac{dnmt3a} clones in these trees, with two out of four trees harbouring \ac{dnmt3a} clones.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/trajectories-fabre-mitchell.pdf}
	\placecaption{Lifelong clonal trajectories inferred for \cite{Fabre2021-uw} (left) and \cite{Mitchell2021-zl}. Fractions of the trajectories showing low variability are represented as full lines, whereas fractions with high variance are represented as dotted lines. A thin grey band represents the carrying capacity from \cite{Lee-Six2018-lp}.}
	\label{fig:trajectories-fabre-mitchell}
\end{figure}

I then fitted biphasic log-linear models to each phylodynamic trajectory in order to quantify early and late growth. Having both these quantities, I first calculated what the expected clone size would be if the early growth was representative of lifelong clonal growth (expected clone size) and compare it with the observed clone size (assuming that the fraction of tips belonging to an expansion are representative of the size of the clone; \figref{fig:expected-observed-ratio-representation}). As per \figref{fig:expected-observed-ratio}, the expected size of the clone vastly exceeds that which was observed, showing that clonal deceleration was essential to explain the observed clone sizes.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/expected-observed-ratio-representation.pdf}
	\placecaption{Representation of the extrapolation of the early growth (expected clone size) and how it compares with the observed clone size.}
	\label{fig:expected-observed-ratio-representation}
\end{figure}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/expected-observed-ratio.pdf}
	\placecaption{Ratio between observed and expected clone size.}
	\label{fig:expected-observed-ratio}
\end{figure}

Next, and considering that extrapolating the trajectory of 34.4\% of clones leads to age at clone onset times that precede birth (\figref{fig:expected-observed-ratio-representation}, \figref{fig:exceeds-within-lifetime}), I inspect the extent to which this happens when considering genes separately. In other words, I assessed whether there as a specific gene bias associated with clonal deceleration, an effect that would cause ages at clonal onset to exceed lifetime. As per \figref{fraction-clones-exceeding-lifetime}, it becomes clear that while some genes such as those involved in splicing, \ac{ptpn11}, \ac{idh1}, \ac{idh2} and \ac{ctcf} are practically unaffected by clonal deceleration, others such as \ac{dnmt3a}, \ac{brcc3} and \ac{tp53} are particularly affected by this, with over 60\% of \ac{dnmt3a} clones exceeding lifetime. Comparatively, \ac{tet2}, the other most prevalently mutated \ac{ch} gene, is remarkably different to \ac{dnmt3a} --- an effect that can be attributable to \ac{tet2} clones growing at higher observed annual growth rates than \ac{dnmt3a} clones, implying that \ac{dnmt3a} clones either grow slower than \ac{tet2} clones or that deceleration has a higher impact on their annual growth rates. 

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/exceeds-within-lifetime.pdf}
	\placecaption{The relationship between clone size, age and annual growth and its effect on age at clone onset.}
	\label{fig:exceeds-within-lifetime}
\end{figure}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/fraction-clones-exceeding-lifetime.pdf}
	\placecaption{Fraction of clones exceeding lifetime stratified by gene.}
	\label{fig:fraction-clones-exceeding-lifetime}
\end{figure}

This gene bias led us to further quantify deceleration in growth inferred from both longitudinal targeted sequencing data and from phylodynamic estimates. For the former, I calculate the historical growth of a gene (i.e. the expected growth excluding samples from the posterior which exceed lifetime --- \figref{fig:historical-growth-representation}) and use the ratio between observed and historical growth --- $deceleration_{long}$ --- as a measure of deceleration. For the latter, I use the late growth as estimated by the biphasic log-linear fit and calculate the expected growth (assuming that the carrying capacity for the clone saturates the expansion grows bigger and that the clone size is proportional to the size of the clades in the tree) to obtain the ratio between the late and expected growth --- $deceleration_{phylo}$. Both of these quantities are analogous in their interpretation --- 1 represents no deceleration, whereas values below 1 represent an increasing amount of deceleration. To gain a better sense of $deceleration_{phylo}$, I use the simulations described in the previous chapter to define two distinct classes of trajectories based on their simulated clonal trajectory --- if the ratio between early and late growth for the simulated clonal trajectory is below 0.80, these trajectories are considered saturating; otherwise, they are considered to be near constant. As per \figref{fig:deceleration-figure-full}, I show that this phenomenon, while not affecting all expansions or genes equally, is remarkably prevalent --- indeed, the real data appears to be a mixture of both near constant and saturating trajectories. Additionally, closer inspection of the differences between historical and observed growth further reveals the remarkable difference between \ac{dnmt3a}, \ac{brcc3} and \ac{tp53} and other genes (\figref{fig:historical-growth-tally}).

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/historical-growth-representation.pdf}
	\placecaption{Fraction of clones exceeding lifetime stratified by gene.}
	\label{fig:historical-growth-representation}
\end{figure}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/deceleration-figure-full.pdf}
	\placecaption{Quantification of deceleration from phylodynamic and longitudinal data. "near constant" and "saturating" were obtained from simulated phylogenetic trees ($deceleration_{phylo}$), "real data (phyl.)" was obtained from real phylogenetic trees ($deceleration_{phylo}$) and "real data (long.)" was obtained from real longitudinal data.}
	\label{fig:deceleration-figure-full}
\end{figure}

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/historical-growth-tally.pdf}
	\placecaption{Observed and historical growth stratified by gene.}
	\label{fig:historical-growth-tally}
\end{figure}

\subsection{Clonal haematopoiesis is characterized by steady clonal onset}

Knowing that clones decelerate, I have to adjust the age at onset estimates similarly to what was done in the previous chapter to avoid having estimates precede birth --- more concretely, I consider it to be impossible for clones to have appeared before the age of -1, an arbitrary cut-off representing conception. I show here that mutations in \ac{srsf2}-P95H and \ac{u2af1} appear much later in life than those in other genes, contributing to the previously reported age association of splicing genes \cite{McKerrell2015-rl} (\figref{fig:ages-at-onset}). Mutations in every other gene, considering the pervasive effects of deceleration, is likely to appear consistently through life.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/ages-at-onset.pdf}
	\placecaption{Age at onset intervals for different drivers.}
	\label{fig:ages-at-onset}
\end{figure}

A key aspect of the age at clonal onset is its close association with the annual growth --- in \figref{fig:age-at-onset-vs-growth} it is clear that clones harbouring mutations which appear later in life grow faster than other mutations; this is especially the case for \ac{srsf2}-P95H and \ac{u2af1}. Additionally, the converse is visible for clones characterized by \ac{dnmt3a} mutations which are projected to appear very early in life, an effect that is impossible to disentangle from the deceleration described in the previous section of the results. Even considering a relative low threshold --- 0.2\%, which would be the theoretical threshold for this longitudinal cohort (\autoref{appendix:theoretical-lod}).

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/age-at-onset-vs-growth.pdf}
	\placecaption{Association between age at onset and annual growth. Points represent the expected value, whereas the coloured lines represent the 90\% credible intervals. The black line and shaded area represent the theoretical limit of detection at a \ac{vaf} of 0.2\%.}
	\label{fig:age-at-onset-vs-growth}
\end{figure}

There was the possibility --- as previously discussed in the previous chapter --- that our estimates, particularly those associated with \ac{u2af1} and \ac{srsf2}-P95H, could be due to a particularly convenient set of values for the population size and number of generations per year. To better clarify this, I also show that this particular age bias remains for a wide array of possible values for the population size and generation time (\figref{fig:ages-at-onset-benchmark}). Additionally, it is possible that our choice of parameters does not allow us to detect the late clonal onset of other drivers, particularly those in \ac{ptpn11} for which there appears to be a bias for a later age at clone onset; however, further studies will have to be conducted to assess this.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/ages-at-onset-benchmark.pdf}
	\placecaption{Age at onset intervals for different drivers given different combinations of values for the population size and number of generations per year.}
	\label{fig:ages-at-onset-benchmark}
\end{figure}

While no association was observable between smoking history and clonal dynamics, I attempted to see if an association could be detected between clone onset and smoking, in order to better understand if the association between smoking and \ac{ch} was mediated by an increased mutation rate, rather than by a change in clonal fitness. However, I found no changes in the distributions for the age at clone onset between individuals smoking during the age at clone acquisition and otherwise (\figref{fig:smoking-age-density}).

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/smoking-age-density.pdf}
	\placecaption{Age at onset distribution stratified by whether the individuals were smoking at the time of clonal acquisiton.}
	\label{fig:smoking-age-density}
\end{figure}

\subsection{Clone fitness and its relation to progression}

The final part of this project focused on analysing whether there was a more concrete connection between the observed growth effects during old age. To this effect, I used the risk coefficients for \ac{aml} onset from \cite{Abelson2018-wh} and the dN/dS values calculated for different sites by Margarete Fabre in a \ac{mds} cohort and in an \ac{aml} cohort to verify whether any association was present between these values --- \ac{aml} risk and selection in \ac{mds} and \ac{aml}, respectively --- and the observed growth rate during old age. As shown in \figref{fig:ch-risk-selection}, there is a clear association between \ac{aml} risk and annual growth rate during old age (\missing{R value}) and between selection in \ac{mds} and \ac{aml} and annual growth rate.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/ch-risk-selection.pdf}
	\placecaption{Association between annual growth and \ac{aml} risk (left) and selection in \ac{mds} and \ac{aml} (right).}
	\label{fig:ch-risk-selection}
\end{figure}

\FloatBarrier

\section{Discussion}

In this work I present a comprehensive study of the dynamics of \ac{ch} by using longitudinal targeted sequencing data and phylogenetic trees derived from single-cell colonies. By doing so, I present distinct growth coefficients that depend on gene identity, as well as a clear perspective on the variability associated with each gene. Additionally, I show that, while clonal trajectories are very consistent during old age, they decelerate over life, an effect that is likely to be associated with an increasingly competitive oligoclonal landscape. I also show how most clones appear consistently through life, except those characterized by mutations in \ac{u2af1} and \ac{srsf2}-P95H. Finally, I show how selection in myeloid malignancies and risk of progression to \ac{aml} is associated with dynamics.

A striking aspect of this analysis arises when considering \ac{dnmt3a} and \ac{tet2}, the two most frequently mutated genes in this cohort (\figref{fig:ch-prevalence-genes}), with both showing up among the most mutated genes in \ac{ch} across several studies \cite{Jaiswal2014-rl,Genovese2014-eu,Dawoud2020-af}: while \ac{dnmt3a} shows very clear signs of deceleration as evidenced in both targeted sequencing data and phylogenetic trees (\figref{fig:historical-growth-tally}; \figref{fig:trajectories-fabre-mitchell}), \ac{tet2} is not nearly as affected by this; in fact, here I show how the prevalence of \ac{tet2} increases with age, whereas that of \ac{dnmt3a} shows no clear sign of being affected by age. A number of reasons can be associated with this: the most obvious one would be that \ac{dnmt3a} clones are simply less fit than \ac{tet2} clones while being exposed to similar mutational stimuli --- \ac{dnmt3a} being less fit than \ac{tet2} would not only translate in \ac{tet2} clones outcompeting \ac{dnmt3a} clones; in fact, it would result in \ac{dnmt3a} being much more prominently affected by rises in the average fitness of the population in relative terms. On top of this, links have been established between infection or inflammation and both \ac{dnmt3a} \cite{Hormaechea_Agulla2019-cd,Hormaechea-Agulla2021-kr} and \ac{tet2} \cite{Cai2018-yi} growth, making it complicated to assign phenomenona such as inflammaging to the differences between both genes \cite{Franceschi2018-be}. However, recent work on \ac{mca} in \ac{ch} and its association with specific mutations has shown that \ac{tet2} is associated with \ac{cnloh} events in the same chromosome as \ac{tet2}. Simply put, \ac{cnloh} happens when, during cell division, chromosome segregation leads to there being two copies of a previously heterozygous mutation in the same cell, resulting in two cells with two and no copies of the mutation, respectively. These events, which are impossible to call given my data, can contribute to the incorrect estimation of the fraction of cells harbouring a \ac{tet2} mutation (if a mutation is homozygous rather heterozygous, the counts would have to be adjusted) and potentially an overestimation of the fitness of clones harbouring such events, particularly considering that these events increase the risk of \ac{aml} \cite{Gao2021-ph}, which is itself associated with clonal dynamics as demonstrated above. Additionally, \ac{dnmt3a} loss-of-function is more heavily associated with increasing \ac{hsc} self-renewal and decreasing differentiation \cite{Jeong2018-pn}, whereas \ac{tet2} loss-of-function leads to an increased myeloid-differentiation bias \cite{Zhang2016-ys}. These effects combined may explain why \ac{dnmt3a} mutations become increasingly less detectable in blood while \ac{tet2} overtakes it as the most prevalent gene. If this is the case and considering that I found no difference between truncating and missense mutations (\figref{fig:ch-trunc}) in both \ac{tet2} and \ac{dnmt3a}, this effect may also impact mutations in these genes which do not lead to loss-of-function. These effects, combined with deceleration, may also explain the discrepancies between this work and that by \etal{Watson} --- while the average lifelong fitness conferred by \ac{dnmt3a} mutations is relatively high, its growth during older ages is much slower than expected given its \ac{vaf} distribution during old age.

Mutations in \ac{u2af1} and \ac{srsf2}-P95H were the ones conferring the highest fitness advantage to clones and appearing only at relatively old ages in contrast with other genes which appeared consistently over life, explaining the previously reported later onset of splicing gene mutations \cite{McKerrell2015-rl}. This particular age bias is further highlighted once their prevalence across other myeloid conditions is considered \cite{fig:prevalence-srsf2-p95h-u2af1} --- while mutations in these drivers appear at a relatively low prevalence in our cohort when compared with haematological malignancies, it is considerably higher than that observable in other studies concerning \ac{ch}, which may be attributable to the fact that this cohort is on average older than other cohorts. Mutations in these genes have also been shown to be practically absent from cases of juvenile or paediatric haematological malignancies, appearing only in adult cases \cite{Kar2013-nw,Takita2012-yo}, a possible consequence of the later age at onset. Specifically considering \ac{srsf2}-P95H, it is worth highlighting that other recurring \ac{srsf2} mutations --- P95A and P95L --- confer a considerably smaller fitness advantage when compared with P95H. This is striking because all three mutations affect the same residue, a proline which interacts directly with \ac{rna} and is responsible for exon recognition \cite{Daubner2012-zp,Kim2015-qz}. P95H, however, is responsible for the most striking changes when considering the number of affected sites \cite{Rahman2020-bh,Zhang2015-xu}. A possible factor contributing to the difference between P95H and other P95 mutations is the fact that a change from proline to histidine is much more dramatic in terms of amino acid polarity when compared with changes to alanine or leucine. Mutations in \ac{ptpn11} are also worthy of some attention --- similarly to \ac{u2af1} and \ac{srsf2}-P95H, these mutations confer a considerable fitness advantage and are associated with syndromes which predispose to oncological diseases \cite{Digilio2002-nq,Tartaglia2001-ru,Kratz2005-qh}, making their prevalence in this cohort --- 2.3\% --- quite surprising.

\begin{figure}[!ht]
	\placefigure{gfx/dynamics-2/prevalence-srsf2-p95h-u2af1.pdf}
	\placecaption{Prevalence of \textit{SRSF2}-P95H (top) and \textit{U2AF1} across different conditions, in other \ac{ch} studies and in the cohort used in to study clonal dynamics in this dissertation. Data sources were collated from different sources: \textit{U2AF1} prevalence in \ac{aml} \cite{Bamopoulos2020-zx}, in \ac{mds} \cite{Wang2020-dr}, in \ac{cmml} \cite{Kar2013-nw}; \textit{SRSF2} in \ac{mds} \cite{Arbab_Jafari2018-st}, in \ac{aml} \cite{Bamopoulos2020-zx}, \textit{U2AF1} and \textit{SRSF2} prevalence in \ac{ch} \cite{Jaiswal2014-rl,Genovese2014-eu,McKerrell2015-rl,Zink2017-zi,Acuna-Hidalgo2017-ng,Young2016-du,Midic2020-zh,Young2019-rz}. To obtain probability of an individual harbouring a mutation on either \ac{srsf2} or \ac{u2af1} of myeloid origin I marginalize the other probabilities ($P(mut_{myeloid}) = P(mut|AML)P(AML) + P(mut|CMML)P(CMML) + P(mut|MDS)P(MDS) + P(mut|CH)P(CH)$).}
	\label{fig:prevalence-srsf2-p95h-u2af1}
\end{figure}

The fact that several expansions in the phylogenetic trees derived for this study have no known driver is also worth highlighting (5 out of 8; \figref{fig:tree-pd34493}; \figref{fig:tree-fig:tree-pd41305}; \figref{fig:tree-pd42176}). This further contributes to the increasingly common knowledge that \ac{ch} is still vastly understudied, similarly to what has been reported elsewhere \cite{Poon2020-ek,Zink2017-zi} and highlights one of the shortcomings of this study --- while focusing on a small panel is considerably more feasible from an economic and labour perspective, it does not permit a comprehensive genetic characterization of \ac{ch} at the genome level or of structural variants \cite{Gao2021-ph,Terao2020-dd,Loh2020-vz,Loh2018-xp}. This could have been relevant to better understand the relative advantage of \ac{tet2} mutations over \ac{dnmt3a} or of \ac{ppm1d} mutations, both of which are associated with \ac{mca} events on the chromosome where they are located. Additional copy-number characterization would have helped to decipher the reason for \ac{jak2} trajectories to be so prone to have a non-constant growth rate as it would have allowed us to detect uni-parental disomy events, frequently associated with \ac{jak2} \cite{Wang2014-qz}. Having access to copy-number information could have also helped in assigning mutations to specific clones, rather than assuming that mutations are mutually exclusive in the same individual. In any case, this is likely to be true for most clones \cite{Miles2020-fz}, but one should expect deviations from this. 

Deceleration appears to be a phenomenon affecting most clones and can have a wealth of possible explanations --- while these have been explored above for \ac{dnmt3a} clones, it still is necessary to highlight a few other factors that may lead to a decreased ability for clones to reach their theoretical carrying capacity. Firstly, the most likely solution is that drivers which we are unable to detect are responsible for filling the bone marrow since, as mentioned, a large fraction of \ac{ch} remains unexplored. Additionally, it is important to consider that the bone marrow is not as homogeneous as we assume it to be --- the stem cell niche, where \ac{hsc} are known to thrive, gets depleted in a non-homogeneous manner as individuals grow older \cite{Crane2017-hl}; it might the case that a clone could be fit enough to out-compete others but has no space where its growth could be viable due to an inability in fostering the constitution or restitution of an adequate stem cell niche. Indeed, the vasculature that enables the existence of stem cell niches is negatively affected by age, which would reduce the ability of \ac{hsc} to exist in the bone marrow \cite{Tanaka2017-pu,Wang2017-rg}. Additionally, the \ac{hsc} population increases considerably with age, as does the myeloid-to-lymphoid bias in cell differentiation \cite{Pang2011-dw}, which can simply lead to a greater denominator concerning the calculation of \ac{vaf} if this expansion happens only in specific sites of the bone marrow --- in a scenario where panmixis is not possible and the population is not homogeneous, a clone may be unable to grow because the local stem cell niche does not permit it, while other regions are still capable of driving clonal expansions. Finally, it is worth considering that murine studies show that an event of \ac{hsc} depletion does not cause considerable harm to the body's ability to produce blood cells, with progenitors and short-term \ac{hsc} playing a key role in the production of peripheral blood cells --- a consistent shift from a predominantly \ac{hsc} population to one composed of progenitors incapable of consistent long-term self-renewal could be responsible for the apparent slowing down of clones without any of them reaching the carrying capacity of the bone marrow \cite{Schoedel2016-bi,Mitchell2021-zl}.

Finally, the association between clonal dynamics and both \ac{aml} onset risk and selection in \ac{aml} and \ac{mds} highlights the connection between a clone's ability to expand within the \ac{hsc} population and transformation and progression to disease, and how this may be unrelated to the relative prevalence of different mutations --- while \ac{tet2} and \ac{dnmt3a} are highly prevalent in \ac{ch}, both their clonal dynamics and their contribution to disease progression are considerably lower than those of \ac{u2af1} and \ac{srsf2}. This has the potential to help clinicians in further stratifying \ac{ch} --- for instance, while a relatively large \ac{dnmt3a} clone may be of little importance, a large \ac{u2af1} clones would warrant closer monitoring of disease progression.

\section{Code availability}

The code for this analysis and the associated notebooks are available in \url{https://gerstung-lab.github.io/ch-dynamics/}, more specifically in "Growth rate coefficients and age at onset inference, possible associations with phenotype", "Analysis of the phylogenetic trees in Mitchell et al. (2021)" and "Investigating the historical growth effect and poor fits".