\chapter{Introduction}

\section{Blood in normal adult individuals}

\subsection{Haematopoiesis in normal adult individuals}

Haematopoiesis is the process that generates the entire population of blood cells and is carried out by \ac{hsc} in the bone marrow. This happens continuously through life and approximately one trillion blood cells are produced every day \cite{Hoggatt2013-jl} --- equivalent to almost 19 quadrillion cells over adult life. \ac{hsc} have two defining abilities --- self-renewal and multipotent differentiation. As such, they are not only able to continuously repopulate the bone marrow, but are also capable of differentiating into all possible lineages of blood cells, including erythrocytes (or \ac{rbc}) and leukocytes (or \ac{wbc}) such as granulocytes, lymphocytes and monocytes \cite{Seita2010-ms,Till1961-mi,Becker1963-wm}. However, despite their very high rate of blood cell generation, it has been estimated that successive self-renewing cell divisions happen every 2-20 months for a population size of 50,000-200,000 \ac{hsc} \cite{Lee-Six2018-lp}. This entails that the number of \ac{hsc} effectively giving rise to the genetic variation in mature blood cells is several orders of magnitude smaller than what their productivity would otherwise suggest. 

As mentioned, \ac{hsc} have the ability to populate the blood, giving rise to a diverse landscape of mature blood cells. The process of \ac{hsc} differentiation --- of haematopoiesis itself --- starts as early as 30 days after conception in the yolk sac \cite{Rieger2012-bh}, the first fetal and extra-embryonic membrane that is formed in all mammals \cite{Dong2018-se}. Later, \ac{hsc} start being formed in the placenta and subsequently in the liver. Only at about 12 weeks after conception do fetal \ac{hsc} start migrating to the spleen and only closer to birth do the first \ac{hsc} appear in the bone marrow \cite{Rieger2012-bh}. Interestingly, \etal{Lee-Six} determined in \cite{Lee-Six2018-lp} that all blood derives from a single common ancestor which predates gastrulation, the process of embryonic development characterized as the formation of the gastrula from the blastocyst happening three weeks after conception \cite{Muhr2020-ep}. This has strong implications for the study of somatic mutations and evolution in the blood as it defines the period over which a single genotype --- that of the germline --- accumulates mutations as this process of development eventually leads to the better studied process of adult haematopoiesis. The most relevant difference between embryonic and adult haematopoiesis is that fetal \ac{hsc} demonstrate a much higher ability to regenerate haematopoiesis \cite{Becker1965-xg,Holyoake1999-nz,Nicolini1999-xe}.

At this point, it is worth noting that when discussing haematopoiesis it is difficult to accurately describe this process as exclusively determined by characteristics intrinsic to the \ac{hsc} population or individual \ac{hsc}. Indeed, there is evidence that this mechanism is highly complex, with a role being played by, among others, the \ac{hsc} niche (the specialized micro-environment where \ac{hsc} reside), extracellular signals, and transcription factors \cite{Wang2016-oq}. While these factors are not a particular concern of this dissertation, a few are mentioned to highlight their multiplicity and importance to the haematopoietic process. Studies on the \ac{hsc} niche show that, for example, endothelial cells and perivascular cells --- cells encircling the endothelial cells of capillaries and microvessels \cite{Crisan2012-nl} --- are localised near \ac{hsc} \cite{Ding2012-hv,Zhou2014-fd} and play a key role in \ac{hsc} regulation through the secretion of stem cell factor and CXCL12, two crucial factors in stem cell maintenance \cite{Ding2012-hv,Sugiyama2006-vc,Greenbaum2013-rx}. Several extracellular signals also play a large role in \ac{hsc} regulation. Particularly, the interaction of c-Kit ligand with c-Kit receptor on \ac{hsc} is essential for the survival and the loss of the latter leads to the death of embryos and depletion of mature blood cells \cite{Huang1990-in}, and a deficiency of transforming growth factor \Beta{} receptor 2 --- responsible for binding to transforming growth factor \Beta{1}, expressed by several peripheral blood cells, stromal cells and endothelial cells \cite{Zhao2014-dv} --- in \ac{hsc} leads to reduced repopulation activity \cite{Yamazaki2011-yv}. As for transcription factors, pre-B cell leukemia transcription factor 1 deletion in \ac{hsc} affects their ability to differentiate and self-renew by maintaining their quiescence \cite{Ficara2008-so}, whereas transcription factors such as \ac{runx1} are essential for \ac{hsc} maintenance and survival in embryonic development \cite{Okuda1996-rd}, losing its importance in the maintenance of \ac{hsc} in adult individuals \cite{Ichikawa2004-ow}. Instead, mutations in such factors result in a few abnormalities in cell differentiation and a mild myelodysplastic phenotype \cite{Ichikawa2004-ow,Growney2005-zx}.

\paragraph{Haematopoiesis as specification.} The best way to think about haematopoiesis is as a cascading loss of multilineage potential and specification \cite{Rieger2012-bh}. \figref{fig:haema} offers a schematic representation of this process. There are many factors and cells contributing to adult haematopoiesis but \ac{hsc} are the only cells capable of both long-term regeneration and differentiation, losing their self renewal ability as they further differentiate \cite{Rieger2012-bh,Benveniste2010-au}. This loss of self-renewal potential happens along the complex and multi-branched lineage that begins with \ac{hsc} and ends, for the purpose of this dissertation, with circulating blood cells. Over the following paragraphs I provide a simplified description of the differentiation process of \ac{hsc}, with a more complete version being present in, for example, \cite{Rieger2012-bh}. It is worth noting that these lineages and categories are well-defined and the method which is typically used to classify the stage of development of individual cells is based on the identification of surface markers through flow cytommetry \cite{Spangrude1988-xs,Ng2017-cp} --- for instance, \ac{hsc} express no Lineage (Lin) or CD38 but express CD34 (they are Lin-CD34+CD38- cells), whereas committed lymphoid progenitors are Lin-CD34+CD45RA+CD10+ cells (cells not expressing Lin, but expressing CD34, CD45RA and CD10) and naive T lymphocytes are CCR7+CD27+CD45RA+CD28+ (cells expressing CCR7, CD27, CD45RA and CD28) \cite{Miller1999-en,Appay2008-sk}. These fine differences between cells impact their function and are dependent on the identification of specific cellular markers through techniques which I did not use during this work; as such, I focus on a simpler and coarser-grained definition of haematopoiesis.

\begin{figure}[!hb]
    \placefigure{intro/blood-cell-formation}
    \placecaption{Simplified representation of haematopoiesis. HSC commit to different paths which can change depending on specific chemical stimuli --- the megakaryocyte-erythroid lineage, giving rise to megakaryocytes and erythrocytes (or red blood cells), the granulocyte-macrophage lineage, giving rise to granulocytes (neutrophils, basophils and eosinophils) and monocytes, and the lymphoid lineage, giving rise to natural killer cells and T- and B-lymphocytes. Dashed arrows represent relatively uncommon transitions, whereas full arrows represent common transitions.}
    \label{fig:haema}
  \end{figure}

\paragraph{Progenitors.} The process of differentiation first gives rise to progenitor cells from stem cells. These cells can have different degrees of potency depending on how many lineages they can give rise to (multipotent, oligopotent or unipotent) and are generally less capable of self-reneweal when compared with stem cells. Particularly for haematopoiesis, \ac{hsc} give origin to multipotent progenitor cells with intermediate-term repopulating capacity, which give rise to multipotent progenitor cells with short-term repopulating capacity. From here on, lineages consistently lose their multilineage potential. First, if cells don't become megakaryocyte-erythrocyte progenitors (responsible for the generation of megakaryocytes, whose fragmentation leads to the production of platelets, and \ac{rbc}), they lose their megakaryocyte-erythrocyte potential, becoming lymphoid-primed or early lymphoid progenitors \cite{Adolfsson2005-iq}. If after losing their megakaryocyte-erythrocyte potential cells do not become common myeloid progenitors they become common lymphoid progenitors \cite{Kondo1997-fb}. However, this does not mean these cells lose their potential to become myeloid cells; indeed, evidence shows that a small fraction of myeloid cells comes from common lymphoid progenitors or even early T- and B-cell progenitors \cite{Schlenner2010-ga,Kawamoto2010-yp}. 

\paragraph{Lymphoid cells.} Most common lymphoid progenitors mature into lymphocytes and further commit to two possible paths: migrating to the thymus to become T-cells, or staying in the bone marrow or migrating to the spleen to become B-cells \cite{Inlay2009-as}. Both T- and B-cells have a wide repertoire of functions but are mainly responsible for the recognition of and response against specific antigen epitopes and are heavily involved in the acquired immune response \cite{Cano2013-gu}. T-cells (characterized as expressing T-cell receptors) can then be further divided into CD4+ cells, which can have roles as diverse as chemokine production, cellular immunity promotion against intracellular pathogens and induction of the transformation of B-cells into plasma cells, and into CD8+ cells, which have an essentially cytotoxic role (upon recognition of the target cell, cytolytic granules are released, leading to the degradation of the target cell) \cite{Cano2013-gu}. B-cells (characterised as expressing B-cell receptors), on the other hand, are responsible for releasing antibodies which facilitate the recognition of cellular targets by other leukocytes (mainly T-cells) and can transform into plasma cells --- which have increased antibody secretion capabilities --- upon helper T-cell signalling \cite{Cano2013-gu,Janeway2001-mb}. 

\paragraph{Myeloid cells.} If cells commit to myeloid lineage, they can further commit to the megakaryocyte-erythrocyte lineage or to the granulocyte-macrophage lineage \cite{Akashi2000-ck}. Intuitively and as mentioned, the megakaryocyte-erythrocyte lineage can further differentiate into megakaryocytes, whose fragmentation generates platelets \cite{Wright1906-af}, and erythrocytes, whose main function is the transport of oxygen to relevant tissues \cite{Jensen2009-be}. The granulocyte-macrophage lineage is responsible for the generation of granulcoytes (neutrophils, eosinophils and basophils) and monocytes. Monocytes are the default lineage for the granulocyte-macrophage lineage and are long-lived cells capable of differentiating into macrophages (immune cells involved in the detection, phagocytosis and destruction of pathogens, as well as in antigen presentation \cite{Mills2012-rc}) \cite{Jakubzick2017-is}. Monocytes and macrophages have also relevant roles in controlling inflammation (with context-dependent pro- or anti-inflammatory capabilities \cite{Rivollier2012-we,Bain2013-qe}) and, by migrating to the tissue, become antigen-presenting cells \cite{Jakubzick2013-az,Plantinga2013-kn}. Granulocytes, which are much more short-lived than monocytes \cite{Athens1961-cl}, can be further divided into neutrophils, eosinophils and basophils, all of which participate in the innate immune response. Neutrophils are circulating \ac{wbc} and primary mediators of the immune response against pathogens of bacterial and fungal origin \cite{Malech2014-dm}. Eosinophils are involved in the immune response to parasitic pathogens, allergic reactions and asthma \cite{Uhm2012-ja,Huang2016-br}. Finally, basophils are more elusive --- until recently, very little was known about their nonredundant roles in the immune system \cite{Miyake2017-rh,Karasuyama2011-ws} --- but recent studies have shown that they play a role in the mediation of type 2 helper T-cell subtype differentiation \cite{Hida2005-lj,Oh2007-yr} and, together with eosinophils, in parasitic infections \cite{Karasuyama2011-ws,Huang2016-br}.

\subsection{The analysis of blood in normal adult individuals}

The diversity described in the previous section is complicated, especially when one wants to consider cell morphology. While flow cytometry can be used to derived simple shape descriptors for individual cells (size or granularity \cite{Shapiro2005-ud}) only through the analysis of cell morphology --- or cytomorphology --- using a \ac{wbs} can clinicians gain a comprehensive perspective on cell morphology. However, while informative, the analysis of a \ac{wbs} is generally not the first way through which experts analyze blood --- after anamnesis, automated blood counts are often the following step in haematological diagnostic.

\subsubsection{Blood counts}

\Ac{cbc} are fully or practically automatic and give clinicians a few standardized parameters that describe the distribution of specific blood components, which I describe below with their common names in the clinic in parentheses. Currently, one can expect a \ac{cbc} to provide values pertaining to the concentration of \ac{rbc} and \ac{wbc} in blood (\ac{rbcc} and \ac{wbcc}, respectively), the concentration of haemoglobin in blood (\ac{hb}), the average size of \ac{rbc} (\ac{mcv}), the width of the distribution that characterises \ac{rbc} volume (\ac{rdw}), the average haemoglobin and haemoglobin concentration in each cell (\ac{mch} and \ac{mchc}, respectively), the percentage of blood occupied by cells and by \ac{rbc} (\ac{pcv} and haematocrit, respectively), the concentration of platelets in the blood and the average platelet size (\ac{plt} and \ac{mpv}, respectively), the number of immature \ac{rbc} (reticulocytes) in the blood (reticulocyte count) and the speed at which \ac{rbc} descend in a standardized tube (\ac{esr}) \cite{Bain2014-oc,Evans1991-ri}. There are normal and expected values for these parameters, but it should be noted that their values are altered for a wide range of characteristics, conditions or circumstances, such as gender \cite{Tong2019-eq,Guillet1998-hw}, ethnicity \cite{Lee2019-pr,Tong2019-eq}, pregnancy \cite{England1976-ff} or menstrual cycles \cite{Guillet1998-hw}. Recently, it was also shown that genetics can explain inter-individual differences in blood counts, with up to 28\% of this variation being attributable to genetics \cite{Vuckovic2020-eh}. Additionally, these parameters also change with age \cite{Kubota1991-at,Mahlknecht2010-mo,Siemons2014-wa}. The automation of \ac{cbc}, together with the amount of possible confounding variables, make \ac{cbc} a tool that is mostly valuable for assessing whether an individual may have a disease or condition, being insufficient for adequately diagnosing most diseases on its own.

\subsubsection{Cytomorphology}

\paragraph{Whole blood slides.} In the case of an abnormality in the \ac{cbc}, it is often required to inspect the morphology of blood cells to further narrow the diagnostic. Indeed, \ac{cbc} can be informative when the absolute and relative proportions of different blood cells are concerned, but provide little information about the morphology of individual cells; this requires the analysis of \ac{wbs} (\figref{fig:whole-blood-slide}). The preparation of \ac{wbs} is generally done by drawing venous blood from an individual, generally from a vein in the antecubital fossa, by a trained professional with adequate patient consent. An anticoagulant is used to allow for the preservation of blood in liquid form and clear inspection of individual cells under a microscope --- the anticoagulant of choice is generally an \ac{edta} salt, due to the fact that calcium is essential for blood coagulation and that \ac{edta} has strong cation-binding properties \cite{Oh2007-yr}. Unfortunately, much like any other coagulant, \ac{edta} has its caveats, being that, after the addition of \ac{edta}, the \ac{wbs} should be prepared as soon as possible to avoid cytomorphological artefacts \cite{Oh2007-yr,Bain2014-oc}. Adequate blood mixing should be ensured prior to the preparation of a \ac{wbs} \cite{Ashenden2012-eu}. Next, blood can be placed on a microscope slide and manually spread or smeared (which is why \ac{wbs} are also known as ”blood smears”). Following air drying, \ac{wbs} should be fixed in methanol with adequate care \cite{Bain2014-oc} and stained. Staining, while not being particularly consistent between laboratories, is generally based on the Romanowsky stain, consisting of old methylene blue and eosin and developed to stain the nucleus of a malarial parasite in purple and the cytoplasm in blue \cite{England1976-ff}. In the United Kingdom, the most commonly used stains are the May-Grünwald-Giemsa stain, a combination of the May-Grünwald stain (which is also quite similar to the earlier Jenner stain formulation) \cite{Krafts2011-xh} with the Giemsa stain \cite{Giemsa1904-dm}, both a modification of the original Romanowsky stain \cite{Bain2014-oc}. In essence, the colours that can be expected from stains derived from the Romanowsky stain are purple for chromatin and nuclei, light purple for neutrophilic granules, deep purple for basophilic granules, orange for eosinophilic granules, blue-grey for monocytic cytoplasm and deep-blue for RNA-rich cytoplasm (as is the case in the basophils) \cite{Bain2014-oc}.

\begin{figure}[!ht]
	\placefigure{intro/whole-blood-slide}
	\placecaption{Image of a whole blood slide.}
	\label{fig:whole-blood-slide}
\end{figure}

\paragraph{Morphological taxonomy.} Having considered the preparation of \ac{wbs} and being aware of a few possible differences in this process, it is important to address their object of analysis - the cytomorphology of circulating blood cells. As mentioned in Section 1.1, the variety of mature blood cells that are generated by HSC is considerable and complex, quickly becoming intractable when considering all possible blood cell phenotypes. Additionally, the main purpose of this is to contextualise the importance of their cytomorphology in diagnosis. Here, I present a hierarchical taxonomy of blood cell types that is simplified but sufficient for the work covered in this dissertation (\figref{fig:blood-cell-examples}). I present this below, considering 4 relevant \ac{wbc} types (one of which is further subdivided) which are common in normal individuals (details on observable abnormalities are considered when discussing \ac{mds} in Section 1.6):
\begin{enumerate}
    \item{Granulocytes} --- granulocytes are a subtype of \ac{wbc} best characterised as having observable and characteristic granules in the cytoplasm \cite{Web2009-xt}. When considering only blood-circulating \ac{wbc} we can further divided granulocytes into neutrophils, eosinophils and basophils:
    \begin{enumerate}
        \item{Neutrophils} --- a granulocyte and the most abundant \ac{wbc} found in peripheral blood, constituting between 50-70\% of all circulating \ac{wbc} \cite{Actor2012-wq}. They are best characterised as measuring 12-15\Mum, a pink cytoplasm (attributable to the fine and invisible secondary granules), the existence of small, visible primary granules, and a nucleus divided into two to five lobes connected by thin filaments, with a small protuberance being observable in some female individuals attributable to chromosome X segregation \cite{Bain2014-oc} (\figref{fig:blood-cell-examples} (top left));
        \item{Eosinophils} --- slightly larger than the neutrophil (measuring 12-17\Mum) and accounting for only 1-5\% of all circulating \ac{wbc} \cite{Simon2007-un}, eosinophiles have their nucleus partitioned into two or three lobes and have a much higher density of granules than neutrophils. The granules in eosinophils are quite different from those found in neutrophils, being more spherical and reddish-pink in colour. When fewer granules are present, the cytoplasm becomes visible, revealing a pale blue cytoplasm \cite{Bain2014-oc} (\figref{fig:blood-cell-examples} (top center left));
        \item{Basophils} --- basophils are the smallest granulocytes (measuring 10 --- 14\Mum) and represented between 0.5-1\% of all circulating \ac{wbc}. The nucleus is often hard to distinguish among the cloud of dark purple granules that carries the bulk of the cytoplasm. The nucleus is divided into more than one lobe \cite{Bain2014-oc} (\figref{fig:blood-cell-examples} (top center right));
    \end{enumerate}
    \item{Lymphocytes} --- lymphocytes constitute 10-15\% of all circulating \ac{wbc} and can be split into smaller (measuring 10 --- 12\Mum) and larger (12 --- 16\Mum) lymphocytes. In the former case their light-blue (basophilic) cytoplasm is barely visible, with the nucleus occupying most of the cell, which is round in shape. Larger lymphocytes are more amorphous and their chromatin is less compact, leading to lightpurple nuclei, and the cytoplasm is more translucent. A few lymphocytes may also display granules in healthy individuals \cite{Bain2014-oc}. While it would be relevant to distinguish between T- and B-cells, these have no differences in regard to their cytomorphology \cite{Cano2013-gu} (\figref{fig:blood-cell-examples} (bottom left));
    \item{Monocytes} --- monocytes are the largest circulating \ac{wbc}, measuring 12 --- 20\Mum. The nuclei in monocytes are purple and their cytoplasm is greyish-blue in colour. On occasion, very fine blue granules may be observable, as well as small vacuoles --- regions, typically one or a group of blobs, of the cytoplasm with no colour \cite{Bain2014-oc} (\figref{fig:blood-cell-examples} (bottom center)).
\end{enumerate}

\begin{figure}[!ht]
	\placefigure{intro/blood-cell-examples}
	\placecaption{Examples of the blood cells considered over this dissertation: white blood cells, divided into granulocytes, which are further divided into neutrophils, eosinophils and basophils, lymphocytes and monocytes, red blood cells and platelets.}
	\label{fig:blood-cell-examples}
\end{figure}

\paragraph{Red blood cells and platelets.} Apart from \ac{wbc}, one expects to find the other, much more abundant type of blood cell when analysing \ac{wbs} --- the \ac{rbc}. \ac{rbc} are, in healthy individuals, round and light pink, with a brighter centre \cite{Bain2014-oc} (\figref{fig:blood-cell-examples} (top right)). This is a characteristic feature of mature \ac{rbc} --- \ac{rbc} are shaped like disks with a concavity with about one third of the total diameter of the cell in their centre, which is known as the central pallor. Due to the decreased thickness of the cell in the central pallor region, light passes more easily, making it look brighter. Finally, when analysing \ac{wbs} one must consider platelets --- these are megakaryocyte fragments and show up as small purple specks across the whole blood slide \cite{Bain2014-oc,Wright1906-af} (\figref{fig:blood-cell-examples} (bottom right)).

Compared with \ac{cbc}, the inspection of \ac{wbs} is far more cumbersome and considerably more subjective with some variance in terms of inter-individual concordance when the identification of specific cell types is concerned, highlighting the role of analysing relatively large collections of blood cells \cite{De_Swart2017-wc}. However, it permits the extraction of morphological characteristics relevant for clinical practice and diagnosis.

\subsection{Studying alterations to haematopoiesis through blood}

It is important to note that while some alterations to the haematopoietic process can be diagnosed with recourse to the analysis of \ac{cbc} or \ac{wbs}, others require more complex methods. These alterations can be manifold and it would be unwise to cover all of them extensively. I focus specifically on the ones discussed on Chapters 3 and 5. 

Firstly, it is important to consider that alterations to haematopoiesis can be assessed indirectly --- for instance, if a haematologist finds an abnormally large \ac{mcv} this is enough to diagnose macrocytic anaemia (large \ac{rbc} with low levels of haemoglobin). However, macrocytic anaemia is itself a symptom of several different conditions --- vitamin B12 deficiency, folate deficiency, megaloblastic anaemia, alcoholism or \ac{mds} are a few that could be mentioned \cite{Nagao2017-bx}. Conversely, an abnormally small \ac{mcv} is enough to diagnose microcytic anaemia, itself caused by different factors --- iron deficiency, recent blood donation, colonic carcinoma or long-term use of aspirin are but few of the possible causes \cite{Short2013-pz}. These two examples, while detected in the blood and relating to the same fundamental aspect --- the size of \ac{rbc} --- are signs of alterations in the haematopoietic process, but not necessarily to the \ac{hsc} population; indeed, only \ac{mds} could be considered an example of such. More particularly, while the diagnosis of \ac{mds} --- a cancer characterised by an absence of healthy blood cells --- relies heavily on the morphological characterisation of blood cells and is aided by \ac{cbc}, it would be incomplete if no molecular characterisation was available \cite{Invernizzi2015-ob}. Indeed, \ac{mds} and other types of blood cancer require comprehensive diagnostic algorithms that concern not only cytomorphology and \ac{cbc}, but also cytogenetics and molecular characterization as discussed in Section 1.5. From these few but meaningful examples, it becomes clear that studying the clinical manifestations of alterations to the \ac{hsc} population can be complicated if one focuses excessively on the phenotype.

\paragraph{Genetics.} The analysis of \ac{wbs} and \ac{cbc} allow us to study haematopoiesis by studying its product --- circulating blood cells. However, it is clear that different modalities of analysis are useful for different tasks --- while the inspection of cytomorphology in a \ac{wbs} can help the haematologist detect the presence of oncological disease in the \ac{hsc} population, such would be complicated if the acquired mutations do not cause any form of phenotypic alteration. For instance, to this date, there is no association between cytomorphology in \ac{wbs} and genetics in \ac{mds}, with the detection of ring sideroblasts in the bone marrow figuring as one of the few methods that would entail a specific genetic association --- with \ac{sf3b1} --- in \ac{mds}. With this in mind, it is no wonder that the investigation of risk factors became indissociable from the genetic and genomic characterisation of individuals.

\Ac{hsc} accumulate mutations as they self-renew. Sometimes, a subset of \ac{hsc} acquires mutations that confer them a growth advantage, leading to their greater rate of proliferation when compared with the rest of the \ac{hsc} population. These mutations --- known as "driver mutations" --- affect specific parts of the genome and are an essential step in oncogenesis \cite{Stratton2009-id}. Consequently, it often becomes necessary to inspect the acquisition of mutations through whole genome or targeted sequencing. To this effect, circulating blood cells are sequenced since they are representative of the \ac{hsc} population. Indeed, it has been shown that bone marrow and peripheral blood \ac{vaf} have a high degree of correlation for shared variants, indicating that peripheral blood can be used to track the clonal composition of \ac{hsc} \cite{Patkar2018-lp,Tong2015-oq,Coffey2019-rr}. Additionally, it is estimated that the life cycle of granulocytes and circulating monocytes lasts a few days and that of lymphocytes corresponds to a few months \cite{Resegotti1957-ut,Lahoz-Beneytez2016-zq,Patel2017-qx}, thus eliminating the possibility of sampling an outdated clonal landscape of the bone marrow, especially considering that annual variations in the \ac{vaf} of different clones at old age are quite small as shown in Chaper 3. However, this should not imply that the molecular characterisation of mutations is sufficient to study alterations to the \ac{hsc} population; indeed, mutations associated with oncological disorders accumulate and even expand significantly in seemingly healthy individuals, a condition known as \ac{ch} that is associated with later onset of haematological malignancies \cite{Jaiswal2014-rl}. 

\section{Clonal haematopoeisis}

\Ac{ch} is characterized as the non-pathological existence of genetically distinct and well-defined subpopulations --- or clones --- of circulating blood cells and \ac{hsc} \cite{Jan2017-rc}. While it was only recently that \ac{ch} was extensively characterized, it is worth noting that one can come across several ways of detecting and consequently defining \ac{ch}, creating some confusion and different operating definitions of \ac{ch}. \Ac{ch} was first discovered as the non-random phenomenon of \ac{cxi} in elderly women \cite{Busque1996-ek,Champion1997-fj} --- specific patterns of \ac{cxi} were being observed in healthy individuals with no alterations to their blood counts when this was thought to be indicative of \ac{mds}. A few years earlier, the theory that \ac{aml} was caused by a progressive formation of leukaemic/pre-leukaemic clones had already been proposed \cite{Fialkow1991-de} and, together with the evidence that different clones could be detected in elderly individuals, led to the theory of \ac{ch} as a stage of the development of blood cancers. In 2012, studies had been able to associate \ac{cxi} with specific genotypes --- mutations in \ac{tet2}, which I further discuss in this section, were shown to be associated with \ac{cxi} \ac{ch} \cite{Busque2012-oc}. 

It was in the next few years that the operating version of \ac{ch} detection I use for this dissertation was proposed --- \ac{ch} is the presence of subpopulations of blood cells with mutations in genes typically associated with myeloid neoplasms \cite{Jaiswal2014-rl,Genovese2014-eu,Xie2014-np,McKerrell2015-rl}, using either exome sequencing \cite{Xie2014-np,Jaiswal2014-rl,Genovese2014-eu} or targeted sequencing \cite{McKerrell2015-rl}. Other possible markers include large chromosomal rearrangements --- \ac{mca} --- in the cells of healthy individuals \cite{Jacobs2012-cj,Laurie2012-fc,Loh2018-xp,Gao2021-ph}, which can be detected using single nucleotide polymorphism array data and were shown to be associated with specific mutations \cite{Loh2018-xp,Gao2021-ph}. Interestingly, mutations in genes such as \ac{tet2} and \ac{ppm1d} were associated with \ac{mca} in the chromosomes where each gene is, raising the possibility that large structural variants play a role on the acquisition of single nucleotide variants or vice-versa \cite{Gao2021-ph}. This raises important issues for targetted sequencing data, used in the analysis presented in Chapter 3 --- while I assumed that mutations are heterozygous and that the fraction of mutated reads is representative of clone size (the fraction of cells harbouring a mutation), this assumption may not hold in the presence of undetected copy number variations. Additionally, the use of specific markers through the years has led to a large fraction of clones in \ac{ch} remaining undetected \cite{Poon2020-hl,Genovese2014-eu,Zink2017-zi}. Consequently, it should be noted that the study of \ac{ch} is, in reality, the study of only a portion of \ac{ch}. Together with recent work by \etal{Mitchell}, the work I present in Chapter 3 highlights this to some extent --- using single-cell colonies one can observe clearly that most expansions in the haematopoietic system are not associated with previously known driver mutations \cite{Fabre2021-uw,Mitchell2021-zl}. 

The prevalence of \ac{ch} and average clone size increases with age (\ac{ch} is detected in less than 1\% of individuals younger than 40 years of age, but in 10-20\% of individuals older than 70) \cite{Jaiswal2014-rl}, showing specific patterns of mutations accumulation \cite{McKerrell2015-rl} and has been shown to be associated with a number of adverse outcomes, including cancer, cardiovascular disease and death \cite{Jaiswal2014-rl,Saleheen2017-eq,Genovese2014-eu,Desai2018-pj,Abelson2018-wh}. While the link between \ac{ch} and cardiovascular disease is still a target of speculation \cite{Evans2021-ae}, with recent research showing that atherosclerosis accelerates somatic mutation and \ac{ch} \cite{Heyde2021-nl}, its connection with cancer is much better understood and intuitive as they both are processes of somatic mutation and \ac{ch} can be used to predict the onset of \ac{aml} \cite{Desai2018-pj,Abelson2018-wh}. However, it should be noted that not all individuals with \ac{ch} develop haematological cancers and that this is generally associated with specific mutations --- for instance, the \ac{tp53} and \ac{u2af1} contribute much more significantly towards the risk of \ac{aml} onset than the gene encoding \ac{dnmt3a} \cite{Abelson2018-wh}. Additionally, signature analysis performed on passenger mutations in \ac{ch} carriers and non-carriers reveals that there is an enrichment for signatures 4 and 6, caused by smoking and \ac{dna} mismatch repair, respectively, in \ac{ch} carriers.

\subsection{The genetics of clonal haematopoiesis}

The association of different genes with different risks of developing haematological malignancies such as \ac{aml} is certainly one of the most interesting aspects of \ac{ch}, requiring close attention. Considering that during this dissertation I studied \ac{ch} through the analysis of the clonal trajectories of 17 genes under positive selection (an explanation on this is offered in Section 1.4), I provide here short descriptions for each and discuss them in the context of \ac{ch}, referring to the genes in their italic format and to the proteins encoded by these genes in their plain text format unless otherwise noted (for example, gene \textit{AB} encodes protein AB). In \tableref{table:genes}, I introduce these genes and group them regarding their role in cell biology providing their average mutation prevalence per individual and in \figref{fig:gene-proportions-study} I show their relative number of mutations.

\begin{table}[!h]
\centering
\caption{Physiological role of the 17 genes studied in the context of clonal haematopoiesis.}
\pgfplotstabletypeset[
font=\footnotesize,
string type,
columns/name/.style={
    column name=Name,
    column type={C{.3\textwidth}}},
columns/ab/.style={
    column name=Abbreviation,
    postproc cell content/.style={@cell content=\textit{##1}},
    column type={C{.15\textwidth}}},
columns/f/.style={
    column name=Role,
    column type={C{.2\textwidth}}},
columns/p/.style={
    column name=Average proportion (range),
    column type={C{.15\textwidth}}},  
columns/chr/.style={
    column name=Gene location,
    column type={C{.1\textwidth}}},  
% Row/column numbering starts from zero
every head row/.style={before row={\toprule},after row=\midrule},
every last row/.style={after row={\toprule}},
every odd row/.style={before row={\rowcolor[gray]{0.9}}}
]\genes
\label{table:genes}
\end{table}

\begin{figure}
    \placefigure{intro/ch-gene-proportions}
    \placecaption{Number of mutations per individual for the 17 studied CH genes across different cross-sectional studies (left) and the sample size for these studies (right). Left: size, colour and shape reflect the number of mutations per individual, the rank based on the number of mutations per individual (yellow = first, red = last) and clinical status, respectively. Grey squares represent studies where these genes were not covered.}
    \label{fig:gene-proportions-study}
\end{figure}

\subsubsection{Epigenetic modifications}

\noindent \textbf{DNA (cytosine-5)-methyltransferase 3A (\textit{DNMT3A})}

\Ac{dnmt3a} encodes a de novo DNA-methylating protein \cite{Yanagisawa2002-op} which is expressed across tissues and enriched in fetal tissues and liver \cite{Robertson1999-dp}. Particularly, DNMT3A has been shown to be involved in the methylation of lysine-4 in histone H3 \cite{Ooi2007-pu} and of CpG islands in early development \cite{Smallwood2011-to}. Its essential role in neurogenesis has been used to suggest that the promoter-independent activity of DNMT3A may be used to maintain the expression of developmental genes by promoting active chromatin states \cite{Wu2010-ig}. Additionally, and through the interaction with Tet genes, DNMT3A is involved in the process of DNA methylation and demethylation that regulates early body plan formation during gastrulation \cite{Dai2016-kl}. While merely demonstrative, this body of evidence points towards \ac{dnmt3a} as a gene which is essential for normal development. Conversely, \ac{dnmt3a} mutations lead to increased oncological pathogenicity --- indeed, in \ac{aml}, which affects one of the few tissues with stem cell potential, \ac{dnmt3a} mutations are present in over 20\% of all individuals and are associated with increased mortality risk \cite{Ley2010-yk}, and mutations, while not leading to differences in global methylation patterns or expression, lead to aberrant or decreased activity \cite{Yan2011-ug}. Individuals with \ac{mds} are more likely to progress to \ac{aml} if they have a \ac{dnmt3a} mutation \cite{Walter2011-ua}, and \ac{dnmt3a} mutations are almost exclusively early or foundational mutations \cite{Shlush2014-di,Miller2013-cw}. Healthy individuals with \ac{dnmt3a} have a higher likelihood of developing \ac{aml} in the future \cite{Gerstung2017-bm,Abelson2018-wh}, a change that also increases with \ac{dnmt3a} clonality \cite{Desai2018-pj}. 

This body of evidence shows that \ac{dnmt3a}, apart from playing an important role in early development, is also a key player in \ac{aml} and clonal development in \ac{hsc}. However, despite the increased \ac{aml} and mortality risk that is associated with \ac{dnmt3a} mutations, these are also among the most prevalent in \ac{ch}, being the most prevalent across several studies \cite{Jaiswal2014-rl,Genovese2014-eu,Zink2017-zi,Bolton2020-ct,Coombs2017-ph,McKerrell2015-rl,Acuna-Hidalgo2017-ng,Desai2018-pj,Young2016-du,Young2019-rz} (0.6126 mutations per individual on average (0.0098-2.9); \tableref{table:genes}; \figref{fig:gene-proportions-study}). Additionally, \ac{ch} with mutations in \ac{dnmt3a} is associated with increased \ac{rdw} \cite{Dawoud2020-af}, a strong predictor of \ac{aml} onset \cite{Abelson2018-wh}. This is counterintuitive --- if \ac{dnmt3a} increases the risk of haematological cancers, how can it be so prevalent in seemingly healthy individuals? A few studies may elucidate this --- \ac{dnmt3a}, much like most mutations in \ac{aml} is frequently mutated with other genes, particularly \ac{npm1} and \ac{flt3}, with a third of all individuals with \ac{aml} and \ac{dnmt3a} mutations also having these two mutations \cite{noauthor_2013-ti,Bezerra2020-zc}. Other studies have shown that, while the loss of \ac{tet2} alone is capable of inducing lymphoid and myeloid disease, its interaction with the loss of \ac{dnmt3a} further enhances this effect \cite{Zhang2016-ys}. Additionally, recent studies have shown that acute and chronic infection drive clonal expansion of \ac{dnmt3a} clones in \ac{ch} through inflammation and particularly for loss-of-function mutations \cite{Hormaechea_Agulla2019-cd,Hormaechea-Agulla2021-kr}. This points to two distinct aspects of \ac{dnmt3a} mutations --- they probably require other mutations to exert pathogenic consequences and infection-driven inflammation potentiates their expansion.

\noindent \textbf{Tet methylcytosine dioxygenase 2 (\textit{TET2})}

\Ac{tet2} is one of the \ac{tet} enzymes, responsible for the regulation of gene expression patterns through epigenetic changes \cite{Rasmussen2016-rt}, namely through the oxidation of 5-methylcytosine (5-mC) to form 5-hydroxymethylcytosine (5-hmC) in both \ac{dna} \cite{Tahiliani2009-gj,Ito2010-el,Guallar2018-eg} and \ac{rna} \cite{He2016-nm,Fu2014-kw}, and through their interaction and mediation of the spliceosome \cite{Rappsilber2002-ll,Tan2009-vw}, while being involved in other forms of epigenetic regulation \cite{Yang2020-yb}. The role of the TET enzyme family is manifold, playing a relevant role in both early and late embryonic development, totipotency induction and maintenance and differentiation \cite{Yang2020-yb}. The role of \ac{tet2} in tumour suppression has also been confirmed \cite{Wang2015-ih} and its associated role in pathogenicity has also been studied: in paediatric \ac{all}, childhood \ac{mds} and adult myeloid malignancies, low expression of \ac{tet2}, \ac{tet2} loss of function or \ac{tet2} mutation-associated low levels of 5-hmC are associated with poor outcome \cite{Zhang2019-on,Coutinho2015-pb,Liu2013-rf}, and its loss has been shown to be associated with hypermutability \cite{Pan2017-ic}. Additionally, loss of \textit{Tet2} in mice leads to enhanced in vivo HSC self-renewal \cite{Moran-Crusio2011-rm}. \ac{tet2} mutations are present in most myeloid malignancies \cite{Delhommeau2009-xp,Abdel-Wahab2009-ir}, being particularly frequent in \ac{cmml} \cite{Abdel-Wahab2009-ir} and in \ac{mds} and \ac{aml}, where its acquisition is likely to be early \cite{Langemeijer2009-ll,Welch2012-uq,Shlush2014-di}.

\Ac{tet2} constitutes one of the most prevalently mutated genes in \ac{ch} \cite{Jaiswal2014-rl,Genovese2014-eu,Zink2017-zi,Bolton2020-ct,Coombs2017-ph,Acuna-Hidalgo2017-ng,Desai2018-pj,Young2016-du,Young2019-rz} (0.3487 mutations per individual on average (0.0025-16176); \tableref{table:genes}; \figref{fig:gene-proportions-study}) and its mutation prevalence is associated with age \cite{Xie2014-np,Buscarlet2017-vh}. \Ac{tet2} is one of the few mutations in \ac{ch} for which a heritable predisposition has been detected with no \ac{tet2} germline mutations \cite{Buscarlet2017-vh,Bick2020-gk}. \ac{tet2} \ac{ch} have also been shown to be driven by inflammation \cite{Zhang2019-wb,Cai2018-yi}, with \ac{tet2} loss-of-function mutations being associated with increased inflammation \cite{Zhang2015-wy,Cai2018-yi}. Additionally, \ac{tet2} \ac{ch} is associated with increased monocyte counts, decreased eosinophil counts, and increased platelet counts \cite{Buscarlet2017-vh,Dawoud2020-af}. 

\noindent \textbf{ASXL transcriptional regulator 1 (\textit{ASXL1})}

The protein encoded by \ac{asxl1} functions as an epigenetic regulator and is involved in histone modifications \cite{Fisher2003-eh}. It is a part of the polycomb repressive complex (PRC) and plays a determinant role in stem cell biology \cite{Avgustinova2016-uz}. Particularly, it has been found that ASXL1 is a part of the polycomb repressive complex 2 (PRC2), which is responsible for the methylation of histone H3, and polycomb repressive complex 1 (PRC1), responsible for the ubiquitination of histone H2A, leading to transcription repression and to the formation of heterochromatin \cite{Schwartz2013-fg,Bowman2018-cd}. The loss of ASXL1 leads to the complete depletion of H3K27me3, a trimethylated state of histone H3 in lysine 27 promoted by PRC2, which leads to myeloid transformation \cite{Abdel-Wahab2012-xt}. Additionally, it should be noted that mutations in other PRC-associated genes are associated with myeloid malignancies --- this indicates that the leukemogenic capacity of \ac{asxl1} is associated with alterations in histone modification \cite{Haferlach2014-bh,Fujino2020-kc}. Mutations in \ac{asxl1} are also indicative of poor prognosis for individuals with aplastic anaemia \cite{Yoshizato2015-th}.

In individuals with \ac{ch}, mutations in \ac{asxl1} are among the most common \cite{Jaiswal2014-rl,Genovese2014-eu,Zink2017-zi,Bolton2020-ct,Coombs2017-ph,Acuna-Hidalgo2017-ng,Desai2018-pj,Young2016-du,Young2019-rz} (0.0663 mutations per individual on average (0.0020-0.3529); \tableref{table:genes}; \figref{fig:gene-proportions-study}) and are associated with smoking, with 69\% of all individuals with \ac{asxl1} mutations having previous smoking experience \cite{Dawoud2020-af}. 

\noindent \textbf{CCCTC-binding factor (\textit{CTCF})}

\Ac{ctcf} is a gene that encodes a protein responsible for delimiting regulatory domains \cite{Jeong2004-xv}. In other words, it acts as a transcriptional insulator by regulating chromatin architecture in a \ac{dna}-methylation-dependent way \cite{Wang2012-gj}, ensuring that the activity of enhancer or silencer elements is restricted to bounded regions of the genome \cite{Kuhn2003-zj}. CTCF is composed of 11 zinc fingers, tertiary structures which have evolved to recognise \ac{dna}, showing a high level of conservation across higher vertebrates \cite{Filippova1996-lz}. However and despite its remarkably high conservation across species, there are few but important human disease-associated conserved CTCF-binding sites across species \cite{Martin2011-iz} or tissues within the same species \cite{Azazi2020-lr}. In normal haematopoiesis, CTCF has been shown to be essential for the maintenance of the \ac{hsc} population \cite{Kim2017-of}, whereas where dynamic CTCF-binding sites --- CTCF-binding sites which change in specific contexts --- are essential for specific lineages and their disruption leads to abnormal cell differentiation \cite{Qi2021-do}. In the context of \ac{aml}, there is a mounting body of evidence showing that CTCF-binding is altered in the context of this disease, but few mechanistic explanations available \cite{Qiu2020-wn}.

In \ac{ch}, \ac{ctcf} is rarely among the mutated genes and observed in cases of therapy-related \ac{ch} \cite{Jaiswal2014-rl,Bolton2020-ct,Coombs2017-ph} (0.0003 mutations per individual on average (0.0000-0.0009); \tableref{table:genes}; \figref{fig:gene-proportions-study}).

\subsubsection{Cell cycle regulation}

\noindent \textbf{KRAS proto-oncogene GTPase (\textit{KRAS})}

The \ac{kras} gene encodes a protein that is a part of the Ras family of proteins and is essential in normal tissue signaling, participating in pathways that are associated with cell development, differentiation and proliferation \cite{Kranenburg2005-sl} and its activity and cellular localization is modulated by the protein kinase C family \cite{Bivona2006-jq} and by phosphodiesterase delta \cite{Chandra2011-ij}, whose activity increases Ras family signaling by concentrating these proteins in the cell membrane. Mutations in \ac{kras} have been shown to be associated with glucose deprivation and, while they do not lead to proliferation, they can trigger apoptosis resistance \cite{Yun2009-an}. In tumours, \ac{kras} is one of the most mutated genes \cite{Kranenburg2005-sl} and has been shown to depend on both nuclear factor kappa B and TANK-binding kinase 1 signalling, both of which are involved in modulating inflammation and immune response \cite{Meylan2009-nc,Barbie2009-hz}. In haematological malignancies, \ac{kras} mutations are associated with poorer survival in \ac{aml} \cite{Ball2019-cs} and in specific subtypes of \ac{cml} (where there is also an enrichment of \ac{kras} mutations) \cite{Vendramini2019-oh}. Additionally, \ac{kras} mutations are also indicative of more aggressive cases of lymphoid neoplasms \cite{Neri1988-xj} and are often subclonal and present upon lymphoid neoplasm relapse \cite{Oshima2016-vy,Malinowska-Ozdowy2015-bv}.

Taking such associations into account, \ac{kras} becomes an obvious candidate for further inspection in \ac{ch}. However, there is little to be said --- \ac{kras}, while detectable in cases of \ac{ch}, shows relatively low prevalence \cite{Jaiswal2014-rl,Genovese2014-eu,Zink2017-zi,Bolton2020-ct,Coombs2017-ph,McKerrell2015-rl,Acuna-Hidalgo2017-ng,Young2016-du,Young2019-rz} (0.0314 mutations per individual on average (0.0000-0.2500); \tableref{table:genes}; \figref{fig:gene-proportions-study}). Interestingly, in individuals with solid tumors, mutations in \ac{kras} were amongst the most common \ac{ch} clones \cite{Conces2019-yq,Wang2019-pl}.

\noindent \textbf{Janus kinase 2 (\textit{JAK2})}

\Ac{jak2} encodes a tyrosine kinase involved in cytokine receptor signaling, whose expression is significantly enriched in circulating \ac{wbc}, among other tissues \cite{Saltzman1998-wi}. JAK2 binds to the growth hormone receptor (GHR), forming a dimer which forms a larger complex with a second JAK2-GHR complex. In its inactivated state, this complex inactivates both JAK2 proteins by having the pseudokinase domain of one JAK2 inhibiting the other and its converse. The activation of GHR triggers conformational changes which physically separate both JAK2 proteins and, consequently, activate them by preventing their inhibition by the pseudokinase domain in the other JAK2 protein. Next, both JAK2 phosphorylate each other on Y1007 \cite{Brooks2014-fh}. This leads to the phosphorylation of other proteins --- namely from the STAT family --- and activates the transcription of target genes whose deregulation leads to oncogenesis \cite{Brooks2010-dc}. In haematological cancer, \ac{jak2} mutations --- particularly V617F --- are typical of cases of \ac{mpn} with subtype-specific prevalence. For instance, in \ac{pv} (a \ac{mpn} subtype), characterised by an abundance of \ac{rbc}, \ac{jak2} mutations are the main driving factor, and \ac{jak2}-V617F shows a strong association with 9p uniparental disomy acquisition (which leads to allele homozygosity while maintaining a normal number of chromosomes) \cite{Wang2014-qz}. In \ac{et}, a subtype associated with higher platelet counts, and \ac{imf} \ac{jak2}-V617F mutations are present in 50\% of the cases \cite{Constantinescu2008-cf}. Importantly, a \ac{jak2}-V617F mutation alone drives \ac{pv} and \ac{imf} phenotypes in mouse bone marrow. The important aspect of \ac{jak2}-V617F is that it leads to constitutive activation of JAK2 --- Y1007 is phosphorylated much more frequently, leading to enhanced STAT-JAK2 pathway activity \cite{Feng1997-jr,Constantinescu2008-cf}

Because of this distinct association with \ac{mpn}, \ac{jak2} mutations are comparatively well studied in \ac{ch}. However, it should be noted that the association of \ac{mpn} with \ac{jak2} mutations does not make the presence of \ac{jak2} mutations sufficient for the development of \ac{mpn} --- indeed, only 5\% of individuals with a \ac{jak2} mutation have a \ac{mpn} diagnosis, a diagnostic which becomes more likely as the mutated \ac{jak2} \ac{vaf} increases \cite{Cordua2019-mo}. \ac{jak2} mutations in \ac{ch} have specific associations with platelet counts, \ac{hb} concentration, hematocrit and leukocyte, neutrophil, thrombocyte and lymphocyte counts \cite{Dawoud2020-af,Cordua2019-mo} and can be further stratified based on \ac{hb} concentration, \ac{wbcc} and platelet counts for \ac{mpn} diagnosis \cite{Sochacki2019-mu,Dawoud2020-af}. Additionally, \ac{jak2}-V617F \ac{ch} leads to cardiovascular disease through proinflammatory mechanisms in murines and artery endothelium degradation in mice \cite{Sano2019-hm,Molinaro_Roberto2020-hd}. In \ac{ch}, \ac{jak2} mutations are present in relatively few individuals \cite{Jaiswal2014-rl,Genovese2014-eu,Zink2017-zi,Bolton2020-ct,Coombs2017-ph,McKerrell2015-rl,Acuna-Hidalgo2017-ng,Desai2018-pj,Young2016-du,Young2019-rz} (0.0176 mutations per individual on average (0.0000-0.1176); \tableref{table:genes}; \figref{fig:gene-proportions-study}).

\noindent \textbf{Protein tyrosine phosphatase non-receptor type 11 (\textit{PTPN11})}

\Ac{ptpn11} is an oncogene that encodes Src homology 2 domain-containing protein tyrosine phosphatase 2 (SHP2), a soluble protein tyrosine phosphatase \cite{Chan2007-xf}. Its function is mostly associated with its role as a phosphatase in the signal transduction of extracellular molecules such as growth factors and cytokines \cite{Chan2008-zh}, but SHP2 has also been shown to act as a scaffolding protein \cite{Bennett1994-wa,Li1994-xn} and it plays both catalytic and non-catalytic roles in triggering apoptosis as a response to DNA damage \cite{Yuan2005-xt}. Germline \ac{ptpn11} mutations are strongly associated with two syndromes which are associated with an increase cancer risk --- Noonan syndrome, specifically associated with an increased predisposition to haematological cancers \cite{Tartaglia2001-ru},  and LEOPARD syndrome, associated with a general increase in cancer risk \cite{Digilio2002-nq}, even though both are characterised by different catalytic changes --- whereas Noonan syndrome is characterised by activating mutations, LEOPARD syndrome is characterised by mutations which negatively affect the catalytic capabilities of SHP2 \cite{Kontaridis2006-ia}. In haematological cancers, several links have been found between \ac{ptpn11} mutations and juvenile cases of B-cell lymphocytic leukaemia \cite{Tartaglia2004-lc}, pediatric AML (where \ac{ptpn11} mutations are associated with specific cellular morphology but not with changes in mortality risk) \cite{Loh2004-fg}, and \ac{jmml}, where a third of cases have \ac{ptpn11} mutations \cite{Tartaglia2003-xw,Kratz2005-qh}, as well as in 8\% of all patients with AML \cite{Fobare2020-et}, where it is closely associated with \ac{npm1} mutations.

Interestingly, \ac{ptpn11} mutations are absent in most \ac{ch} studies \cite{Jaiswal2014-rl,Genovese2014-eu,Zink2017-zi,Bolton2020-ct,Coombs2017-ph,Acuna-Hidalgo2017-ng,Young2016-du,Young2019-rz} (0.0067 mutations per individual on average (0.0000-0.0294); \tableref{table:genes}; \figref{fig:gene-proportions-study}), further highlighting their prevalent role in juvenile and pediatric oncological diseases and in a few cases of adult AML.

\subsubsection{DNA damage regulation}

\noindent \textbf{Tumor protein 53 (\textit{TP53})}

\Ac{tp53} encodes a transcription factor that also acts as a tumor suppressor gene \cite{Surget2013-xh}, responding to a wide variety of insults to the cell such as DNA damage, hypoxia or oncogene activation \cite{Oren2003-gg} with at least twelve known isoforms across several tissues \cite{Khoury2011-yk,Bourdon2005-ne}. The pathways involving p53, the protein encoded by \ac{tp53}, are notably complex, with different p53 isoforms and closely related proteins such as p73 and p63 controlling the transcription of specific p53 isoforms \cite{Bourdon2005-ne}. Additionally, the ubiquitousness of p53 --- expressed in every cell excluding erythrocytes --- does not represent a universal function; indeed, its function is dependent on the tissue where it is expressed, maintaining nonetheless a universal role as a tumour suppressor \cite{Surget2013-xh}. To further complicate the severity of \ac{tp53} mutations, p53 can also be differentially expressed rather than mutated to lead to worse or better prognosis in different cancer types \cite{Bourdon2011-hc,Song2009-oq,Surget2013-xh}. In myeloid cancers specifically, it is known that \ac{tp53} mutations can be found in 5-10\% of de novo \ac{aml} cases, but in about half of \ac{aml} cases with a complex karyotype, which is generally associated with poor outcomes \cite{Hou2015-jz,Rucker2012-oj,Bowen2009-ep}. Therapy-related cases of \ac{mds} are enriched in \ac{tp53} mutations \cite{Wong2015-ki}. Additionally, multiple p53 mutations, indicative of biallelic targeting, are associated with a worse prognosis in and characterise clinically distinct subtypes of \ac{mds} \cite{Bernard2020-fj}. Lastly, very large clones harbouring \ac{tp53} mutations are associated with a worse prognosis in \ac{aml} patients \cite{Short2020-qu}. As mentioned before, the function of p53 in myeloid malignancies is further affected by p53 pathway dysfunctions are highly common in \ac{aml} \cite{Quintas-Cardama2017-tj,Abramowitz2017-lg} independent from \ac{tp53} mutation status.

In \ac{ch}, \ac{tp53} is detected in a modest proportion of individuals \cite{Jaiswal2014-rl,Genovese2014-eu,Zink2017-zi,Bolton2020-ct,Coombs2017-ph,McKerrell2015-rl,Acuna-Hidalgo2017-ng,Desai2018-pj,Young2016-du,Young2019-rz} (0.0723 mutations per individual on average (0.0000-0.3); \tableref{table:genes}; \figref{fig:gene-proportions-study}), being, together with \ac{ppm1d}, expectedly more prevalent in \ac{ch} cases that result from therapy \cite{Zehir2017-gh}.

\noindent \textbf{Protein phosphatase Mg\textsuperscript{2+}/Mn\textsuperscript{2+}-dependent 1D (\textit{PPM1D})}

\Ac{ppm1d} encodes a nuclear phosphatase and its overexpression is cytostatic, an effect which depends on its dephosphorylation activity \cite{Fiscella1997-ay}. This nuclear phosphatase is responsible for dephosphorylating the active forms of Checkpoint kinase 1 and p53 after \ac{dna} repair pathways have been triggered --- this ensures that the cell is able to return to homeostasis \cite{Lu2005-it}. Intuitively, the overexpression of PPM1D leads to oncogenic phenotypes \cite{Li2002-dl}, specifically by the suppression of p53 activity \cite{Bulavin2002-ft}. \ac{ppm1d} mutations are usually truncating and activating and are especially prevalent in ovarian and breast cancers \cite{Bolton2020-ct}, with cancers that have multiple \ac{ppm1d} copies or that overexpress PPM1D are associated with a poorer prognosis \cite{Lambros2010-yo,Castellino2008-wr,Yang2015-kz}. PPM1D overexpression and activating mutations are also present in a large number of \ac{mpn} and are thought to lead to \ac{mpn} progression \cite{Marcellino2019-ze}. 

In \ac{ch}, \ac{ppm1d} mutations are highly associated with therapy-related \ac{ch} \cite{Genovese2014-eu,Zink2017-zi,Bolton2020-ct,Coombs2017-ph,Desai2018-pj} (0.0153 mutations per individual on average (0.0000-0.0482); \tableref{table:genes}; \figref{fig:gene-proportions-study}), an effect that is thought to be driven by the ability of \ac{ppm1d} mutant cells to thrive in cytotoxic environments \cite{Hsu2018-ta}. Additionally, \ac{ppm1d} \ac{ch} is associated with increased monocyte counts \cite{Dawoud2020-af}.

\noindent \textbf{BRCA1/BRCA2-containing complex subunit 3 (\textit{BRCC3})}

\ac{brcc3} encodes a deubiquitinating enzyme responsible for the deubiquitination of lysine-63 (K63) \cite{Dong2003-ei}, a residue whose ubiquitination is typically associated with altered protein function \cite{Meyer2020-yq}. BRCC3 plays a crucial role in the repair of double strand breaks and mutations in \ac{brcc3} or \ac{brcc3} loss has been shown to be associated with impaired \ac{dna} repair \cite{Chen2006-ub}. Additionally, loss of BRCC3 deregulates the activity of the inflammasome (oligomeric protein complexes that trigger inflammatory responses \cite{Broz2016-zn}) \cite{Py2013-wr}. BRCC3 mutations are more typically associated with \ac{mds} \cite{Huang2015-id} rather than with \ac{aml} \cite{noauthor_2013-ti}, but specific \ac{aml} subtypes show a higher prevalence of \ac{brcc3} mutations, which are also indicative of a better prognosis \cite{Jahn2017-dh,Meyer2020-yq}.

In the context of \ac{ch}, \ac{brcc3} is not particularly interesting despite its association with \ac{mds}. It is usally present in small numbers in \ac{ch} studies that consider it \cite{Jaiswal2014-rl,Genovese2014-eu,Acuna-Hidalgo2017-ng} (0.0003 mutations per individual on average (0-0.0005); \tableref{table:genes}; \figref{fig:gene-proportions-study}). 

\subsubsection{Splicing factors}

\noindent \textbf{Splicing factor 3b subunit 1 (\textit{SF3B1})}

\Ac{sf3b1} expresses subunit 1 of the splificing factor complex 3B, a constitutive part of the larger complex of the U2 small nuclear ribonucleoproteins which participates in pre-mRNA splicing \cite{Fica2013-vn,Kramer1995-mw,Isono2001-yf} through branchpoint recognition \cite{Gozani1998-ox}. Mutations in \ac{sf3b1} are associated with the repression of BRD9, a tumor suppressor associated with chromatin remodelling, possibly by altering its role in branchpoint recognition \cite{Inoue2019-ra}. \ac{sf3b1} mutations are relatively uncommon in most haematological cancers except \ac{mds} --- where they are associated with a specific \ac{mds} subtype \cite{Papaemmanuil2011-nr,Hellstrom_Lindberg2015-zs} --- and \ac{cll}, where they lead to distinctively altered pre-mRNA splicing and were associated with poorer survival \cite{Wang2011-wu,Quesada2011-xc}. Importantly, in \ac{mds}, a high frequency of mutations in the splicing machinery (which includes \ac{sf3b1}, \textit{SRSF2} and \textit{U2AF1} is thought to happen in the first few stages of the disease \cite{Haferlach2014-bh,Papaemmanuil2013-yt,Nagata2019-hn}) \cite{Yoshida2011-zp}. More specifically, two hotspots constitute frequently mutated \ac{sf3b1} mutants --- K666N, which is associated with increased risk of mortality and specific pre-mRNA splicing alterations \cite{Dalton2020-mg}, and K700E, which is also associated with altered pre-mRNA splicing and leads to impaired erythropoiesis \cite{Obeng2016-bw}.

In \ac{ch}, SF3B1 mutations are found at low prevalence \cite{Jaiswal2014-rl,Genovese2014-eu,Zink2017-zi,Bolton2020-ct,Coombs2017-ph,McKerrell2015-rl,Acuna-Hidalgo2017-ng,Desai2018-pj,Young2016-du,Young2019-rz} (0.0211 mutations per individual on average (0-0.1471); \tableref{table:genes}; \figref{fig:gene-proportions-study}) and show a strong positive association with age \cite{Bolton2020-ct,Xie2014-np,McKerrell2015-rl}. 

\noindent \textbf{Serine and arginine rich splicing factor 2 (\textit{SRSF2})}

\Ac{srsf2} encodes a splicing enzyme involved in splicing and spliceosome assembly \cite{Fu1992-wy} and is involved in the recognition of specific mRNA sites \cite{Liu2000-ax} and also in transcriptional elongation \cite{Lin2008-tj}. In human disease, \ac{srsf2} mutations play a role that is typically associated with alterations in splicing --- indeed, this is observed in \ac{aml} \cite{Yoshimi2019-ax} and in most \ac{mds} \cite{Yoshida2011-zp,Makishima2012-wr}, especially \ac{cmml} \cite{Patnaik2013-cv,Meggendorfer2012-jn}. Alterations in splicing are also observable in \ac{srsf2}-mutant \ac{mds} \cite{Rahman2020-bh}. \ac{srsf2} mutations, together with \textit{U2AF1} mutations, are remarkably absent in childhood \ac{mds} and \ac{jmml} \cite{Hirabayashi2012-jj,Kar2013-nw}, a finding that is important once one considers the strong association between splicing machinery \ac{ch} and age \cite{Yoshida2011-zp}. In all cases, mutations in \ac{srsf2}-P95 are the most common, with \ac{srsf2}-P95H being the one having a much more preponderant role in the progression to \ac{mds}, \ac{aml} and \ac{mpn} \cite{Wang2021-wg,Smeets2018-yv,Nagata2019-hn}. Unsurprisingly, \ac{srsf2}-P95H also leads to the most noticeable changes in pre-mRNA binding of all \ac{srsf2} mutations \cite{Zhang2015-xu,Komeno2015-pq}.

Similarly to \ac{sf3b1}, \ac{srsf2} mutations are present in few individuals with \ac{ch} \cite{Jaiswal2014-rl,Genovese2014-eu,Zink2017-zi,Bolton2020-ct,Coombs2017-ph,McKerrell2015-rl,Acuna-Hidalgo2017-ng,Desai2018-pj,Young2016-du,Young2019-rz} (0.0155 mutations per individual on average (0-0.0882); \tableref{table:genes}; \figref{fig:gene-proportions-study}) despite showing a strong association with age and being detectable only in older individuals \cite{Bolton2020-ct,McKerrell2015-rl}.

\noindent \textbf{U2 small nuclear RNA auxiliary factor 1 (\textit{U2AF1})}

\Ac{u2af1} synthesises one of the two subunits responsible for the formation of the U2AF splicing factor --- U2AF1 and U2AF2 \cite{Palangat2019-nv}. It has important roles in normal haematopoiesis, with competitive transplantation assays in mice showing that \textit{U2af1}-deficient cells are unable to compete with wildtype \textit{U2af1} cells and transcriptome analysis revealing downregulation of \ac{hsc} maintenance genes and upregulation of genes associated with DNA damage and apoptosis in U2af1-deficient mice \cite{Dutta2020-ah}. Studies using human cells have also shown that wildtype U2AF1 acts as a non-canonical translational repressor through the interaction of the U2AF heteromer with mRNA and that a specific mutation, S34F, leads to increased IL-8 secretion, which is increased in cases of relapsed/refractory \ac{aml} \cite{Palangat2019-nv}, a haematological cancer where spliceosome mutations are common \cite{Papaemmanuil2016-jy}. Interestingly, it has also been shown that, despite the presence of a \ac{u2af1} mutation, a wildtype copy is always required to ensure cell survival \cite{Fei2016-cc,Wadugu2020-df}, making \ac{u2af1} mutations always monoallelic. In haematological malignancies, spliceosome mutations are relatively common in \ac{mds} \cite{Yoshida2011-zp}, with U2AF1 mutations being one of the most prevalent in \ac{mds} --- especially in younger cases of \ac{mds} --- and associated with increased mortality risk despite not being particularly represented in higher or lower risk \ac{mds} \cite{Li2018-vz}. \ac{u2af1} mutations are also associated with poor outcomes after \ac{aml} with a first complete remission, increasing both the risk of death and relapse \cite{Saygin2018-bb}, affecting the splicing of genes which, among others, are typically mutated in haematological malignancies \cite{Przychodzen2013-vb} and lead to lineage specific alterations in splicing \cite{Yip2017-kf}. 

It should be noted that the large majority of the aforementioned studies on \ac{u2af1} mutations focus on the S34F hotspot. However, as shown in Chapter 3, it is another hotspot, Q157, that is the most prevalent in \ac{ch}. It is interesting that this bias is observable across several studies, especially when both hotspots are prevalent in equal proportions and are located in the same tertiary structure responsible for binding to RNA \cite{Adema2016-fa}. In \ac{ch}, \ac{u2af1} mutations are generally present in relatively few cases or completely absent \cite{Jaiswal2014-rl,Genovese2014-eu,Zink2017-zi,Bolton2020-ct,Coombs2017-ph,Acuna-Hidalgo2017-ng,Desai2018-pj,Young2016-du,Young2019-rz} (0.0127 mutations per individual on average (0-0.0882); \tableref{table:genes}; \figref{fig:gene-proportions-study}), showing an association, as is the case with other splicing genes, with progression to therapy-related myeloid neoplasms while mutation prevalence shows no association with prior therapy. This points towards splicing genes as facilitators of this specific type of myeloid neoplasms \cite{Bolton2020-ct}. 

\subsubsection{Glucose metabolism}

\noindent \textbf{Isocitrate dehydrogenase (NADP+) 1 and 2 (\textit{IDH1} and \textit{IDH2})}

\Ac{idh1} and \ac{idh2} are the genes responsible for encoding NADP+-dependent enzymes responsible for decarboxylating isocitrate into alpha-ketoglutarate \cite{Nekrutenko1998-ld}, an important step in the citric acid cycle which produces NADPH. In the liver, the activity and expression of IDH1 is correlated with that of NADPH-dependent processes, namely fatty acid and cholesterol synthesis \cite{Shechter2003-vd}. Two mutations are of particular relevance to the study of IDH1/2 in disease --- R132 in \ac{idh1} and R140 in \ac{idh2}. These mutations frequently co-occur in \ac{aml}, with mutations in \ac{npm1} or \ac{dnmt3a} and using them to stratify \ac{aml} patients leads to a decrease in overall survival when compared with other \textit{IDH1/2} mutations \cite{Meggendorfer2018-me,Mardis2009-gl}. The transformation capability of these mutations has to do with alterations in their metabolic products --- both mutations, while maintaining the NADPH production, lead to the production of 2-hydroxyglutarate from alpha-ketoglutarate \cite{Ward2010-fi}. 2-hydroxyglutarate inhibits H3K9 demethylase KDM4C, an enzyme responsible for histone H3 demethylation and blocks differentiation \cite{Lu2012-ix}. Consequently, a mutation which seemingly affects only the citric acid cycle, is responsible for halting cell differentiation and promoting transformation. Clones with mutations in \textit{IDH1/2} are able to reconstitute in \ac{aml} patients which are treated with chemotherapy without this implying an increased disease risk in the future \cite{Wiseman2016-kw}. Together with their frequent co-clonality, mutations in \textit{IDH1/2} are unlikely to be the sole drivers of \ac{aml} \cite{Hasserjian2020-qj}.

In \ac{ch}, \textit{IDH1/2} mutations are present in very small numbers or completely absent \cite{Jaiswal2014-rl,Genovese2014-eu,Zink2017-zi,Bolton2020-ct,Coombs2017-ph,McKerrell2015-rl,Acuna-Hidalgo2017-ng,Desai2018-pj,Young2016-du,Young2019-rz} (0.0132 mutations per individual on average (0-0.0882) and 0.0039 mutations per individual on average (0-0.0294) for \ac{idh2} and \ac{idh1}, respectively; \tableref{table:genes}; \figref{fig:gene-proportions-study}). 

\subsubsection{G-protein coupled receptor-mediated signaling}

\noindent \textbf{G-protein subunit beta 1 (\textit{GNB1})}

\Ac{gnb1} encodes the beta-subunit of heterotrimeric guanine nucleotide-binding proteins (G proteins), a set of intracellular proteins which transduce extracellular signals received by G protein coupled receptors \cite{Hurowitz2000-nk}. The complex formed by the beta-subunit and the gamma-subunit of G proteins acts as a repressor of the alpha-subunit by decreasing its affinity for guanosine diphosphate. In haematological cancers, mutations in \ac{gnb1} can show lineage specificity --- whereas K57 mutations have been found exclusively in myeloid neoplasms, I80 mutations are found almost exclusively in B-cell neoplasms. \Ac{gnb1} mutations are also concentrated on the interface of GNB1 with the alpha-subunit \cite{Yoda2015-eq}, thus making it likely that \ac{gnb1} mutations lead to constitutive alterations to G protein-mediated cell signaling. 

In \ac{ch}, \ac{gnb1} is prevalent at relatively low frequencies \cite{Jaiswal2014-rl,Genovese2014-eu,Zink2017-zi,,Acuna-Hidalgo2017-ng} (0.0009 mutations per individual on average (0-0.002); \tableref{table:genes}; \figref{fig:gene-proportions-study}).

\subsubsection{Protein degradation}

\noindent \textbf{Cbl proto-oncogene (\textit{CBL})}

CBL acts as a positive regulator of receptor protein-tyrosine kinase ubiquitination, thus being involved in their degradation. By doing so it contributes to the termination of the signaling triggered by these receptors \cite{Joazeiro1999-do}. More specifically, CBL binds to tyrosine-phosphorylated substrates through its Src homology 2 domain (SH2) and recruits ubiquitinating enzymes through its Really Interesting New Gene (RING) domain \cite{Thien2001-wm}. In myeloid neoplasms, \ac{cbl} mutations are detected later during disease progression and are associated with transformation \cite{Valent2019-bx} except in \ac{cmml} \cite{Schnittger2009-kw} and especially in \ac{jmml} \cite{Loh2009-li,Muramatsu2010-ec}, while also being associated with 11q uniparental disomy in myeloid neoplasms \cite{Sanada2009-jh}.

In \ac{ch}, \ac{cbl} is relatively common \cite{Jaiswal2014-rl,Genovese2014-eu,Zink2017-zi,Bolton2020-ct,Coombs2017-ph,Acuna-Hidalgo2017-ng,Young2016-du,Young2019-rz} (0.0493 mutations per individual on average (0.0002-0.1765); \tableref{table:genes}; \figref{fig:gene-proportions-study}).

\subsubsection{Other determinants and features of \ac{ch}}

The aforementioned genes constitute less than half of the genes that would generally be studied in \ac{ch}. Nonetheless, the examples provided are representative --- indeed, the differential and oftentimes age-dependent prevalence of genes paints a landscape that is complex enough. This complexity, however, cannot prevent us from drawing a few remarks that allow us to better draw the patterns of \ac{ch}.

\paragraph{The age-dependency of spliceosome mutations.} \Ac{u2af1}, \ac{srsf2} and \ac{sf3b1} are commonly associated with more advanced age. From this, two theories can be posited --- either mutations in these genes appear at a much later stage in life or they appear consistently through life but are only able to expand in older individuals due to specific stimuli such as that of inflammaging \cite{Franceschi2018-be} or specific insults. In either case there is a specific bias contributing towards the late appearance of these clones. While it could also be possible that these clones grow consistently through life at a very slow rate, this is highly unlikely --- while being able to detect clones with a \ac{vaf} as small as 2\%, \etal{Jaiswal} where only able to detect one larger than 6\% \cite{Jaiswal2014-rl}, whereas another study, while being able to detect clones as small as 1\%, could only find \ac{u2af1} at a \ac{vaf} of at least 5\% \cite{Zehir2017-gh}. These data, while somewhat anecdotal, point towards a late expansion of \ac{u2af1}. Other data also supports this evidence for specific spliceosome mutations --- \ac{u2af1}-Q157 and \ac{srsf2}-P95H are largely absent during childhood and early adulthood and remain practically undetected in cases of childhood \ac{mds} and juvenile \ac{cmml} \cite{Kar2013-nw,Hirabayashi2012-jj}, as is the case with other splicing-associated mutations such as \ac{sf3b1} \cite{Lin2014-qr};

\paragraph{The impact of life events in \ac{ch}.} Cancer therapy can lead to increased \ac{ch} prevalence as shown in a few studies \cite{Bolton2020-ct,Zehir2017-gh}; indeed, mutations in genes such as \ac{ppm1d} and \ac{tp53} are more likely to appear in blood after cancer therapy. \Ac{asxl1}, on the other hand, shows a distinct association with smoking \cite{Dawoud2020-af}, while \ac{dnmt3a} and \ac{tet2} \ac{ch} expand more quickly in cases of chronic infection and inflammation, respectively \cite{Hormaechea-Agulla2021-kr,Cook2020-xs}. What this entails for the work discussed in Chapter 3, which studies the evolutionary dynamics of \ac{ch}, is that life events can have a meaningful impact on \ac{ch}, making the expectation of a perfect extrapolation to the past more complicated. Other mechanisms inherent to a pool of clonal HSC with different fitness such as competition is also likely to make these inferences more complicated \cite{Beerenwinkel2007-us} by leading to the underestimation of fitness effects. It is also worth noting that competition is obfuscated by the fact that a large majority of \ac{ch} still goes by undetected \cite{Poon2020-ek,Zink2017-zi,Genovese2014-eu}. These expansions with no known drivers can be caused by, for example, large-scale chromosomal alterations --- deletions, duplications and copy-neutral loss of heterozygosity --- some of which are germline mutations \cite{Loh2020-vz,Loh2018-xp,Gao2021-ph,Terao2020-dd}.

A final factor that is worth noting is that clinical phenotypes can be associated with mutations in healthy individuals. Indeed, \ac{ch} has been shown to be associated with several distinct blood counts (in some cases with contradictory evidence being presented between studies; \tableref{table:blood-counts-ch}). Importantly, these associations are either observed when studying \ac{ch} in general or when considering specific genes (such as the association between \ac{ch} and \ac{rdw} \cite{Jaiswal2014-rl,Dawoud2020-af} or between \ac{jak2} \ac{ch} and an increase in \ac{wbcc} \cite{Cordua2019-mo}, respectively) or within groups of individuals which progressed to a specific haematological condition (such as the association between \ac{ch} in \ac{aml} progressors and a decrease in \ac{rbc} and increase in \ac{rdw} \cite{Abelson2018-wh}). Furthermore, chromosome Y loss \cite{Dumanski2015-ze} and mutations in genes such as \ac{asxl1} \cite{Dawoud2020-af} can be associated with smoking history. Complex relationships between genetics and phenotype are evident. However, despite the presence of associations between \ac{ch} and clinical phenotypes, this is still poorly understood --- for instance, when there is a clear association (\ac{asxl1} and smoking is perhaps the best example), there may not be any known mechanism.

\begin{table}[!h]
\centering
\caption{Known associations of clonal haematopoiesis with blood indices.}
\pgfplotstabletypeset[
font=\footnotesize,
string type,
columns/p/.style={
    column name=Blood index,
    column type={C{.2\textwidth}}},
columns/i/.style={
    column name=Increase,
    column type={C{.35\textwidth}},
    string replace={nan}{}},
columns/d/.style={
    column name=Decrease,
    column type={C{.35\textwidth}},
    string replace={nan}{}},
every head row/.style={before row={\toprule},after row=\midrule},
every last row/.style={after row={\toprule}},
every odd row/.style={before row={\rowcolor[gray]{0.9}}}
]\bloodCountsCH
\label{table:blood-counts-ch}
\end{table}

As mentioned thus far, \ac{ch} is a mutational process that leads to the formation of distinct clones and can lead to the onset of cancer, a progression that is highly dependent on the mutated genes --- it is necessary to better understand how evolutionary theory can help in explaining these differences.

\section{Evolutionary theory and somatic evolution}

The application of the theory of natural selection as posited by Charles Darwin in the 19th century \cite{Darwin1859-yw} and initially developed by Greek and Islamic scholars before him \cite{noauthor_1898-xh,Zirkle1941-bw} has been studied in cancer development since the 1970s by Peter C. Nowell as he observed that specific karyotypes were much more prevalent than others in individual tumours \cite{Nowell1976-ob}. About 20 years earlier, Nowell and a collaborator --- David A. Hungerford --- were responsible for one of the better known chromosomal abnormalities in chronic myelogenous leukaemia: the Philadelphia chromosome, characterized as a slightly shorter chromosome 22 and an elongated chromosome 9 as a result of translocation \cite{Nowell2007-ux}. However, it was the later observation of intra-tumoural evolution that drove the interest in cancer evolution among molecular biologists \cite{Cannataro2018-ye}. This discovery was followed by another important finding --- that proto-oncogenes (genes that, once mutated, can become tumorigenic) were mutated in healthy cells \cite{Stehelin1976-uu,Shih1982-gu}. It was only a matter of time before genetic models of tumorigenesis were considered --- for instance, Eric R. Fearon and Bert Vogelstein observed a consistent increase in the prevalence of mutant \textit{RAS} and 17p loss as colorectal carcinomas progressed, also noting that the number of chromosomal alterations increased with progression \cite{Fearon1990-rk} and \etal{Califano} observed that in head and neck tumours there was a specific progression of chromosomal arms that could consistently delineate tumour progression \cite{Califano1996-kj}. The interest in genetic models of cancer has not weaned --- indeed, it has incorporated technological developments such as sequential CRISPR, through which \etal{Wang} were able to show a likely sequence of mutation acquisition steps that are more likely to drive myeloid leukaemia \cite{Wang2021-wg} and single-cell colonies, which allowed researchers to identify genetic models of tumorigenesis through phylogenetic reconstruction in acute myeloid leukaemia and myeloproliferative neoplasms, respectively \cite{Morita2020-rq,Williams2020-ji,Van_Egeren2021-tb}. Also recently, it has been shown why mutator sites --- sites which, when mutated, increase the mutation rate --- happen later during cancer progression: if they happen at population sizes which are too small (at early stages of cancer development) they are deleterious due to the promotion of increased genomic instability \cite{Raynes2018-hg}. For instance, mutations in \ac{tp53} which promote genomic instability are rarely founding mutations in colorectal cancer \cite{Kinzler1996-mp} and, despite being highly prevalent in ultraviolet light-exposed skin, \ac{tp53} mutations have a tendency to disappear once exposure is ceassed \cite{Jonason1996-sp}. Considering this complexity and while the earlier genetic models of cancer assumed that a sequential and discrete acquisition of mutations drove tumour progression, with each stage-specific clone overtaking the entire population, the more recent genetic models did a better job at recognizing the influence of more complex subclonal architectures, progressive acquisition of mutations, and the role of competition in tumours \cite{Cannataro2018-ye,Schwartz2017-hg}. Additionally, while the first cancer genetic models were built on the idea of punctuated equilibrium, where one observes a sequence of steady states intercalated with catastrophic events, more recent genetic models are in general more gradualist, incorporating a more continuous vision of cancer evolution. This debate, however, is far from settled \cite{Beerenwinkel2015-xr,Sterelny2007-kp}. 

\subsection{The Wright-Fisher model}

To model evolution, different models are available, one of which is the Wright-Fisher model. The Wright-Fisher model considers a population of diploid $N$ hermaphroditic individuals, each of which is characterized by a set of $P$ genetic positions (loci) which can take on different alleles (different versions of the same genetic position). The Wright-Fisher model defines how this population progresses through generations considering that generations do not overlap and that all individuals give rise to their offspring at exactly the same time through random mating --- an individual on a given generation will have a combination of the alleles present in the previous generation. However, this process of allelic inheritance is not completely random --- the allele counts in a generation are determined by their prevalence in the previous generation. Indeed, considering that two alleles (A and B) exist in a population of diploid $N$ hermaphroditic individuals, the probability of allele A having $k$ counts in the next generation is given by $\binom{2N}{k}p^kq^{2N-k}$, where $p$ and $q$ are the fraction of alleles A and B in the present generation, respectively. In other words, the counts $X$ of allele A are distributed according to a binomial distribution with probability $p$ and $2N$ trials ($X \sim Binom(2N,p)$), making $E[X] = np$ and $Var[X] = npq$ \cite{Gillespie2004-xo}. 

\paragraph{Fitness advantage and selection.} In this scenario, the way that the distributions of alleles $A$ and $B$ evolve is stochastic --- while clearly defined and dependent on the initial prevalence of each allele, there is no way of predicting the long-term outcome. Indeed, either A or B can become fixated in the population (a situation where either B or A disappears completely from the population, respectively). This behaviour is known as stochastic or genetic drift and describes how alleles evolve when no allele is fitter than the other (the probability of either becoming fixated in the population is identical). However, in the circumstance where, despite identical proportions $p$ and $q$, allele A is still more likely to be represented in the following generation, it is said that allele A is being positively selected. This positive selection is parametrized by a fitness advantage $s > 0$ and is used to scale $p$ such that $p' = (1+s)p$, leading to $X' \sim Binom(2N,p(1+s))$ and $E[X'] = np(1+s)$. This has quite simple consequences --- by increasing the probability of the binomial sampling of A counts, the population of clones with A will increase at a greater rate than that of B, leading to the eventual fixation of A in the population and depletion of B. However, it should be noted that the effect of the absolute fitness is affected by the average fitness of the population $\bar{w} = p^2(1+s) + 2pq(1+hs) + q^2(1) = p^2 + p^2s + 2pq + 2pqhs + q^2 = 1 + p^2s + 2pqhs$, following the Hardy-Weinberg law ($1 = p^2 + 2pq + q^2$) and where $h$ is the fraction of heterozygous (carrying both A and B alleles) individuals in the population. Having the average population fitness $\bar{w}$ and the absolute fitness $s$, the relative allele frequency at time $t+1$ is expected to be $p(t+1) = p(t)\frac{1+s}{\bar{w}} = p(t)\frac{1+s}{1 + p^2s + 2pqhs}$. As $p$ becomes closer to $1$ (and $q$, consequently, approaches $0$), $p(t+1) = p(t)\frac{1+s}{1+s} = p(t)$. In other words, the rate of growth of a clone will saturate as its frequency approaches 1. To better formalize the advantage of $A$ over $B$, assuming that A and B can confer the fitness advantage $s$ and $r$, respectively, and remembering that $p(t+1) = p(t)\frac{1+s}{\bar{w}}$, the ratio between allele frequencies can be given by $\frac{p(t+1)}{q(t+1)} = \frac{p(t)\frac{1+s}{\bar{w}}}{q(t)\frac{1+r}{\bar{w}}} = \frac{p(t)(1+s)}{q(t)(1+r)}$ which becomes, by induction and for any time increment $t_i$, $\frac{p(t+t_1)}{q(t+t_i)}=\frac{p(t)}{q(t)}(\frac{1+s}{1+r})^{t_i}$. Taking the logarithm of the latter expression, $log(\frac{p(t+t_i)}{q(t+t_i)}) = log(\frac{1+s}{1+r})t + log(\frac{p(t)}{q(t)})$. This linear function of $t$ reveals that A is $e^{m} = \frac{1+s}{1+r}$ times fitter than B, with $m$ being the slope of this linear function of $t$. 

\paragraph{Mutations.} Extending the Wright-Fisher model to account for fitness advantages enables the study of populations evolving under selection and given an initial allele distribution. However, this is not necessarily useful if the focus is on somatic mutations --- in humans, the amount of somatic mutations at birth is quite low \cite{SpencerChapman_2021_gjz4x4,Mitchell2021-zl}, increasing linearly with age at an apparently constant rate in healthy individuals \cite{Abascal_2021_gjvqfm,Mitchell2021-zl}. The Wright-Fisher model can thus be extended to include the possibility of mutations --- a change of alleles in a specific locus --- at every generation. This is parametrized as the mutation rate $u$ which defines the probability of an allele changing in a given locus. This extension permits the simulation of a process that is more similar to the acquisition of somatic mutations through life. On each generation, we would have prior to the sampling process described above a given number of diploid sites $p$ on $N$ individuals under some probability $u$ of acquiring a mutation --- in effect, on each generation, there is a probability of a new mutation being acquired of $2Npu$.

\paragraph{Multiple sites.} An important and final (for the purpose of this dissertation) aspect of evolution is still missing from this model --- that of multiple sites. The possibility of an individual having a specific combination of alleles is important since it allows us to account, for instance, for the presence of multiple alleles conferring fitness advantages or the more generic effect of epistasis, where the effect of a specific allele depends on the presence of other alleles. In terms of the simulation itself, a clone (defined as a genetically-defined and distinct set of individuals from a population) is characterized by a set of $n$ mutations $S_{m} = {m_1,m_2,...,m_n}$ and each of these mutations confers a fitness advantage $s_i \geq 0$, where $s_i$ is the fitness advantage of mutation $i$ ($m_i$). To combine multiple fitness effects, one can assume that these are additive --- for example, if a clone $c_1$ is characterized by mutations $m_{42}$ and $m_{13}$, its fitness advantage is $s_{42} + s_{13}$ and its probability $p_{c_1}$ in the next generation ($t+1$) is given by $p_{c_1}(t+1) = \frac{1+s_{c_1}}{\bar{w}}p_{c_1}(t) = \frac{1+s_{42}+s_{13}}{\bar{w}}p_{c_1}(t)$. Now, rather than having a binomial distribution with $2N$ trials for a population of $N$ diploid individuals, it is multinomial distribution parametrized by a set of probabilities $S_{p_c} = {p_{c_1},p_{c_2},...,p_{c_n}}$, each multiplied by the fitness advantage conferred by the mutations on their respective clones. One advantage of this approach is that the expected value and the variance of the proportion of the counts of any given clone $c_i$ for the next generation is analogous to the binomial case above: $E[X_{c_i}] = np'_{c_i}$ and $Var[X_{c_i}] = np'_{c_i}(1-'p_{c_i})$, where $p'_{c_i} = (1+s_{c_i})p_{c_i}$. 

\paragraph{Somatic evolution.} As mentioned, humans accumulate mutations consistently through life at an approximately constant rate (\cite{Abascal_2021_gjvqfm,Mitchell2021-zl}). A few mutations --- known as drivers --- confer a significant fitness advantage and are positively selected. However, most mutations will confer a negligible fitness advantage and are only able to grow over life thanks to the stochastic dynamics of genetic drift or by appearing in already fit clones. This process, where neutral mutations grow thanks to the fitness effect of driver mutations is known as "hitchhiking" and these mutations are known as passenger mutations. In the context of somatic mutation, a slow, constant process of asexual reproduction (where descendants have a single parent in the previous generation) sometimes leads to the presence of multiple drivers on different clones, leading to clones which are fitter than others and to competition between them. This process, akin to the natural selection posited by Darwin, will lead to the formation of complex mutational landscapes within each individual. In cases where the population is well-mixed and homogeneous, it is possible that very fit clones or clones harbouring more than one driver mutation take over the entire population \cite{Beerenwinkel2015-xr,Grossmann2020-ka}.

The Wright-Fisher model allows for a simple but powerful and informative query --- given a set of initial circumstances, how does a population evolve in terms of its allele frequency over time? However, only in the laboratory can evolution be studied in such a way. In real world settings, the available alternative is observing the current genetic diversity and look back into the past, thus motivating the coalescent.

\subsection{The coalescent}

The coalescent is a phylogenetic reconstruction of the relations between a finite set of individuals (in most cases characterized by their genetic sequences) up to their \ac{mrca}. Typically, the coalescent is used to derive statistics about the population, rather than to precisely describe phylogenetic relations \cite{Gillespie2004-xo}. The simplest way to build an example of the coalescent is to take a sample of $n$ individuals from the current and fixed-size population of $N$ individuals and to "go back in time" --- starting from the present generation $t$, a sample with replacement of the $N$ individuals is drawn for every individual in the sample at generation $t$ with probability $p$ to get to the number of parents in generation $t-1$. Repeating this iteratively, until only one parent is identified --- in other words, until the \ac{mrca} for the sample is found --- leads to the construction of a coalescent that recapitulates the population structure observed in the present. In the coalescent, the probability $p$ represents the probability that a member of the population on generation $t-1$ is the parent of any member of the population on generation $t$.

If $p$ is the same (uniform) for all parents in generation $t-1$, any single one has the same chance of being a parent of any one of the children in generation $t$ --- in other words, the population is evolving neutrally. If $p$ is not uniform (i.e. one of the possible parents has $x$ times the probability of being a parent) this leads to more children in generation $t$ having the same parent, which would be the expected case when a sample of the population is under positive selection. From this description, it is possible to see how this is remarkably similar to the Wright-Fisher model assuming neutrality --- the population is constant and well-mixed, each generation is discrete and non-overlapping and populations evolve under neutral evolution \cite{Gillespie2004-xo}. The fact that the coalescent looks back in time has, as described, good reasons from a theoretical and practical point of view. However, computational reasons also motivate this --- it is much easier to simulate the previous generation of a sample with $n$ elements than the next generation of a population of size $N$ (here $N \gg n$).

The coalescent is helpful to study a number of properties of a population --- for instance, it is possible to calculate the probability that a coalescence will not happen at a given generation $t-1$ for a sample of $n$ individuals from a (diploid) population of size $2N$ in generation $t$: $p_{\mathrm{coal}=0} = \frac{2N-1}{2N}\frac{2N-2}{2N}...\frac{2N-(n-1)}{2N} = (1 - \frac{1}{2N})(1 - \frac{2}{2N})...(1 - \frac{n+1}{2N})$ and, by ignoring terms which are similar to $\frac{1}{4N^2}$ or smaller and considering that $\sum^A_{a=1}{i} = \frac{A(A-1)}{2} = \binom{A}{2}$, $p_{\mathrm{no\ coal}} = 1 - \frac{1}{2N} - \frac{2}{2N} -...- \frac{n-1}{2N} = 1 - \binom{n}{2}\frac{1}{2N}$. This makes the probability of observing at least one coalescent event in generation $t-1$ $p_{\mathrm{coal}>0} = \binom{n}{2}\frac{1}{2N}$. In other words, as $N$ increases, the probability of a coalescence on the previous generation decreases, leading to trees that span more generations. Another consequence of this is that, as the coalescent progresses through generations, the probability of a coalescent event decreases as $n$ also decreases, leading to trees with short branches and many coalescent events near the initial sample and longer branches as the \ac{mrca} approaches. One can generalize this probability to the probability of no coalescence over $\Delta t$ generations: $P(T > \Delta t) = p_{\mathrm{coal}=0}^{\Delta t} = (1-\binom{n}{2}\frac{1}{2N})^{\Delta t}$. Using a similar formulation, Kingman showed, in 1982, that the coalescent times are independent and exponentially distributed such that their probability density function is $f(t_i) = \binom{n}{2}e^{-\binom{n}{2}t_i}$, where $t_i$ is the time during which there are $i$ branches in the coalescent. This allows the calculation of the expected value and variance of the times between coalescent events $T_i$, $E[T_i] = \frac{2}{i(i-1)}$ and $Var[T_i] = (\frac{2}{i(i-1)})^2$ \cite{Kingman1982-tc}. At this point, it is also important to note that the coalescent can also approximate the Wright-Fisher model under neutrality for large enough $N$ and with mutations, highlighting how the coalescent is analogous to genetic drift progressing backwards in time \cite{Wakeley2008-wh}. The coalescent and some of its properties --- particularly the distribution of times between coalescent events --- are the foundation of the estimation of the evolution of the \ac{eps}.

When studying the coalescent, we are not necessarily interested in any specific tree --- in reality and for most cases, a distribution of likely solutions, rather than a single solution, explains the data \cite{Gillespie2004-xo}. The coalescent is, for the most part, used to estimate two parameters of great importance for evolving populations --- the time between generations and the \ac{eps}. The \ac{eps} is the size of an ideal population (a population under the assumptions of random mating, constant population size and fixed generation times) necessary to generate the amount of diversity observed in the genetically characterized sample \cite{Gillespie2004-xo}. The \ac{eps} can then be used to measure fluctuations in the population size and whether the population is under the effect of any form of selection --- as the real population approaches the characteristics of the ideal population, the \ac{eps} also approaches the real population size.

\subsection{Phylogenetics and the phylodynamic estimation of populations}

While the coalescent can be useful, sometimes the interest lies in the estimation of the tree that best describes the population structure. Phylogenetics is concerned with the estimation of the tree that best represents the evolutionary relationship between different individuals, otherwise known as the phylogenetic tree or phylogeny. In phylogenies, it is often assumed that the number of mutations scales linearly with time --- in other words, the rate of mutation accumulation is constant along a phylogeny \cite{Roy2014-ez}. While this is often hard to verify as it requires validation through orthogonal methods or exclusion of some samples \cite{Roy2014-ez}, it is a relatively fair assumption in human somatic evolution in healthy tissues \cite{Abascal_2021_gjvqfm,Mitchell2021-zl,Fabre2020-as}. As such, this permits not only the timing of mutations from phylogenetic trees, but also the estimation of the dynamics of the population through the use of phylodynamics. 

Phylodynamics is, historically, the field concerned with how specific processes --- immunological, evolutionary or epidemiological --- affect viral phylogenies \cite{Volz2013-ey}. Particularly and at its inception, changes to the viral population were suggested as having a sizeable impact on viral phylogenies \cite{Grenfell2004-ip} --- indeed and thinking about the coalescent, where the probability of a coalescent event is in part determined by the size of the population, this is quite a reasonable statement. Following this line of thought also reveals a few more conceptual conclusions from the aspect of a phylogeny --- if the population increases exponentially with time, then coalescent events would have been more probable early on; if, on the other hand, the population remains constant, most coalescent events would occur closer to the time of sampling. It should be noted that, for phylogenetics in general and especially for phylodynamics, the size of the population describes the \textit{effective} population size, rather than population size, as it still assumes, much like the coalescent, an ideal population. For this reason, the population sizes I discuss in the Introduction and in Chapters 2 and 3 refer specifically to the size of the population capable of generating offspring.

While phylodynamics is concerned with the causes of a specific phylogenetic tree structure, the starting point for phylodynamic analyses is often the phylogenetic tree itself, with the purpose of phylodynamics being the reconstruction of a few of the aspects of the population. For instance, the time for two members of a phylogeny to share a coalescent event --- directly observable from a phylogenetic tree --- is assumed to be exponentially distributed with rate $\lambda(n) = \binom{n}{2}\frac{1}{N} = p_{\mathrm{coal}>0}$ --- in other words, the expected time for a coalescence to occur between two members is the inverse of the probability of a coalescent event occurring and can be used to infer how the population evolved. From this, it follows that the expected time until the \ac{mrca} of a phylogenetic tree can be calculated, assuming that coalescent event probabilities are independent of one another, as $\frac{1}{\lambda(n)} + \frac{1}{\lambda(n-1)} + ... + \frac{1}{\lambda(2)}$, which can be stated as $2N(1-\frac{1}{n})$; a natural consequence of this is that, as $n \rightarrow \inf$, the expected time until the phylogenetic tree converges on its \ac{mrca} is $2N$ \cite{Volz2013-ey}. 

The previous paragraph worked under the assumption that $N$ was constant through time --- that the population size did not change along the phylogeny. However, in a fair number of cases and particularly in my case, where the main interest is the evolutionary dynamics of fit clones in the \ac{hsc} population, this can be unrealistic --- not only does the population of specific clones change, but so does the population of \ac{hsc} in general during gestation and development \cite{Lee-Six2018-lp}. As such, one has to think of these models in terms of a changing population size, $N(t)$: $\lambda(n,t) = \binom{n}{2}\frac{1}{N(t)}$. I focus on the analyses of specific groups of branches in the tree, all of which harbour the same driver mutation. This group is otherwise known as a clade and unless, other driver mutations are acquired, I can assume that each individual clade is evolving under neutral pressure. This is helpful since it allows the assumption that a constant growth rate drives the expansion --- for instance, if this growth rate is exponential with initial population size $N_0$ and rate $r$, it is possible to say that the distribution of times between coalescent events is $\lambda(n,t) = \binom{n}{2}\frac{1}{N_0 e^{-rt}}$ \cite{Volz2013-ey}. This example can then be expanded to include changes in rates and dynamics, making the growth of the \ac{eps} more flexible and less parametric; this can be useful to capture, for example, cases of competition or catastrophic events --- in the case of competition, one expects growth to be faster when no other clones are present and competing for the same resources, while in the case of a catastrophic event it can be expected for the \ac{eps} to grow more slowly or even to decrease. Another, perhaps better known example in the field of phylodynamics, is that of flu seasonality --- during colder months, the \ac{eps} of H1N1 circulates through the population much more effectively, leading to a temporary increase in the effective population size \cite{Karcher2020-bg}.

\subsection{The neutral theory (of cancer evolution)}

Even if an excellent genetic model for tumour progression could be devised, a seemingly unintuitive factor of the evolution of mutagenesis is still at play --- a dramatically high number of mutations are acquired each year \cite{Kimura1968-xd}, with most having no detectable fitness advantage. This led Motoo Kimura to posit one of the core tenets of molecular evolution, known as the neutral theory of molecular evolution --- the majority of mutations accounting for intra- and inter-individual variability confer no advantage whatsoever \cite{Kimura1979-ps}; in fact, most mutations are, from an evolutionary perspective, completely useless, and most of the variation observed can be attributed to mutation and stochastic drift. In other words, most mutations are neutral and, in mathematical terms and according to Kimura, their fitness $s$ is smaller than the inverse of the \ac{eps} assuming diploidy: $s<\frac{1}{2N_e}$, with $\frac{1}{2N_e}$ corresponding to the fixation probability for a neutral mutation. Additionally, the probability of fixation for a mutation under positive selection is $\rho = \frac{1-e^{-4spN_e}}{1-e^{-4sN_e}}$ for an allele with initial frequency $p$ \cite{Kimura1962-df} --- assuming that all alleles begin with a single cell, $\rho = \frac{1-e^{-2s}}{1-e^{-4sN_e}}$. Immediately, what becomes evident is that the fixation probability for both depends largely on the population size: in both cases, this probability increases as the \ac{eps} decreases, making the effect of fitness highly dependent on population size --- at very small \ac{eps}, drift overpowers the effect of fitness. While this may be unremarkable, it is important when considering the effect of founding populations when analysing evolution --- a small, physically bounded population can give rise to a dominating and neutral genotype by simply having no pressures that would actively select against it: an example of this is brought by Stephen Jay Gould, who, in his essay “Unenchanted Evening”, discusses Partula snails, a Polynesian snail species which acquired a high level of diversity purely due to the effect of founding populations managing to subsist without contact with other Partula snail populations \cite{Gould1995-yr}.

\subsubsection{Positive selection in cancer under the neutral theory}

The steady accumulation of mutations through life is verifiable in humans \cite{Lee-Six2018-lp,Williams2020-ji,Yoshida2020-zi} --- indeed, between hundreds and thousands of mutations are acquired through life in both normal and cancerous tissue. And this effect of neutral evolution does not stop at somatic mutations in normal tissue --- in human cancer one can also find evidence for the neutral theory of molecular evolution. \etal{Martincorena}, by analysing the evolutionary pressure under which 7,664 tumours across 29 cancer types, determined that each tumour has a modest number of drivers (between 1 and 10) and that about 5\% of all nonsynonymous mutations are under positive selection \cite{Martincorena2017-ii}, thus extending the theory of neutral molecular evolution to cancer. To do this, they built on a popular measure to measure selection --- dN/dS \cite{Miyata1980-mt}. dN/dS is the ratio between the rate of non-synonymous mutations and the rate of synonymous mutations and it quantifies the excess number of positively selected mutations --- for example, a dN/dS of 1.05 shows that 5\% of the nonsynonymous mutations are under positive selection. \etal{Martincorena} extended this model into dNdScv, accounting for different mutation rates across genes and different mutation types (missense, nonsense and essencial splice site mutations), thus accounting for the potentially different selection pressures that may affect truncating mutations when compared to missense mutations and the effect of replication time and expression levels on the mutation rate of different genes \cite{Lawrence2013-rw}. Additionally, dNdScv incorporates context-aware nucleotide substitution models, contemplating tri-nucleotide (the base before and after the substitution site) or penta-nucleotide (the two bases before and after the substitution site) contexts, to account for the effect of different mutational processes \cite{Alexandrov2013-kl}, showing that using simpler substitution models leads to biased estimates of positive selection \cite{Martincorena2017-ii}. 

The term “neutral theory of molecular evolution” is perhaps unintuitive --- by saying that cancer evolution follows the theory of neutral molecular evolution, one is not trying to imply that clones are devoid of a fitness advantage when they sweep an entire population of cells (indeed, \etal{Martincorena} finds an average of 4 driver mutations per tumour); rather, what is implied by this is that the majority of variation is driven not by selection, but by stochastic drift. A more practical example of this is amino acid substitutions --- while a leucine-to-alanine substitution (both apolar, small amino acids) may not lead to any change in fitness, a leucine-to-histidine is much more likely to be deleterious, with histidine being a large, basic amino acid. The generalization of this is best understood as amino acid substitutions being more likely under a neutral evolution model as the physico-chemical distance between the original and mutant allele decreases; conversely, mutations which confer a fitness advantage are more likely to correspond to a higher physico-chemical distance between original and mutated allele \cite{Kimura1979-ps}. The main conclusion one should leverage from the neutral theory of molecular evolution is that there often are cases where mutation prevalence in specific genes may not correspond to an effective increase in fitness; drift can account, especially at earlier stages through the effect of founding populations, for some of this variation, while many nonsynonymous mutations may confer no advantage, creating difficulties in the selection of potential targets for cancer therapy. It becomes clear how, through the lenses of the neutral theory of molecular evolution, prevalence can fail to be representative of fitness. 

\section{The dynamics of clonal haematopoiesis}

Several cross-sectional studies on \ac{ch} have been conducted, helping researchers gain a perspective on the prevalence and possible associations of \ac{ch} with relevant clinical factors. However, few have been able to draw conclusions regarding the dynamics of \ac{ch}. Some studies have used clone size to draw conclusions regarding the timing of each clone, assuming that larger clones appeared earlier \cite{Li2018-vz}. However, these estimates rely heavily on the assumption that the fitness advantage conferred by each mutation is relatively similar. Others have used clone size in combination with copy number alterations to rank mutations on a per-individual basis, further aggregating these estimates for each cancer type to render genetic models of mutational acquisition for different tumours \cite{Jolly2018-yb,Gerstung2020-kf} and even specifically to haematological cancers \cite{Papaemmanuil2013-yt}. While it seems intuitive to test these methods on \ac{ch}, it is unlikely that the mutational landscape is composed of a large clone accumulating more mutations --- in fact, it is much more likely that \ac{ch} is oligoclonal, with few co-occurring mutations on the same clone \cite{Miles2020-fz,Mitchell2021-zl,Saiki2021-sq}. As such and despite recent developments in the study of cancer evolution, different approaches have to be considered for \ac{ch}.

In 2016, \etal{Young} presents the first instance of longitudinal sequencing in \ac{ch}. However, having only access to two timepoints per individual and a relatively small cohort, it avoids drawing any inference pertaining to the dynamics \ac{ch}, noting instead that there appears to be a high degree of heterogeneity --- while some clones grow considerably, others remain stable or even decrease in size \cite{Young2016-du}. In 2020, \etal{Suarez-Gonzalez} investigate how these clones evolve after bone marrow transplant, reaching similar conclusions to those by \etal{Young} \cite{Suarez-Gonzalez2020-hn}. Later in 2020, \etal{Mencia-Trinchant} looked at how \ac{ch} mutations were acquired and evolved in two astronauts before, during and after spaceflight. To do this, they whole-genome sequence both individuals at high temporal resolution and show how different genes and sites grow differently, with two distinct \ac{dnmt3a} clones showing dramatically different trajectories within the same individual \cite{Mencia-Trinchant2020-eb}. However and similarly to \cite{Young2016-du}, the very small sample size prohibits any form of major conclusions.

In 2020, it was van \etal{Zeventer} that provided corroborating evidence to and elaborated on the aforementioned heterogeneity when looking at \ac{ch} in a control-matched cohort of anaemic older individuals. In their work, they quantify the absolute \ac{vaf} difference using two timepoints and show that this difference is statistically significant for mutations in \ac{tet2}, \ac{tp53} and \ac{asxl1}. Here, they also try to assess whether specific mutations in \ac{ch} constitute a risk factor for the acquisition of novel mutations and present results that point towards \ac{asxl1} and \ac{tet2} mutations as risk factors for additional \ac{ch} mutations and towards the protective effect of \ac{dnmt3a} \cite{Van_Zeventer2020-pw}. In the same year, \etal{Watson} present the first systematic study on the gene-specific evolutionary dynamics of \ac{ch} by aggregating a wealth of cross-sectional studies on \ac{ch}. Here, they make use of the Fokker-Planck equation --- a mathematical model that describes the temporal evolution of probability density functions under both deterministic and stochastic forces (such as fitness advantages and genetic drift, respectively) --- in combination with well-established molecular evolution theory to model how the \ac{vaf} density of different genes and sites evolves. In their work, they derive and apply a model that builds on this to infer the fitness that is conferred by mutations in different sites under a few assumptions, particularly that both fitness advantages and mutation rates remain constant through life. While both of these assumptions are apparently reasonable they may not be true for all driver mutations as I will show in Chapter 3. Additionally, the authors do not consider small clones as they may fall below each "[studies'] limit of reliable variant detection" and be detection artefacts; however, if low \ac{vaf} are, rather than technical artefact, biological fact, this may lead to a slight over-estimation of fitness advantage. Either way, the work developed by \etal{Watson} is still remarkable and able to capture the average fitness of clones over life. In this work, it is shown that different mutations have fairly different fitness advantage, even if they affect the same gene (this is particularly made clear in \ac{dnmt3a}, with fitness advantages ranging between 10\% and almost 20\%) \cite{Watson2020-pz}.

A 2021 article from \etal{Poon} --- from the same group as \cite{Watson2020-pz} --- presents a different take on the dynamics of \ac{ch} --- using the \ac{vaf} distribution of synonymous mutations (the key assumption in this case is that the accumulation of synonymous mutations can only be caused if they are passengers in a clone carrying driver mutations) and propose that this can be used to infer the growth rate of different clones. Additionally, by association with the estimation of fitness effects directly from the \ac{vaf} distribution, they are able to uncover a wealth of undetected driver mutations by showing that only 10\% of cases where the synonymous \ac{vaf} distributions are likely to be due to their co-occurrence with driver mutations are in individuals with detectable drivers which reached the same \ac{vaf}. Finally and as far as \ac{ch} is concerned (since they generalize their analysis to oesophagus and other tissues), they also show that mutations which confer smaller fitness advantages are likelier to appear earlier in life, whereas highly fit variants are thought to be biased towards older age based on \ac{vaf} distributions alone \cite{Poon2020-ek}.

In 2021, a preprint was posted which discussed the association between clonal dynamics and insulin resistant and cholesterol levels in obese individuals. To do this, they look at how \ac{vaf} levels evolve in 40 individuals over 20 years, showing that most trajectories (42/77) are static, with a large fraction of them showing considerable growth (30/77) and the remaining trajectories decreasing over time (5/77) \cite{Van_Deuren2021-ll}. However, they do not try to quantify the gene- or site-specific rate growth for any mutation, mentioning that the same mutation appears to grow at different rates across different individuals, a conclusion similar to that posited by \etal{Young} years earlier \cite{Young2016-du}. A later 2021 preprint by \etal{Robertson} follows a longitudinal sequencing experimental protocol similar to the one I worked with --- 89 older individuals (70-90 years) were followed over a median period of 12 years and the blood of each individual was error-corrected sequenced three times over this period at high depth (median: 1930x). They then calculate the growth coefficients of different trajectories and use these to calculate an aggregate on a per-gene basis. Similarly to \cite{Watson2020-pz}, they assume a simple birth-death and differentiation process for the \ac{hsc} population where the fitness conferred by each mutation can be understood as a bias towards self-renewal in symmetric divisions. In this work, they only infer the fitness advantage of clones which are growing faster than would be expected from genetic drift alone, which may be problematic as this is controlled on a per-clone (rather than per-site or per-gene) basis and may lead to inflated estimates of the fitness advantage of each gene. In this work, they reach conclusions similar to the ones I present in Chapter 3 and that are conflicting with those presented by \cite{Watson2020-pz} --- that mutations in splicing genes confer greater fitness advantages, while those genes involved in \ac{dna} methylation (\ac{dnmt3a}) confer much smaller fitness advantages \cite{Robertson2021-sw}. This highlights an important difference between both approaches --- modelling \ac{ch} dynamics using cross-sectional or longitudinal data: the observable clonal dynamics during old-age may not be representative of lifelong clonal dynamics. Conversely, estimating the fitness advantage conferred by a mutation using cross-sectional data does not imply that all clones with this mutation expand at a given rate, nor does it imply that this rate is constant through life; rather, it implies that an average fitness advantage is capable of explaining the observed distribution of \ac{vaf} for this mutation in individuals with \ac{ch}. I show in Chapter 3 why this can be a problematic assumption that misses some key aspects of \ac{ch}.

\section{The diagnosis and prognostis of myelodysplastic syndromes}

Myelodysplasias --- or \ac{mds} --- are a broadly defined group of oncological diseases characterized by the absence of healthy myeloid cells and presence of immature cells in the blood. This can be contrasted with more neoplastic conditions such as \ac{aml}, which are characterized by an abnormal growth of myeloid cells. The incidence of \ac{mds} was 4.1 cases per 100,000 in the United States of America in 2015 \cite{Zeidan2019-jj} and 4.44 cases per 100,000 in the United Kingdom \cite{Roman2016-ad}. However, it is important to note that this is highly age-dependent --- indeed, while \ac{aml} is more prevalent than \ac{mds} in individuals younger than 60, this trend is inverted in individuals over 60, with \ac{mds} accounting for a much larger fraction of myeloid malignancies in this age group \cite{Zeidan2019-jj}. \ac{mds} is also more prevalent in white and male individuals \cite{Zeidan2019-jj}. Apart from age, ethnicity and sex, other risk factors also contribute towards the incidence of \ac{mds}, particularly exposure to industrial or agricultural chemicals \cite{Nisse2001-qf,Lv2011-iv,Strom2005-pq}, smoking \cite{Lv2011-iv,Strom2005-pq}, obesity \cite{Ma2009-ws} and even proximity to an industrial plant \cite{Nisse2001-qf}. \Ac{t-mds}, an aetiologically-defined \ac{mds} as being caused by cancer therapy \cite{Kuendgen2021-in}, while being relevant, is not discussed --- for the remainder of this dissertation, mentions of \ac{mds} refer to primary \ac{mds} unless otherwise noted.

\paragraph{Prognosis and treatment.} \Ac{mds} can be further divided into a fairly broad range of clinically-relevant subtypes that affect prognosis and treatment. This depends on the relative frequency of blood and bone marrow blasts, number of cytopenias (\ac{mds} is not diagnosed if there are no cytopenias and the current diagnostic criteria include three possible cytopenias --- low \ac{hb} concentration, low platelet count and low \ac{wbc}/neutrophil count), number of dysplastic lineages as observed in a bone marrow slide, cytogenetics and presence of ring sideroblasts in the bone marrow \cite{Arber2016-os}. This highlights the multiplicity of tests that have to be performed in order to confirm the diagnostic of \ac{mds} and is indicative of the inter-observer diagnostic heterogeneity --- indeed, while cytogenetics or the detection of cytopenias can be relatively unambiguous, the inter-individual (inter-expert) concordance for the detection of dysplasias is between 0.7-0.8, but can drop to 0.5 when the task is that of classifying specific abnormal cell types \cite{Font2013-lx, Parmentier2012-wm, Della_Porta2015-hw}. As such, it is reasonable that \ac{mds} can require independent confirmation when there is a low degree of dysplasia or only one dysplastic lineage is identified \cite{Arber2016-os}.

\paragraph{Molecular characterization of \ac{mds}.} Recently, thanks to the advent of next-generation sequencing technologies, it is now possible to characterise the specific mutational landscape of \ac{mds}. Particularly, it has been shown that \ac{mds} harbours driver mutations in up to 90\% of cases, with over 60\% of \ac{mds} cases having mutations in splicing genes \cite{Papaemmanuil2013-yt,Haferlach2014-bh}. Except for mutations in \ac{sf3b1}, \ac{dnmt3a}, \ac{jak2} and \ac{mpl}, driver mutations were more frequently observed in high risk subtypes of \ac{mds} such as refractory anaemia with excess blasts types 1 and 2, a \ac{mds} subtype characterised by having an excess of blasts in peripheral blood and in the bone marrow \cite{Haferlach2014-bh}. Additionally, the likely clonal architecture of \ac{mds} also hints that some mutations are likelier to appear earlier (clonal) rather than later (subclonal) in \ac{mds} --- while mutations \ac{idh2} and \ac{sf3b1} are likely to appear earlier, mutations in \ac{nras} and \ac{idh1} are likely to appear later as \ac{mds} progresses --- with clonal and subclonal mutations equally affecting prognosis \cite{Papaemmanuil2013-yt}. Finally, \ac{sf3b1}, a relatively prevalent gene in \ac{mds}, is associated with refractory anaemia with ring sideroblasts, a \ac{mds} subtype characterised by the presence of ring sideroblasts in the bone marrow \cite{Papaemmanuil2011-nr, Papaemmanuil2013-yt, Haferlach2014-bh}. However, and despite their informative potential, the contribution of molecular characterisation to prognosis is reduced; indeed, for a model predicting leukaemia-free survival, the \ac{auroc} using information on peripheral blood counts, expert evaluation of bone marrow, cytogenetics, and demographics was of 80\% and adding the molecular characterisation increased the \ac{auroc} by only 2\% \cite{Papaemmanuil2013-yt}: while it is clear that the potential of the molecular characterisation of \ac{mds} can be useful, more traditional methods carry most of the information required for both diagnosis and prognosis.

\paragraph{The unrealized potential of \ac{wbs} in \ac{mds} prognostic assessment.} While the inspection of both a \ac{wbs} and a bone marrow slide are essential for the diagnostic of \ac{mds} \cite{Greenberg2012-en,Arber2016-os}, prognosis makes no use of the information in \ac{wbs} --- neither the \ac{ipss} nor its updated version, the \ac{ipss-r}, two systems that are fundamental for the risk assessment of \ac{mds}, include any aspect that could be further assessed from the analysis of a \ac{wbs} \cite{Greenberg2012-en}. Indeed, assessing the risk --- both of death and of progression to \ac{aml} --- of any form of \ac{mds} has been simplified into an algorithm that considers cytogenetic evaluation, which is influenced by both the number and the type of abnormalities, the relative frequency of blasts in the bone marrow, haemoglobin and platelet levels and absolute neutrophil counts \cite{Greenberg2012-en}. Nonetheless, it is worth noting that this risk assessment, while relatively simple, is remarkably powerful --- very-low risk \ac{mds} patients live over 10 times longer than those with very high risk \ac{mds} (8.8 years vs. 0.8 years, respectively) \cite{Greenberg2012-en}. Additionally, when the \ac{ipss} is used as the single variable in a model predicting leukaemia-free survival the \ac{auroc} differs only by 6\% from that of a model incorporating information on peripheral blood counts, expert evaluation of the bone marrow, cytogenetic and demographics, and molecular characterization (\ac{auroc} = 76\% vs. \ac{auroc} = 82\%, respectively) \cite{Abelson2018-wh}. It is worth noting that recent publications using cytomorphology (either automated or derived by an expert) in combination with machine-learning focus specifically on the bone marrow, showing that there is more information to be mined from cellular morphology than what has been thus far inferred from human expert inspection \cite{Bruck2021-fx,Nagata2020-lh}, but make no effort to expand this analysis to \ac{wbs}. Together with the aforementioned fact that inter-individual concordance can be lacking in \ac{wbs} and bone marrow slide analysis \cite{Font2013-lx, Parmentier2012-wm, Della_Porta2015-hw}, the need for reproducible and more objective diagnostic criteria becomes clearer.

\section{Machine-learning-based computer vision methods and their application in cytomorphology}

Computer vision is the field dedicated to the computational analysis of images. Its reproducibility, objectivity and quantitative nature are key aspects for scientific discovery, particularly in the life sciences. Here, I focus on a brief history of what I believe are key developments in the field of computer vision methods as applied to the analysis of cells, focusing particularly on cell detection and cell morphometry, i.e. the measurement and quantification of cell shape (cytomorphology).

\subsection{Computer vision and morphometry in biological sciences}

\subsubsection{From early adaptations to high-throughput applications}

One of the first applications of computer vision in a biological field can be traced back to 1966, when C.W. Gilbert coded a short computer program that would enable him to automatically derive a karyogram from a set of chromosomes \cite{Gilbert1966-km}. In this work, measuring chromosome lengths was still done by humans, with the machine being only responsible for the correct designation of chromosomal pairs. While far away from the sophistication of image analysis as it is known today, I include it here for historical purposes as the first instance, to the best of my knowledge, of a computer program that dramatically reduced the human labour associated with the analysis of a biological image. In 1980, \etal{Howe} used a much more refined method --- using a microscope attached to a densitometer, they registered the grey level of neutrophils under different preparations at a sequence of depths, using this to calculate not only neutrophil density, but also neutrophil motility \cite{Howe1980-ua}. This represented a remarkable advancement --- rather than relying on the laborious observation and possibly erroneous appreciation of cells under a microscope, researchers now had access to an automatic method capable of accessing most, if not all, of the available information on an image. While temporally distant from the advances observed during this century, these were the changes that eventually led to cytomorphology as a valuable tool. 

Here, the applications of cytomorphology should be made clearer --- while it is evident that cytomorphology is concerned with the analysis of the morphology of individual cells, it is not the individual cell that is relevant for analysis --- in most cases, researchers are concerned with general morphometric trends of cell populations, rather than with the morphometry of individual cells (exceptions to this include the analysis of very small multicellular organisms using techniques such as volumetric electron microscopy \cite{Vergara2021-nx,Witvliet2021-gv}). For instance, \etal{Fuchs} analyzed in which ways cellular morphology was associated with expression in a genome-wide RNA inhibition experiment \cite{Fuchs2010-pk}, \etal{Gustafsdottir} developed a cell painting assay and used a similar approach to measure how cellular morphology was affected by specific bioactive compounds \cite{Gustafsdottir2013-ul} and \etal{Simm} repurpose high-throughput imaging and cellular morphometry to infer the activity of compounds on biological components different from the ones measured initially \cite{Simm2018-nh}. These examples, while highlighting the already established importance of cellular morphology, are indicative of the immense potential of high-throughput imaging in deriving novel and important biological conclusions such as the discovery of novel genes involved in \ac{dna} damage repair \cite{Fuchs2010-pk} or discovering new and unbiased approaches to study cell states associated with specific diseases and compounds \cite{Gustafsdottir2013-ul}. This has motivated different software programs dedicated to cell detection and morphometric characterization which provide an easy interface for computationally-inexperienced users such as CellProfiler \cite{Carpenter2006-hy} and Ilastik \cite{Sommer2011-ds}.

\subsection{Detecting cells}

\subsubsection{Segmentation using convolutional features}

Typically, cellular morphometry requires cells to be segmented --- in other words, each pixel in the cell must be correctly identified as belonging to the cell. While works using fluorescence microscopy can make this task remarkably simple by using markers for both the nucleus and cytoplasm proteins \cite{Fuchs2010-pk,Gustafsdottir2013-ul}, other imaging modalities have motivated the widespread popularity of \ac{ml} algorithms for cell segmentation. Classic \ac{ml} approaches to this problem generically involve three steps, the second of which is optional: feature extraction, feature selection and training of a classification algorithm. As an example, I progressively show how the feature extraction and algorithm training steps work using an image of a haematoxylin and eosin stained colorectal adenocarcinoma \cite{Graham2019-em} (\figref{fig:segmentation-example-image-labels}).

\begin{figure}[!ht]
	\placefigure{intro/segmentation-example-image-labels}
	\placecaption{Small section of a stained colorectal adenocarcinoma histopathology slide (left), the segmented nuclei (center) and the overlay of both images (right).}
	\label{fig:segmentation-example-image-labels}
\end{figure}

\paragraph{Feature extraction.} For each individual pixel, a set of $n$ features is calculated, some of which are based on convolutions (the weighted sum of a $p \times p$ window of pixels) and incorporate neighbouring pixel information. Features from the pixel itself usually pertain to colour, intensity or hue, whereas features based on convolutions capture the local structure of the image and can be used to detect, among other local features, edges \cite{Canny1986-pi} and structure \cite{Sato1998-sy}. Before introducing these, however, it is important to grasp the concept of a Gaussian filter and how it affects an image --- for a given pixel $ij$ in an image, its vicinity is weighted by its density calculated from a two-dimensional Gaussian with standard deviation $\sigma$ centred on pixel $ij$. The result is of applying a Gaussian filter to an image is a blurred image which becomes increasingly more blurry as $\sigma$ increases since the vicinity also increases with $\sigma$ (\figref{fig:segmentation-example-gaussian-filter}). The reason why Gaussian filters are so important in feature extraction is because they make the image smoother; in other words, they eliminate high-frequency spatial information (large, local changes in pixel values).

\begin{figure}[!ht]
	\placefigure{intro/segmentation-example-gaussian-filter}
	\placecaption{Result of applying a Gaussian filter with increasing values of $\sigma$ to the image in \figref{fig:segmentation-example-image-labels}.}
	\label{fig:segmentation-example-gaussian-filter}
\end{figure}

By reducing noise at a local level, Gaussian filters are often times used as a preprocessing step for feature extraction, as is the case with some edge features (\figref{fig:segmentation-example-edges}). These features capture specific aspects of the colour gradient at a given pixel. The Laplacian of Gaussian --- after smoothing the grey-scale image with a Gaussian filter to avoid high frequencies, the Laplacian of the image ($L(x,y) = \frac{\delta^2I}{\delta x^2} + \frac{\delta^2I}{\delta y^2}$ or the second spatial derivative of an image) --- is obtained using the following convolutional filter:
	
\begin{center}
    $\begin{bmatrix}0 & -1 & 0\\-1 & 4 & -1\\0 & -1 & 0\end{bmatrix}$
\end{center}

\begin{figure}[!ht]
	\placefigure{intro/segmentation-example-edges}
	\placecaption{Result of calculating the Laplacian of Gaussian (top) and the difference of Gaussians (bottom) with increasing values of $\sigma$ of the image in \figref{fig:segmentation-example-image-labels}.}
	\label{fig:segmentation-example-edges}
\end{figure}

Other edge features, such as the difference of Gaussians, offer different ways of measuring of the local colour gradient --- however, instead of applying an additional filter over a Gaussian-filtered image, it applies two Gaussian filters with different values for $\sigma$ ($\sigma_1$ and $\sigma_2$, where $\sigma_1 > \sigma_2$) on the original image and subtracts the blurrier image ($\sigma_1$) to the less blurry image ($\sigma_2$). This is equivalent to the calculation of the local gradient and approximates the Laplacian of Gaussians \cite{Bundy1984-uu} (\figref{fig:segmentation-example-edges}). Typically, to define both values of $\sigma$, $\sigma_1 = \sigma_2 * 1.6$.

Texture features capture the local organization or intensity distribution of pixels in a given region or surrounding a specific pixel. A popular measure of texture at a pixel is given by the eigenvalue of its structure tensor. The two-dimensional structure tensor is a $2 \times 2$ matrix containing the weighted sum of pixel gradients surrounding a pixel and the eigenvalue of this matrix provides a value for the magnitude of the distribution of the gradient. As visible in \figref{fig:segmentation-example-texture}, increasing $\sigma$ allows for the extraction of texture features at a larger scale. An additional feature describing texture is the determinant of the Hessian matrix of each pixel --- the Hessian is the square matrix of second-order derivatives at the level of each pixel. This matrix, similarly to the structure tensor, characterizes how the gradient is locally distributed --- by calculating its determinant it is thus possible to derive the magnitude of this local gradient.

\begin{figure}[!ht]
	\placefigure{intro/segmentation-example-texture}
	\placecaption{Result of calculating the first eigenvalue of the structure tensor (top) and the determinant of the Hessian matrix (bottom) with increasing values of $\sigma$ from the image in \figref{fig:segmentation-example-image-labels}.}
	\label{fig:segmentation-example-texture}
\end{figure}

\paragraph{Feature selection.} Feature selection is a step that can help in reducing the amount of necessary features, which can be essential if image segmentation has to be fast or even happen in real-time in high-throughput scenarios. It can also help with \ac{ml} model training if computational resources are scarce. Approaches to perform unsupervised feature selection --- i.e. selecting which features may be the most important for classification without having access to annotated data --- are reviewed elsewhere \cite{Solorio-Fernandez2020-cx} and, considering that they are not the focus of this dissertation, are not discussed here. 
	
\paragraph{Training of a classification model.} After characterizing each pixel, the set of $n$ features is used to classify whether a pixel belongs or not to the object of interest (a cell, in this case). This can be a binary classification problem (i.e. does the pixel belong to the cell?) or a multiple class classification problem, also known as semantic segmentation (i.e. to which object does this pixel belong to, if any?). For this example, features similar to those extracted by Ilastik \cite{Sommer2011-ds} and focus on a specific task --- that of segmenting the nuclei in the original image in \figref{fig:segmentation-example-image-labels} (left image). I use two distinct sets of features for this prediction --- one which only considers colour intensity features (the colour intensity values after applying a Gaussian filter) for small values of $\sigma$ ($[0.3,0.5,1.0]$), and a second one which considers colour intensity (Gaussian filter), edge (Laplacian of Gaussian and difference of Gaussians) and texture features (the eigenvalues of the structure tensor and the Hessian determinant) for a larger range of values for $\sigma$ ($[0.3,0.5,1.0,1.6,3.5]$). I train a random forest to discriminate between different labels (pixel belongs to nuclei (positive) vs. pixel belongs to the background (negative)) --- a random forest is a \ac{ml} algorithm that is an ensemble of decision trees, each of which is trained using a randomly selected set of features and samples and validated using samples which were not used to train it \cite{Breiman2001-yz}. Its main advantages in general \ac{ml} settings are its speed (during both training and inference), high scalability and good generalization and low chance of overfitting. For semantic segmentation in particular, it can be useful since it can be easily deployed in tasks with multiple classes.

After training, I apply this model to two similar images --- one obtained from the same slide, immediately adjacent to the image used for training, and another from a different slide. In \figref{fig:segmentation-example-result} I show the results of this very simple and illustrative experiment --- one can see that the algorithm performs almost perfectly when applied to the training data, but performs poorly when applied to other images. For the first test image (extracted from the same slide as the training data), it is possible to see that a larger region of pixels which do not belong to nuclei have a high probability of belonging to the nuclei in both models (red circles in \figref{fig:segmentation-example-result}), whereas for the second test image a large number of cells remains undetected (green circles in \figref{fig:segmentation-example-result}). Additionally, it is possible to see that in the second model, where larger values of $\sigma$ were used, predictions appear to be smoother and less noisy. This set of observations highlights a few relevant aspects of training these algorithms:

\begin{itemize}
    \item Generalization is highly dependent not only on the features used for characterizing pixels, but also on the training data itself and how similar it is to the test data
    \item Extracting features considering relatively large vicinities can be highly valuable as it allows for smoother predictions and these provide better contextual information
    \item It is not always the case that increasing the number of features leads to large increases in performance --- in the example provided in \figref{fig:segmentation-example-result}, the performance gain from increasing the number of features is not as high as one would hope. By measuring the \ac{iou} (a measure of segmentation quality discussed in Chapter 4) between predictions and ground truth, it is possible to better assess this --- the \ac{iou} for the model which uses more features is only 0.3\% higher in both cases.
\end{itemize}

\begin{figure}[!ht]
	\placefigure{intro/segmentation-example-result}
	\placecaption{Segmentation results on a training image (first column) and two test images (second and third columns). The input data is shown on the first row, the ground truth is shown on the second row and the third and fourth rows show the prediction results using two distinct sets of convolutional features.}
	\label{fig:segmentation-example-result}
\end{figure}

This chapter highlights how unsuitable convolutional features may be for segmentation tasks --- it is hard to know which features are useful \textit{a priori}. The advent of the \ac{cnn}, where convolutional features are automatically learned by a \ac{ml} algorithm at an immense scale, marks an important shift in this --- while the human design of convolutional features is cumbersome and oftentimes inneffective, convolutions can still be extremely useful in \ac{cv}. 

\subsubsection{Deep convolutional neural networks}

More traditional segmentation methods such as the one described above, with separate stages for feature extraction and classification, are seldom used nowadays, particularly in bioimage analysis. This is due to the rapid rise of open source and easily deployable \ac{dl} methods \cite{Lucas2021-hc}. \Ac{dl}, while having significant variability in terms of implementation, consists in the use of large and deep \ac{ann} in machine-learning applications. \Ac{ann} are learning algorithms which can be generically decomposed into a sequence of matrix operations and non-linear transformations, organized into "layers". As the number of layers increase, so does the representational power of an \ac{ann} --- this is one of the main strengths of \ac{dl} as a framework, their theoretical capability of approximating any function \cite{Nielsen2015-wn}. This is enabled by the non-linearities introduced in each layer --- examples of this include the sigmoid function or the \ac{relu}, which outputs the identity for any input greater than $0$ and $0$ otherwise. As for matrix operations, the two most relevant ones are the matrix multiplication --- used in "linear" \ac{ann} --- and convolutions. Convolutions in particular are the basic construct of \ac{cnn}, \ac{dl} networks which learn and apply a set of convolutional filters on each layer --- rather than having a fixed set of parameters such as the Laplacian detailed above, convolutions in \ac{cnn} are iteratively learned. This gives \ac{cnn} the ability to intrinsically learn how to extract features from an image, an advantage over previous approaches that greatly reduces the burden on and the bias of researchers. Combined with max-pooling --- an operation that selects the highest value in a $p \times p$ sliding window --- \ac{cnn} learn relevant convolutional features while learning that these should correspond to the local maximum if they are relevant for classification. The consequent application of several convolutions leads to an increase in the \textit{depth} of an image --- in other words, the number of channels in the image increase as richer representations are built --- whereas the applications of max-pooling operations extracts the most relevant features out of an $p \times p$ sliding window, shrinking the images in terms of their height and width and decreasing its resolution. The last layer of \ac{cnn} networks for classification usually consists on an operation that extracts the most relevant feature of each channel, using this vector to classify the image according to a pre-designated set of labels (\figref{fig:cnn}). \ac{cnn} were first formulated in these approximate terms by Fukushima in 1980 and offered one of the first translation-invariant solutions to image recognition \cite{Fukushima1980-bk}. The optimization of these models, as is the case for many other models, can be done in many different ways given an approriate loss function. However, the size of modern \ac{ann} makes gradient-free optimization methods unfeasible as their ability to find a good solution decreases with the size of the problem --- particularly, \etal{Rios} test gradient-free optimizers on problems with up to 300 parameters, showing that solutions at this relatively small scale struggled to solve problems with more variables \cite{Rios2013-bx}. Immediately this becomes a problem for training \ac{cnn} --- most modern models have, at least, millions of parameters \cite{Li2021-xu}. For this reason, \ac{cnn} and \ac{ann} in general remained relatively unpopular for a very long time, regaining their popularity with the advent of gradient-based methods \cite{Kurenkov_undated-pu}.

\begin{figure}
	\placefigure{intro/dnn}
	\placecaption{Schematic representation of a convolutional neural network. Differently coloured arrows correspond to different operations.}
	\label{fig:cnn}
\end{figure}

\paragraph{Neural network training and backpropagation.} As mentioned in the previous paragraph, \ac{cnn} models are composed of many layers, all of which are applied consecutively. To train these models using gradient-based methods, the calculation of the gradient of the loss function with respect to the parameters is necessary. However, the naive calculation of these gradients, where each gradient is calculated independently of others, lacks the efficiency to make \ac{cnn} training a reasonable endeavour. Backpropagation, the application of the chain rule to compute gradients as these are propagated backwards (from output layers to input layers), was first shown to be effective in training \ac{ann} in the 80s and remains to this date the most widely used gradient-based optimization family of methods in \ac{ann} training \cite{Rumelhart1986-tl,LeCun1989-ro}. The chain rule is, for two differentiable functions $f$ and $g$ and their respective outputs $y = f(u)$ and $u = g(x)$, $\frac{dy}{dx} = \frac{dy}{du}\frac{du}{dx}$ and, as mentioned, can be applied sequentially to calculate \ac{ann} gradients with respect to the network coefficients \cite{Rumelhart1986-tl}. In other words, the derivative of a composite function can be decomposed into the product of the derivative of both functions which make up the composite function. 

Given an image $I_i$ and its respective label $C_i \in \{0,1\}$, the interest of training a \ac{cnn} $F$ for classification lies in determining whether $I_i$ belongs to class of interest $C_i$ by calculating $P_i = F(I)$, the probability that $I_i$ belongs to $C_i$. $P_i$ is ideally close to $C_i$, a proximity that is measured by a loss function $L_i(C_i,P_i)$ such as the cross-entropy, $C_i\log(P_i) + (1-C_i)\log(1-P_i)$. This can be generalised to multiple instances of image-label pairs ($\{\{I_1,C_1\},\{I_2,C_2\},...,\{I_n,C_n\}\}$), making the cross-entropy $\frac{1}{n}\sum_{i=1}^{n}C_i\log(P_i) + (1-C_i)\log(1-P_i)$. Given that $F$ and the cross-entropy is differentiable, it is possible to calculate the gradient of the loss function with respect to the parameters of the network using the chain-rule and backpropagation, making the training of \ac{ann} a tractable endeavour by consecutive applications of the chain rule. 

It is worth noting that, while backpropagation in \ac{ann} and in the \ac{cnn} dates back to the 1980s, the groundbreaking implementation of a \ac{cnn} that can be considered as the "dawn" of \ac{dl} for image classification was AlexNet, one of the first \ac{cnn} to be implemented in a \ac{gpu} that dominated image classification for several years, combining remarkable classification performance with comparatively fast training and inference \cite{Krizhevsky2012-by}. The reason why \ac{gpu} cards are so popular for this is because, unlike older technologies such as the \ac{cpu}, \ac{gpu} cards have many cores and are ideal for highly parallelizable computations such as matrix operations and convolutions, two of the key components of \ac{ann} and \ac{cnn}.

\paragraph{Image segmentation with U-Net.} It was a few years after AlexNet that U-Net, one of the first \ac{dl} networks to become widely popular in bioimage segmentation, was first introduced \cite{Ronneberger2015-do}. Rather than outputting a single class for each image, U-Net produces a segmentation map --- a classification for each pixel that predicts whether it belongs to a specific class, making ideal for tasks such as that specified above in \figref{fig:segmentation-example-result}. Since then, other algorithms have become widely available for bioimage segmentation or instance detection \cite{Lucas2021-hc}, but I focus on U-Net since it is the one I used in the work presented in Chapters 4 and 5. The U-Net architecture belongs to a larger subset of \ac{cnn} architectures known as fully-convolutional \ac{cnn}, referring to the fact that, unlike "vanilla" classification \ac{cnn}, which have a linear classifier as their last layer, fully-connected \ac{cnn} have only convolutional operations. The U-Net architecture owes its name to its typical "U" shape, combining an encoder --- responsible for extracting relevant convolutional features from the image --- and a decoder --- responsible for combining the rich information from several layers of the encoder (\figref{fig:u-net}). The encoder itself is composed of convolutions and max-pooling operations, whereas the decoder takes this information and upscales it to the original resolution of the image. As convolutions add on to an input, the retrieved features become more and more complex --- to maintain a resolution that is close to that of the original image and to make good use of the features retrieved earlier during the segmentation, the decoder incorporates information from the encoder through skip connections (green in \figref{fig:u-net}), where an array from the encoder is concatenated to an identically-sized array of the decoder, going through a consequent set of convolutional operations. The combination of a U-shaped architecture, where progressively complex features are retrieved and then operated on through convolutions and upscaling, together with skip connections, which enable the U-Net to maintain a good degree of resolution, are what give it its remarkable performance --- it is no surprise that for the 2018 Data Science Bowl competition --- a massive competition to achieve the best performing results in divergent nuclear segmentation --- the top two solutions were either U-Net models or relied heavily on them, with the first and second places achieving a \ac{iou} of 63\% and 61\%, respectively \cite{Seferbekov_undated-br,jacobkie_undated-tc}. 

\begin{figure}
	\placefigure{intro/u-net}
	\placecaption{Schematic representation of a U-Net. The input is processed iteratively as feature representations are increasingly retrieved (convolution + maxpool). This rich feature representation is then upscaled and further convolved (upscale + convolution) and the output of these and previous layers is concatenated in order to preserve resolution (identity + concatenation).}
	\label{fig:u-net}
\end{figure}

\subsection{Machine and deep-learning --- practical considerations}

In Chapters 4 and 5, a considerable portion of the methodology rests on \ac{ml} and \ac{dl}. As such, it is important to discuss a few recurring features which come into play when training such models, especially in conditions where data acquisition and annotation is complicated. Particularly, it is worth highlighting that these methods, while extremely powerful, require vast amounts of data during training. This is a strategy that generically ensures the success of data-driven approaches --- by using increasingly larger amounts of data from diverse sources and sites, it is more likely that the model generalizes to newer data. However, in the life and biomedical sciences, this represents a circumstance which can be prohibitive: both fields are typically characterized either by the cumbersome and laborious acquisition and/or labelling of digital data and by having a reduced number of centres dedicated to the scanning and sharing of histopathology slides. While the former is more easily solvable given sufficient time, the latter is more complicated --- anecdotally, it was in 2016 that Yoshua Bengio advised against the training of more radiologists as they would soon (in 5 years) be outcompeted by artificial intelligence \cite{Creative_Destruction_Lab2016-zf} and in 2017 that Andrew Ng first hailed the end of the radiology profession \cite{Ng2017-tz}, rumours which were greatly exaggerated --- indeed, it is quite common to see the performance of \ac{dl} models drop when confronted with independent test sets (data which the model has never seen) when the task is to classify X-ray images (it should nonetheless be noted that radiologists may also perform poorly when looking at data from different centres) \cite{Rajpurkar2021-bj}. The reason for this is fairly illustrative --- most cohorts will be derived from single centres, with the assembly of large, multi-centre cohorts of digitalized image data representing a fairly modern and recent endeavour as they require significant coordination, standards and data storage and sharing protocols --- examples of this include The Cancer Genome Atlas. The reason why multi-centre approaches became desirable is due to the high variability in technical aspects concerning staining, image quality, image compression and scanning conditions and equipment \cite{Van_der_Laak2021-id}.

\paragraph{Underfitting and overfitting.} A classification model is not only concerned with learning a function of the data that returns the correct class; its objective is also to avoid a relatively large drop in predictive performance on data that was not used to train the model due to overfitting (the model learns a representation of the data that is perfect for the training data but relatively useless for new data), and underfitting (the model is far too simple and fails to capture the complex relationships that would allow it to correctly classify new examples, but does not perform any worse on new data). A more mathematically correct terminology for this would be to talk about the bias and variance of a model --- simply put, the bias measures the distance between predictions and target classes, whereas the variance measures differences in the trained model when different sets of data are used (i.e. if subsets of the training data are removed, how much does this impact the classification). Both of these are the two sides of the bias-variance trade-off --- it is not possible to minimize both at once, with low bias models running the risk of overfitting (high variance) and low variance models running the risk of underfitting (high bias) \cite{James2013-py}. While underfitting can be solved with a more complex parametrization (in the case of \ac{dl} this often translates into increasing the number of parameters that the model has to learn), there are different strategies to handle overfitting.

\paragraph{Regularization and dropout.} There are several ways to prevent overfitting when collecting or annotating more data is unfeasible. Some of these ways are more generic and can be applied to a diverse array of \ac{ml} models, whereas others are only applicable to models that work on image data. Regularization techniques belong to the former and their objective is to constrain or shrink coefficients, forcing models to learn much simpler representations of the data. More formally, given an optimization problem where the objective is to find a set of parameters $\theta$ that minimizes $l(X,y)$, an objective or loss function of the data $X$ and its corresponding annotations $y$ that measures the distance between predictions and target annotations (in other words, $\mathrm{arg min}_{\theta}l_{\theta}(X,y)$), the regularization minimizes the magnitude of the coefficients. Typically, this regularization can be based on the $L_1$-norm ($L_1(\theta) = \sum_i^k{|\theta_i|}$) or on the $L_2$-norm ($L_2(\theta) = \sum_i^k{(\theta_i)^2}$) and, in the case of \ac{dl} and other gradient-descent based methods, can be added to the objective function when minimizing it, making it $\mathrm{arg min}_{\theta}(l_{\theta}(X,y)+\lambda L_2(\theta))$, where $\lambda$ is how much the regularization contributes to the value of the loss function \cite{James2013-py}. The reason why I represented the $L_2$, rather than the $L_1$ regularization is because the standard application of gradient-descent algorithms require the loss function to be completely differentiable and $L_1$ is not differentiable whenever any parameter approximates $0$. Dropout is another method that can help in the training of \ac{dl} networks --- for each training iteration, a proportion $p$ of all neurons is randomly selected and set to $0$, forcing other neurons to learn representations which are both sparse and uncorrelated from those of other neurons \cite{Srivastava2014a}. 

\paragraph{Reducing overfitting with image data and in deep-learning.} The wealth of methods to reduce overfitting when working with image data has become considerable, with these methods becoming increasingly complex. A comprehensive description and survey of these methods --- image augmentation methods --- can be found in \cite{Shorten2019-hr}. Here, I focus only on giving a brief overview of these techniques. Simpler manipulations (from a computational perspective) of the image (such as hue and brightness manipulation, addition of Gaussian noise, random depletion of pixels or image regions, and random image cropping, resizing and flipping) are still fairly common, but others involving \ac{dl}-based augmentation techniques have become increasingly popular, including within the field of biomedical research \cite{Shorten2019-hr}. The aforementioned manipulations focus on the changing the image as a digital object and most operations are linear or boolean transformations operating on pixels individually without assuming any form of dependence or structure. These can be useful when, for example, the objective is to train a model that classifies histopathology images and is capable of generalizing to different amounts of stains (assuming these are linearly separable), or to different illumination conditions during digitalization. The other methods, based on \ac{dl}, typically learn how to generate new examples or transform existing examples of training data by learning about the contextual information on the image, similarly to how other \ac{cnn} function.

\paragraph{Early stopping, fine-tuninig and transfer-learning.} Early stopping --- halting the training after some epochs of confirmed convergence --- has also been considered as good method to reduce over-fitting, but results seem mixed in terms of improving the predictive performance \cite{Prechelt2012-xf}. Fine-tuning and transfer learning, two training schemes which rely on \ac{dl} networks pre-trained on large collections of data, have also become increasingly popular. While similar, both rely on slightly different principles --- the focus of fine-tuning is to, given a pre-trained network, slightly adjust all the weights with a relatively low learning-rate in such a way that the predictive performance of the network is maximized for a new task; in other words, rather than randomly initializing the weights in a \ac{ann}, fine-tuning initializes the weights to an already good solution. Transfer learning focuses on training only a set of layers in a pre-trained model under the assumption that the feature representation that non-trained layers derive from the data is sufficient for the problem at hand. While transfer learning has the advantage of being less expensive from a computational perspective, it is also more dependent on how useful the pre-trained feature representation is \cite{transfer-learning-fine-tuning}. 

\paragraph{Class imbalance.} To further alleviate the burden of the biases present in the training data, simple aspects such as class balance should also be considered --- indeed, the existence of considerably more examples of one class compared to the rest may lead to a poorly calibrated model if these proportions are not reflected in the real world \cite{Van_Calster2019-zp}. For instance, if a model was very accurate at discriminating between foxes and dogs but was trained on a dataset featuring many dogs and very few foxes (i.e. a city), it may perform poorly where foxes are much more common than dogs (in a forest). To prevent this, weighing each instance according to its class probability (such that lesser represented classes contribute as much to the learning as more represented classes) can be helpful, with other methods such as data sampling schemes focusing on under-represented or worse performing classes also being popular \cite{Johnson2019-cf}.

\paragraph{Rotation- and scale-invariance.} Finally, it is important to consider that, while \ac{dl} networks try to learn representations that are rotation- and scale-invariant, this invariance is not perfect and its relevance is context-dependent. For instance, if a model is trained for face identification (determining whether a face belongs to an individual), it is often the case that rotation-invariance is not necessary because the input is seldom rotated or flipped, but scale-invariance is convenient because a face can be captured at different distances. However, if the model is trained to detect or segment cells spread over a microscope slide obtained at constant resolution, then rotation-invariance becomes a key concern, whereas variations in scale are virtually non-existent. 

\paragraph{Test time augmentation.} \Ac{tta} uses this information to improve the predictive performance of algorithms during inference --- with \ac{tta}, the data is artificially augmented during inference --- scaled, rotated, flipped, shifted or warped --- in order to create a consensus prediction. While relatively simple, \ac{tta} has been shown to improve predictive performance in classification, segmentation and object detection tasks \cite{Moshkov2020-rc,Shorten2019-hr}.

\subsection{Morphometry --- characterizing cell shape}

Having segmentation masks for different cells can greatly facilitate morphometric tasks. However, it is worth considering that morphometry does not necessarily require accurate segmentation maps --- recent approaches have made use of \ac{dl} to create abstract representations of different cell types \cite{Sommer2017-uj,Wei2021-gw}. Despite this, in this dissertation (Chapters 4 and 5) I use only simpler and relatively more intuitive measures of cellular shape, providing nonetheless a few examples of \ac{dl}-assisted morphometry along this section. 

\subsubsection{Shape descriptors}

Morphometry measures the shape and other tangential aspects of objects (such as size or orientation), a non-trivial task that humans excel at --- for instance, humans can recognize a shape no matter its orientation or position without putting much thought into it; however, if one seeks to define shape in computational terms, it should be clear that shape remains the same no matter its scale, orientation or position \cite{Kendall1977-yx}. Additionally, one can also opt for making certain assumptions; for instance, if we assume all objects are circular, the radius (the average distance between the centre and the edges of the object) would be sufficient to quantify their size, but this can also be done using the area of the object (the number of pixels contained in the segmentation map). On the other hand, under the more general assumptions --- the object is fully connected (there is a path between all the pixels belonging to the object) and little else --- several methods and tools are available. Particularly and as mentioned, Ilastik and CellProfiler are intuitive and easy to use software tools offering a helpful user interface for the segmentation, detection and characterization of different objects in images \cite{Sommer2011-ds,Carpenter2006-hy}. To characterize the shape of different cells, methods and tools such as Ilastik and CellProfiler use different sets of features capable of capturing different aspects of cellular morphology. As an example and to provide a better intuition, I have generated a set of 25 random blobs which I characterize with a few relatively intuitive features regarding size and shape, while using the Amsterdam Library Of Textures \cite{Burghouts2009-ln} to illustrate texture features:

\begin{itemize}
    \item \textbf{Size} --- size can be captured by features such as the \textbf{perimeter} or \textbf{area} of a segmented cell. Other measures include the average distance between the center and the edge of the cell (the \textbf{\ac{cdf}} of a cell; \figref{fig:shape-size-examples}).
    \item \textbf{Shape} --- shape is considerably less trivial to capture than size and many different descriptors are available, some of which have been reviewed elsewhere \cite{Mingqiang2008-wv}. 
    \begin{itemize}
        \item \textbf{Solidity} is characterized as the ratio between a cell and its convex hull (the convex polygon delimiting the pixels in a given cell) --- this provides a rough quantification of the amount of convex defects in a cell (\figref{fig:shape-size-examples})
        \item The \textbf{eccentricity} of a cell provides a quantification of how circular a cell is by calculating the ratio between its major and minor axis of variation (\figref{fig:shape-size-examples})
        \item Features based on \textbf{central moments} (shape descriptors which are translation-invariant and are mathematically defined as $M_{pq} = \sum_y^n\sum_y(x-\bar{x})(y-\bar{y})$, where $M_{pq}$ is the central moment of orders $p$ and $q$, $x$ and $y$ are the coordinates of the pixels in the cell and $\bar{x}$ and $\bar{y}$ are the coordinates of the centroid). These can be used to compute \textbf{moments invariant to rotation and scale} which provide a quantification of shape abnormality \cite{Mingqiang2008-wv} (for example, in the bottom right corner of \figref{fig:shape-size-examples} I show the trend captured by one of these invariant moments, calculated as $(M_{30} + 3 M_{12})^2 + (M_{03} + 3 M_{21})^2$)
        \item Using the \ac{cdf} described above and calculating its standard deviation can also help to characterize shape, while using aspects of the \ac{cuf}, which measures the curvature of the cell edge, can also be helpful
        %\item Zernike moments (quantities based on Zernike polynomials ($Z_{nm} = \frac{n+1}{\pi}\sum_r\sum_{\theta}f(r\mathrm{cos}\theta,r\mathrm{sin}\theta).R_{nm}(r)e^{nm\theta}$, where $R_{nm}(r)$ is the orthogonal radial polynomial as described in \cite{Mingqiang2008-wv} and pixels are the input as sets of angles and normalised radii in relation to the centroid as $\theta$ and $r \ge 1$, respectively) which are robust, rotation- and scale-invariant and avoid information redundancy).
    \end{itemize}
    \item \textbf{Texture} can be quantified using the \textbf{\ac{glcm}} of an object. The \ac{glcm} is, for a given distance in pixels $d$ and a given angle $\alpha$, a matrix which quantifies how many times in the object grey-level $i$ occurs at a distance $p$ and angle $\alpha$ from grey-level $j$. In practical terms, the grey-levels are usually quantized into $k$ distinct values and the \ac{glcm} is coded as a four-dimensional array and is normalized for each combination of $d$ and $\alpha$,  $P_{ijd\alpha}$. Using these features, one can assess for given $d$ and $\alpha$ whether the texture of an object has higher contrast (larger local differences in grey-levels: $\sum_{i,j=0}^{k}{P_{ij}(i-j)^2}$) or correlation (relatively conserved local structure: $\sum_{ij=0}^{k}P_{ij}\frac{(i-\mu_i)(j-\mu_j)}{\sqrt{\sigma_i^2\sigma_j^2}}$, where $\mu$ and $\sigma$ the mean and standard deviation, respectively; \figref{fig:texture-examples})
    \item \textbf{Colour intensity} is perhaps the simplest of features to explain --- it generally characterizes whether the colour is distributed more or less homogenously through the cell or if there are specific shifts given an expected colour distribution.
\end{itemize}

\begin{figure}[!ht]
	\placefigure{intro/size-shape-examples}
	\placecaption{Examples of the same 25 blobs sorted by different morphometric features.}
	\label{fig:shape-size-examples}
\end{figure}

\begin{figure}[!hb]
	\placefigure{intro/texture-examples}
	\placecaption{Examples of the same 12 texture images from the Amsterdam Library Of Textures \cite{Burghouts2009-ln} sorted by different texture features.}
	\label{fig:texture-examples}
\end{figure}

The relation between these features and the characterization of blood cells is evident --- as mentioned in Section 1.1.2, each cell type has specific features --- while lymphocytes are generally smaller, monocytes have a tendency to be considerably larger and less round \cite{Bain2005-zg} (\figref{fig:blood-cell-examples}). Additionally, the morphology of the nucleus is also of great relevance for the quick identification of granulocytes (whose morphology can be relevantly quantified by their solidity and size), together with their relatively high cytoplasmic granularity (which can be characterized by texture features). Finally, \ac{rbc} can be smaller or larger and their shape can be altered in specific conditions \cite{Aslinia2006-en,Bain2014-oc}. In summary, these features can be used to characterize cells in a way that allows their use in quantitative analyses. Indeed, while some works today still rely on human expert-based characterization of cells \cite{Nagata2020-lh}, it is easy to see why automated methods are increasingly gaining traction.

\subsubsection{Learning features with convolutional neural networks}

Other recent works have used \ac{dl} classifiers to characterize cells --- a \ac{cnn} that has been trained to correctly identify specific objects or conditions in images implicitly creates accurate and informative numerical representations of these objects or conditions. Such \ac{dl} applications are reviewed elsewhere \cite{Moen2019-py,Pratapa2021-ee} and are not used during this work; consequently, I only mention a few interesting applications. 

\etal{Fu} trained a pan-cancer detection algorithm based on histopathology images and used these features to predict driver mutations, changes to gene expression, lymphocyte infiltration and survival. They additionally showed that whole genome duplication events could be detected using the feature representation learned to detect cancer type and tissue, determining that the size of the nucleus was larger in these events, with specific histological signatures being observed for cancers with different driver mutations \cite{Fu2020-mx}. Other works have also showed how unsupervised \ac{dl} approaches --- particularly autoencoders, an \ac{ann} architecture which learns how to map images to a low-dimensional representation that is sufficient to reconstruct the input image --- to characterize cells \cite{Sommer2017-uj,Wei2021-gw}. \etal{Wei} showed how these technologies were relatively applicable across datasets, noting that these autoencoder-derived features could easily be used in classification tasks \cite{Wei2021-gw}, whereas \etal{Sommer} used a similar approach to show how changes in cell state were captured by such unsupervised approaches \cite{Sommer2017-uj}.

\subsection{Digital cytomorphology in blood cancers}

The morphometric technologies described above have already been applied to some instances in the study of blood cancers. \etal{Eckardt} showed that using relatively small sections of bone marrow slides and \ac{dl} algorithms was sufficient to not only predict \ac{aml} but also the presence of mutations in \ac{npm1}, the most commonly mutated gene in \ac{aml} \cite{Eckardt2021-fb}. In this work, \etal{Eckardt} also determined that this specific \ac{aml} subtype was associated with specific morphological signatures through the use of occlusion sensitivity maps, a method that detects which regions of the image were responsible for specific predictions in \ac{cnn}-based methods \cite{Zeiler2013-jk}. In recent years, very large collections of \ac{wbc} in the bone marrow and in peripheral blood have been annotated and successfully used in prediction, showing that \ac{cnn} models can be used to accurately predict cell type \cite{Matek2019-ld,Matek2021-mp,Kouzehkanan2021-mz}, highlighting the power of large sets of training data of labelled images. Another work, focusing on both \ac{wbc} and \ac{rbc} in blood smears, annotated each cell according to its dysplastic status, showing that \ac{cnn} models could predict this with high accuracy \cite{Mori2020-dn}.

Looking specifically into works that seek to use cell morphology to better understand \ac{mds} and as mentioned earlier, two publications have used machine-learning to assess how cytomorphology can be associated with clinical aspects of \ac{mds} and genetics. While focusing on bone marrow cytomorphology and disregarding \ac{wbs}, important discoveries were made --- \etal{Nagata} used human annotation of cytomorphology combined with machine-learning to reveal how specific cytomorphological profiles present in the bone marrow are associated with high- or low-risk \ac{mds}, showing additionally that cytomorphological profiles were associated with specific genetic signatures (for instance, they show that the presence of both trilineage dysplasia and pancytopenia is associated with a higher probability of exclusive \ac{tet2} mutation, while monocytosis as evaluated by a \ac{cbc} was more prevalent in both \ac{tet2} and \ac{srsf2} mutants) \cite{Nagata2020-lh}. \etal{Bruck}, used a more computationally complex protocol to analyse the cytomorphology of the bone marrow and how it can be used to predict between healthy individuals, individuals with \ac{mds} and those with \ac{mpn}, as well as the presence of specific mutations. In it, they analyse bone marrow slides using a classification network, similar to what has already been done with histopathology by \etal{Fu} \cite{Fu2020-mx}. However, it should be noted that \etal{Bruck} do not validate their approach on an independent test set, relying instead on cross-validated metrics \cite{Bruck2021-fx}. 

\section{The aims of this dissertation and outline}

Changes to the haematopoietic system have a wide range of consequences, leading to changes in the evolutionary landscape of the bone marrow and to alterations in the cytomorphology of circulating blood cells. In this dissertation, I present ways through which these alterations can be studied through blood, focusing in particular on the computational study of evolutionary and cytomorphological aspects (\figref{fig:overview}). More concretely, I have divided the work I developed into two distinct avenues --- "The longitudinal dynamics and natural history of clonal haematopoiesis" (Chapters 2 and 3) and "The cytomorphology of myelodysplastic syndromes" (Chapters 4 and 5). More concretely, I cover the following:

\begin{itemize}
	\item Chapter 2 - I show a theory of somatic evolution of the haematopoietic system based on relatively simple assumptions of growth can be used to recapitulate dynamic aspects of clonal evolution such as fitness and age at clone onset in 1) longitudinal samples simulated from Wright-Fisher simulations and 2) phylogenetic trees built from Wright-Fisher simulations
	\item Chapter 3 - I show how applying these methods to a cohort of over 300 healthy individuals allows us to understand the effect of different mutations on clonal evolution during old age and how \ac{ch} evolves through life in the context of an increasingly oligoclonal landscape
	\item Chapter 4 - I show how image analysis can be used to automatically characterize thousands of cells \textit{per} \ac{wbs} and characterize them individually
	\item Chapter 5 - I show how the blood cell detection and characterization protocol developed in Chapter 4 can be used to discriminate between different diseases, while also enabling the identification of specific, genetically-characterised subtypes of \ac{mds}.
\end{itemize}

\begin{figure}[!hb]
  \placefigure{intro/overview}
  \placecaption{Schematic representation of the two projects included in this dissertation --- "the longitudinal dynamics and natural history of clonal haematopoiesis" (Chapters 2 and 3), where I study the evolutionary aspects of clonal haematopoiesis using longitudinal targeted sequencing and single-cell phylogenies, and "the cytomorphology of myelodysplastic syndromes" (Chapters 4 and 5), where I use computer vision and machine-learning methods to better derive cytomorphological markers of myelodysplastic syndromes from whole blood slides.}
  \label{fig:overview}
\end{figure}
